<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <meta name="generator" content="pandoc" />

        <meta name="author" content="Tao Wu" />
        <meta name="author" content="Xue-Song Liu (Corresponding author)" />
    
        <meta name="date" content="2024-01-22" />
    
    <title>Precise metabolic dependencies of cancer through deep learning and validations</title>

        <script src="report_main_files/header-attrs-2.22/header-attrs.js"></script>
        <script src="report_main_files/jquery-3.6.0/jquery-3.6.0.min.js"></script>
        <meta name="viewport" content="width=device-width, initial-scale=1" />
        <link href="report_main_files/bootstrap-3.3.7/css/bootstrap.min.css" rel="stylesheet" />
        <script src="report_main_files/bootstrap-3.3.7/js/bootstrap.min.js"></script>
        <script src="report_main_files/navigation-1.1/tabsets.js"></script>
        <link href="report_main_files/readthedown-0.1/readthedown.css" rel="stylesheet" />
        <link href="report_main_files/readthedown-0.1/readthedown_fonts_embed.css" rel="stylesheet" />
        <script src="report_main_files/readthedown-0.1/readthedown.js"></script>
        <script src="report_main_files/core-js-2.5.3/shim.min.js"></script>
        <script src="report_main_files/react-17.0.0/react.min.js"></script>
        <script src="report_main_files/react-17.0.0/react-dom.min.js"></script>
        <script src="report_main_files/reactwidget-1.0.0/react-tools.js"></script>
        <script src="report_main_files/htmlwidgets-1.6.2/htmlwidgets.js"></script>
        <link href="report_main_files/reactable-0.4.4/reactable.css" rel="stylesheet" />
        <script src="report_main_files/reactable-binding-0.4.4/reactable.js"></script>
    
    
        <style type="text/css">code{white-space: pre;}</style>
    <style type="text/css">
      pre > code.sourceCode { white-space: pre; position: relative; }
      pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
      pre > code.sourceCode > span:empty { height: 1.2em; }
      .sourceCode { overflow: visible; }
      code.sourceCode > span { color: inherit; text-decoration: inherit; }
      div.sourceCode { margin: 1em 0; }
      pre.sourceCode { margin: 0; }
      @media screen {
      div.sourceCode { overflow: auto; }
      }
      @media print {
      pre > code.sourceCode { white-space: pre-wrap; }
      pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
      }
      pre.numberSource code
        { counter-reset: source-line 0; }
      pre.numberSource code > span
        { position: relative; left: -4em; counter-increment: source-line; }
      pre.numberSource code > span > a:first-child::before
        { content: counter(source-line);
          position: relative; left: -1em; text-align: right; vertical-align: baseline;
          border: none; display: inline-block;
          -webkit-touch-callout: none; -webkit-user-select: none;
          -khtml-user-select: none; -moz-user-select: none;
          -ms-user-select: none; user-select: none;
          padding: 0 4px; width: 4em;
          background-color: #ffffff;
          color: #a0a0a0;
        }
      pre.numberSource { margin-left: 3em; border-left: 1px solid #a0a0a0;  padding-left: 4px; }
      div.sourceCode
        { color: #1f1c1b; background-color: #ffffff; }
      @media screen {
      pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
      }
      code span { color: #1f1c1b; } /* Normal */
      code span.al { color: #bf0303; background-color: #f7e6e6; font-weight: bold; } /* Alert */
      code span.an { color: #ca60ca; } /* Annotation */
      code span.at { color: #0057ae; } /* Attribute */
      code span.bn { color: #b08000; } /* BaseN */
      code span.bu { color: #644a9b; font-weight: bold; } /* BuiltIn */
      code span.cf { color: #1f1c1b; font-weight: bold; } /* ControlFlow */
      code span.ch { color: #924c9d; } /* Char */
      code span.cn { color: #aa5500; } /* Constant */
      code span.co { color: #898887; } /* Comment */
      code span.cv { color: #0095ff; } /* CommentVar */
      code span.do { color: #607880; } /* Documentation */
      code span.dt { color: #0057ae; } /* DataType */
      code span.dv { color: #b08000; } /* DecVal */
      code span.er { color: #bf0303; text-decoration: underline; } /* Error */
      code span.ex { color: #0095ff; font-weight: bold; } /* Extension */
      code span.fl { color: #b08000; } /* Float */
      code span.fu { color: #644a9b; } /* Function */
      code span.im { color: #ff5500; } /* Import */
      code span.in { color: #b08000; } /* Information */
      code span.kw { color: #1f1c1b; font-weight: bold; } /* Keyword */
      code span.op { color: #1f1c1b; } /* Operator */
      code span.ot { color: #006e28; } /* Other */
      code span.pp { color: #006e28; } /* Preprocessor */
      code span.re { color: #0057ae; background-color: #e0e9f8; } /* RegionMarker */
      code span.sc { color: #3daee9; } /* SpecialChar */
      code span.ss { color: #ff5500; } /* SpecialString */
      code span.st { color: #bf0303; } /* String */
      code span.va { color: #0057ae; } /* Variable */
      code span.vs { color: #bf0303; } /* VerbatimString */
      code span.wa { color: #bf0303; } /* Warning */
    </style>
    
    
    <!-- tabsets -->
    <script>
      $(document).ready(function () {
	  window.buildTabsets("toc");
      });
      $(document).ready(function () {
	  $('.tabset-dropdown > .nav-tabs > li').click(function () {
	      $(this).parent().toggleClass('nav-tabs-open')
	  });
      });
    </script>

    <!-- code folding -->
    
    <!-- code download -->
    
    <!-- tabsets dropdown -->

    <style type="text/css">
      .tabset-dropdown > .nav-tabs {
	  display: inline-table;
	  max-height: 500px;
	  min-height: 44px;
	  overflow-y: auto;
	  background: white;
	  border: 1px solid #ddd;
	  border-radius: 4px;
      }
      
      .tabset-dropdown > .nav-tabs > li.active:before {
	  content: "";
	  font-family: 'Glyphicons Halflings';
	  display: inline-block;
	  padding: 10px;
	  border-right: 1px solid #ddd;
      }
      
      .tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
	  content: "&#xe258;";
	  border: none;
      }
      
      .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
	  content: "";
	  font-family: 'Glyphicons Halflings';
	  display: inline-block;
	  padding: 10px;
	  border-right: 1px solid #ddd;
      }
      
      .tabset-dropdown > .nav-tabs > li.active {
	  display: block;
      }

      .tabset-dropdown > .nav-tabs > li.active a {
  	  padding: 0 15px !important;
      }

      .tabset-dropdown > .nav-tabs > li > a,
      .tabset-dropdown > .nav-tabs > li > a:focus,
      .tabset-dropdown > .nav-tabs > li > a:hover {
	  border: none;
	  display: inline-block;
	  border-radius: 4px;
	  background-color: transparent;
      }
      
      .tabset-dropdown > .nav-tabs.nav-tabs-open > li {
	  display: block;
	  float: none;
      }
      
      .tabset-dropdown > .nav-tabs > li {
	  display: none;
	  margin-left: 0 !important;
      }
    </style>
    
</head>

<body class="preload">

   	
         <!-- readthedown start -->   
   <div id="content" data-toggle="wy-nav-shift">
     <nav id="nav-top" role="navigation" aria-label="top navigation">
       <a role="button" href="#" data-toggle="wy-nav-top"><span class="glyphicon glyphicon-menu-hamburger"></span></a>
     </nav>
         
   
        
      <h1 class="title">Precise metabolic dependencies of cancer through
deep learning and validations</h1>
      
         <!-- readthedown authors -->
   <div id="sidebar">
    <h2><a href="#content">Precise metabolic dependencies of cancer
through deep learning and validations</a></h2>
    <div id="toc">
      <ul>
      <li><a href="#dependencies"
      id="toc-dependencies">Dependencies</a></li>
      <li><a href="#the-deepmeta-framework"
      id="toc-the-deepmeta-framework">The DeepMeta framework</a></li>
      <li><a href="#model-performance-evaluation"
      id="toc-model-performance-evaluation">Model performance
      evaluation</a></li>
      <li><a href="#tcga-analysis" id="toc-tcga-analysis">TCGA
      analysis</a>
      <ul>
      <li><a href="#pan-cancer-metabolic-dependency"
      id="toc-pan-cancer-metabolic-dependency">Pan-cancer metabolic
      dependency</a></li>
      <li><a
      href="#metabolic-vulnerabilities-of-cancers-with-undruggable-driver-genes"
      id="toc-metabolic-vulnerabilities-of-cancers-with-undruggable-driver-genes">Metabolic
      vulnerabilities of cancers with undruggable driver genes</a></li>
      </ul></li>
      </ul>
    </div>
    <div id="postamble" data-toggle="wy-nav-shift" class="status">
                  <p class="author"><span class="glyphicon glyphicon-user"></span> Tao
Wu</p>
                        <p class="author"><span class="glyphicon glyphicon-user"></span> Xue-Song
Liu (Corresponding author)</p>
                        <p class="date"><span class="glyphicon glyphicon-calendar"></span> 2024-01-22</p>
          </div>
   </div>
     

   
      
   
<!-- Don't indent these lines or it will mess pre blocks indentation --> 
<div id="main">
<div id="dependencies" class="section level1">
<h1>Dependencies</h1>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(yardstick)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggprism)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(parallel)</span></code></pre></div>
</div>
<div id="the-deepmeta-framework" class="section level1">
<h1>The DeepMeta framework</h1>
<p>The metabolic vulnerability of each specific sample is dependent on
the sample status and the metabolic network context. Here, we present
the DeepMeta framework to predict the metabolic gene dependence based on
the metabolic pathway information characterized by enzyme networks and
sample status information defined by gene expression profile. Thus,
DeepMeta has two inputs, the sample-specific enzyme network and the gene
expression information (<code>Fig 1a</code>).</p>
<p><img
src="https://picgo-wutao.oss-cn-shanghai.aliyuncs.com/image-20240122090715501.png" /></p>
<p>We used the graph attention network (GAT) module to extract
information from the sample-specific enzyme network to obtain the
embedding of the metabolic enzyme gene node, and used the fully
connected neuron network module to extract information from the
expression profiles of cells to get the sample state embedding (see
Methods). Here, the features of gene node in metabolic enzyme network
was implemented by binary vector which denotes the involvement of the
gene in chemical and genetic perturbation (CPG) gene sets5. These two
parts of information are then merged to predict metabolic dependencies
for each specific samples (<code>Fig 1a</code>). DeepMeta was trained on
cell line CRISPR-Cas9 screening data of DepMap6 (23Q2 version). We used
the post-Chronos7 gene dependency probability score to set up a
classification problem for our model. The dependency probabilities value
is between 0 and 1. Sample-gene pair with dependency score &gt;0.5 were
defined as the positive class and the sample-gene pair with dependency
score &lt;=0.5 were defined as the negative class. We choose cancer cell
lines with matched cancer type or tissue-specific Genome-Scale Metabolic
Models (GSMs) and genes existing in corresponding enzyme network and
positive in at least 3 cell lines. In together, 343,029 (760 cell lines
and 1063 genes) labeled samples were available. We randomly partitioned
cell lines into training (90%) and testing (10%) sets, and in the
training set, we randomly sampled 70% of data for training, 30% to
monitor training process and tune hyper-parameters.</p>
<p>To access the impact of different parts on the performance of the
model, we conducted ablation experiments, and train new models using
random metabolic node features, without the enzyme network or random
sample expression as input, then compared the performance of these model
with DeepMeta using ten-folds cross validation. First, we calculated
metrics in every CV fold:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>get_roc <span class="ot">&lt;-</span> <span class="cf">function</span>(dt,<span class="at">need=</span><span class="st">&quot;all&quot;</span>){</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>  dt <span class="ot">&lt;-</span> dt <span class="sc">%&gt;%</span> </span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">truth =</span> <span class="fu">ifelse</span>(label <span class="sc">==</span> <span class="dv">1</span>, <span class="st">&quot;Class1&quot;</span>,<span class="st">&quot;Class2&quot;</span>),</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">pred_label =</span> <span class="fu">ifelse</span>(preds <span class="sc">==</span> <span class="dv">1</span>, <span class="st">&quot;Class1&quot;</span>,<span class="st">&quot;Class2&quot;</span>))</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>  dt<span class="sc">$</span>truth <span class="ot">&lt;-</span> <span class="fu">factor</span>(dt<span class="sc">$</span>truth)</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>  dt<span class="sc">$</span>pred_label <span class="ot">&lt;-</span> <span class="fu">factor</span>(dt<span class="sc">$</span>pred_label)</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>  f1 <span class="ot">&lt;-</span> <span class="fu">try</span>(<span class="fu">f_meas</span>(dt,truth,pred_label)[<span class="st">&quot;.estimate&quot;</span>] <span class="sc">%&gt;%</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>              <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> <span class="fu">unname</span>() <span class="sc">%&gt;%</span> <span class="fu">round</span>(.,<span class="dv">2</span>),<span class="at">silent =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>  kappa <span class="ot">&lt;-</span> <span class="fu">try</span>(<span class="fu">kap</span>(dt,truth,pred_label)[<span class="st">&quot;.estimate&quot;</span>] <span class="sc">%&gt;%</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> <span class="fu">unname</span>() <span class="sc">%&gt;%</span> <span class="fu">round</span>(.,<span class="dv">2</span>),<span class="at">silent =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>  f1 <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="st">&#39;try-error&#39;</span> <span class="sc">%in%</span> <span class="fu">class</span>(f1),<span class="cn">NA</span>,f1)</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>  kappa <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="st">&#39;try-error&#39;</span> <span class="sc">%in%</span> <span class="fu">class</span>(kappa),<span class="cn">NA</span>,kappa)</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (need <span class="sc">==</span> <span class="st">&quot;all&quot;</span>){</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>    roc <span class="ot">&lt;-</span>  <span class="fu">roc_auc</span>(dt, truth, preds_raw)[<span class="st">&quot;.estimate&quot;</span>] <span class="sc">%&gt;%</span> </span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>      <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> <span class="fu">unname</span>() <span class="sc">%&gt;%</span> <span class="fu">round</span>(.,<span class="dv">2</span>)</span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>    pr <span class="ot">&lt;-</span>  <span class="fu">pr_auc</span>(dt, truth, preds_raw)[<span class="st">&quot;.estimate&quot;</span>] <span class="sc">%&gt;%</span></span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>      <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> <span class="fu">unname</span>() <span class="sc">%&gt;%</span> <span class="fu">round</span>(.,<span class="dv">2</span>)</span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">c</span>(roc,pr,f1,kappa))</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>  }<span class="cf">else</span>{</span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>    res <span class="ot">&lt;-</span> <span class="cf">switch</span>(</span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>      need,</span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>      <span class="st">&quot;roc&quot;</span> <span class="ot">=</span> <span class="fu">roc_auc</span>(dt, truth, preds_raw)[<span class="st">&quot;.estimate&quot;</span>] <span class="sc">%&gt;%</span> </span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a>        <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> <span class="fu">unname</span>() <span class="sc">%&gt;%</span> <span class="fu">round</span>(.,<span class="dv">2</span>),</span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a>      <span class="st">&quot;pr&quot;</span> <span class="ot">=</span> <span class="fu">pr_auc</span>(dt, truth, preds_raw)[<span class="st">&quot;.estimate&quot;</span>] <span class="sc">%&gt;%</span></span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a>        <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> <span class="fu">unname</span>() <span class="sc">%&gt;%</span> <span class="fu">round</span>(.,<span class="dv">2</span>),</span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a>      <span class="st">&quot;f1&quot;</span> <span class="ot">=</span> f1,</span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a>      <span class="st">&quot;kappa&quot;</span> <span class="ot">=</span> kappa</span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(res)</span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-33"><a href="#cb2-33" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-34"><a href="#cb2-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-35"><a href="#cb2-35" aria-hidden="true" tabindex="-1"></a>get_gene_metric <span class="ot">&lt;-</span> <span class="cf">function</span>(dt,gene_name,need){</span>
<span id="cb2-36"><a href="#cb2-36" aria-hidden="true" tabindex="-1"></a>  dt <span class="ot">&lt;-</span> dt <span class="sc">%&gt;%</span> </span>
<span id="cb2-37"><a href="#cb2-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">truth =</span> <span class="fu">ifelse</span>(label <span class="sc">==</span> <span class="dv">1</span>, <span class="st">&quot;Class1&quot;</span>,<span class="st">&quot;Class2&quot;</span>),</span>
<span id="cb2-38"><a href="#cb2-38" aria-hidden="true" tabindex="-1"></a>           <span class="at">pred_label =</span> <span class="fu">ifelse</span>(preds <span class="sc">==</span> <span class="dv">1</span>, <span class="st">&quot;Class1&quot;</span>,<span class="st">&quot;Class2&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb2-39"><a href="#cb2-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(genes <span class="sc">==</span> gene_name)</span>
<span id="cb2-40"><a href="#cb2-40" aria-hidden="true" tabindex="-1"></a>  dt<span class="sc">$</span>truth <span class="ot">&lt;-</span> <span class="fu">factor</span>(dt<span class="sc">$</span>truth)</span>
<span id="cb2-41"><a href="#cb2-41" aria-hidden="true" tabindex="-1"></a>  dt<span class="sc">$</span>pred_label <span class="ot">&lt;-</span> <span class="fu">factor</span>(dt<span class="sc">$</span>pred_label)</span>
<span id="cb2-42"><a href="#cb2-42" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-43"><a href="#cb2-43" aria-hidden="true" tabindex="-1"></a>  f1 <span class="ot">&lt;-</span> <span class="fu">try</span>(<span class="fu">f_meas</span>(dt,truth,pred_label)[<span class="st">&quot;.estimate&quot;</span>] <span class="sc">%&gt;%</span></span>
<span id="cb2-44"><a href="#cb2-44" aria-hidden="true" tabindex="-1"></a>              <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> <span class="fu">unname</span>() <span class="sc">%&gt;%</span> <span class="fu">round</span>(.,<span class="dv">2</span>),<span class="at">silent =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-45"><a href="#cb2-45" aria-hidden="true" tabindex="-1"></a>  kappa <span class="ot">&lt;-</span> <span class="fu">try</span>(<span class="fu">kap</span>(dt,truth,pred_label)[<span class="st">&quot;.estimate&quot;</span>] <span class="sc">%&gt;%</span></span>
<span id="cb2-46"><a href="#cb2-46" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> <span class="fu">unname</span>() <span class="sc">%&gt;%</span> <span class="fu">round</span>(.,<span class="dv">2</span>),<span class="at">silent =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-47"><a href="#cb2-47" aria-hidden="true" tabindex="-1"></a>  f1 <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="st">&#39;try-error&#39;</span> <span class="sc">%in%</span> <span class="fu">class</span>(f1),<span class="cn">NA</span>,f1)</span>
<span id="cb2-48"><a href="#cb2-48" aria-hidden="true" tabindex="-1"></a>  kappa <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="st">&#39;try-error&#39;</span> <span class="sc">%in%</span> <span class="fu">class</span>(kappa),<span class="cn">NA</span>,kappa)</span>
<span id="cb2-49"><a href="#cb2-49" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-50"><a href="#cb2-50" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="cf">switch</span>(</span>
<span id="cb2-51"><a href="#cb2-51" aria-hidden="true" tabindex="-1"></a>    need,</span>
<span id="cb2-52"><a href="#cb2-52" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;roc&quot;</span> <span class="ot">=</span> <span class="fu">roc_auc</span>(dt, truth, preds_raw)[<span class="st">&quot;.estimate&quot;</span>] <span class="sc">%&gt;%</span> </span>
<span id="cb2-53"><a href="#cb2-53" aria-hidden="true" tabindex="-1"></a>      <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> <span class="fu">unname</span>() <span class="sc">%&gt;%</span> <span class="fu">round</span>(.,<span class="dv">2</span>),</span>
<span id="cb2-54"><a href="#cb2-54" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;pr&quot;</span> <span class="ot">=</span> <span class="fu">pr_auc</span>(dt, truth, preds_raw)[<span class="st">&quot;.estimate&quot;</span>] <span class="sc">%&gt;%</span></span>
<span id="cb2-55"><a href="#cb2-55" aria-hidden="true" tabindex="-1"></a>      <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> <span class="fu">unname</span>() <span class="sc">%&gt;%</span> <span class="fu">round</span>(.,<span class="dv">2</span>),</span>
<span id="cb2-56"><a href="#cb2-56" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;f1&quot;</span> <span class="ot">=</span> f1,</span>
<span id="cb2-57"><a href="#cb2-57" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;kappa&quot;</span> <span class="ot">=</span> kappa</span>
<span id="cb2-58"><a href="#cb2-58" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb2-59"><a href="#cb2-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb2-60"><a href="#cb2-60" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-61"><a href="#cb2-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-62"><a href="#cb2-62" aria-hidden="true" tabindex="-1"></a>test_dtV2 <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/test_dtV2.rds&quot;</span>)</span>
<span id="cb2-63"><a href="#cb2-63" aria-hidden="true" tabindex="-1"></a>train_dtV2 <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/train_dtV2.rds&quot;</span>)</span>
<span id="cb2-64"><a href="#cb2-64" aria-hidden="true" tabindex="-1"></a>all_dt <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(test_dtV2,train_dtV2)</span>
<span id="cb2-65"><a href="#cb2-65" aria-hidden="true" tabindex="-1"></a>all_dt_summ <span class="ot">&lt;-</span> all_dt <span class="sc">%&gt;%</span> </span>
<span id="cb2-66"><a href="#cb2-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(gene) <span class="sc">%&gt;%</span> </span>
<span id="cb2-67"><a href="#cb2-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">pos_counts =</span> <span class="fu">mean</span>(is_dep <span class="sc">==</span> <span class="dv">1</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb2-68"><a href="#cb2-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb2-69"><a href="#cb2-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(pos_counts <span class="sc">&lt;</span> <span class="fl">0.9</span> <span class="sc">&amp;</span> pos_counts <span class="sc">&gt;</span> <span class="fl">0.1</span>)</span>
<span id="cb2-70"><a href="#cb2-70" aria-hidden="true" tabindex="-1"></a>all_dt_summ <span class="ot">&lt;-</span> <span class="fu">left_join</span>(</span>
<span id="cb2-71"><a href="#cb2-71" aria-hidden="true" tabindex="-1"></a>  all_dt_summ,</span>
<span id="cb2-72"><a href="#cb2-72" aria-hidden="true" tabindex="-1"></a>  all_dt <span class="sc">%&gt;%</span> <span class="fu">select</span>(id,gene) <span class="sc">%&gt;%</span> <span class="fu">distinct_all</span>()</span>
<span id="cb2-73"><a href="#cb2-73" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-74"><a href="#cb2-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-75"><a href="#cb2-75" aria-hidden="true" tabindex="-1"></a>all_cv <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;own_model&quot;</span>,<span class="st">&quot;random_feat&quot;</span>,<span class="st">&quot;random_exp&quot;</span>,</span>
<span id="cb2-76"><a href="#cb2-76" aria-hidden="true" tabindex="-1"></a>            <span class="st">&quot;no_gnn&quot;</span>,<span class="st">&quot;gnn_model_random_y&quot;</span>,</span>
<span id="cb2-77"><a href="#cb2-77" aria-hidden="true" tabindex="-1"></a>            <span class="st">&quot;ml_logistic&quot;</span>,<span class="st">&quot;ml_rf&quot;</span>,<span class="st">&quot;ml_svm&quot;</span>)</span>
<span id="cb2-78"><a href="#cb2-78" aria-hidden="true" tabindex="-1"></a>cv_res <span class="ot">&lt;-</span> <span class="fu">vector</span>(<span class="st">&quot;list&quot;</span>,<span class="dv">8</span>)</span>
<span id="cb2-79"><a href="#cb2-79" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>){</span>
<span id="cb2-80"><a href="#cb2-80" aria-hidden="true" tabindex="-1"></a>  sub_res <span class="ot">&lt;-</span> <span class="fu">vector</span>(<span class="st">&quot;list&quot;</span>,<span class="dv">10</span>)</span>
<span id="cb2-81"><a href="#cb2-81" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>){</span>
<span id="cb2-82"><a href="#cb2-82" aria-hidden="true" tabindex="-1"></a>    pre <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">paste0</span>(<span class="st">&quot;~/DeepMeta/data/cv/&quot;</span>,</span>
<span id="cb2-83"><a href="#cb2-83" aria-hidden="true" tabindex="-1"></a>                           all_cv[i],<span class="st">&quot;/fold_&quot;</span>,j<span class="dv">-1</span>,<span class="st">&quot;.csv&quot;</span>)) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>X)</span>
<span id="cb2-84"><a href="#cb2-84" aria-hidden="true" tabindex="-1"></a>    pre_roc <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(i <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">7</span>),<span class="fu">get_roc</span>(pre,<span class="at">need =</span> <span class="st">&quot;roc&quot;</span>),<span class="cn">NA</span>)</span>
<span id="cb2-85"><a href="#cb2-85" aria-hidden="true" tabindex="-1"></a>    pre_pr <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(i <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">7</span>),<span class="fu">get_roc</span>(pre,<span class="at">need =</span> <span class="st">&quot;pr&quot;</span>),<span class="cn">NA</span>)</span>
<span id="cb2-86"><a href="#cb2-86" aria-hidden="true" tabindex="-1"></a>    pre_f1 <span class="ot">&lt;-</span> <span class="fu">get_roc</span>(pre,<span class="at">need =</span> <span class="st">&quot;f1&quot;</span>)</span>
<span id="cb2-87"><a href="#cb2-87" aria-hidden="true" tabindex="-1"></a>    pre_kap <span class="ot">&lt;-</span> <span class="fu">get_roc</span>(pre,<span class="at">need =</span> <span class="st">&quot;kappa&quot;</span>)</span>
<span id="cb2-88"><a href="#cb2-88" aria-hidden="true" tabindex="-1"></a>    dt <span class="ot">&lt;-</span> pre <span class="sc">%&gt;%</span> </span>
<span id="cb2-89"><a href="#cb2-89" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(genes <span class="sc">%in%</span> all_dt_summ<span class="sc">$</span>id) <span class="sc">%&gt;%</span> </span>
<span id="cb2-90"><a href="#cb2-90" aria-hidden="true" tabindex="-1"></a>      <span class="fu">group_by</span>(genes) <span class="sc">%&gt;%</span></span>
<span id="cb2-91"><a href="#cb2-91" aria-hidden="true" tabindex="-1"></a>      <span class="fu">summarise</span>(<span class="at">pos_counts =</span> <span class="fu">sum</span>(label <span class="sc">==</span> <span class="dv">1</span>),</span>
<span id="cb2-92"><a href="#cb2-92" aria-hidden="true" tabindex="-1"></a>                <span class="at">neg_counts =</span> <span class="fu">sum</span>(label <span class="sc">==</span> <span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb2-93"><a href="#cb2-93" aria-hidden="true" tabindex="-1"></a>      <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> <span class="fu">filter</span>(pos_counts<span class="sc">&gt;</span><span class="dv">0</span> <span class="sc">&amp;</span> neg_counts <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb2-94"><a href="#cb2-94" aria-hidden="true" tabindex="-1"></a>    pre_gene_roc <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene =</span> <span class="fu">unique</span>(dt<span class="sc">$</span>genes)) <span class="sc">%&gt;%</span></span>
<span id="cb2-95"><a href="#cb2-95" aria-hidden="true" tabindex="-1"></a>      <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb2-96"><a href="#cb2-96" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">roc =</span> <span class="fu">ifelse</span>(i <span class="sc">==</span> <span class="dv">8</span>,<span class="cn">NA</span>,</span>
<span id="cb2-97"><a href="#cb2-97" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">get_gene_metric</span>(pre,<span class="at">gene_name =</span> gene,<span class="st">&quot;roc&quot;</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb2-98"><a href="#cb2-98" aria-hidden="true" tabindex="-1"></a>      <span class="fu">ungroup</span>()</span>
<span id="cb2-99"><a href="#cb2-99" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb2-100"><a href="#cb2-100" aria-hidden="true" tabindex="-1"></a>    sub_res[[j]] <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb2-101"><a href="#cb2-101" aria-hidden="true" tabindex="-1"></a>      <span class="at">fold =</span> <span class="fu">paste0</span>(<span class="st">&quot;Fold-&quot;</span>,j),</span>
<span id="cb2-102"><a href="#cb2-102" aria-hidden="true" tabindex="-1"></a>      <span class="at">ROC =</span> pre_roc,</span>
<span id="cb2-103"><a href="#cb2-103" aria-hidden="true" tabindex="-1"></a>      <span class="at">PR =</span> pre_pr,</span>
<span id="cb2-104"><a href="#cb2-104" aria-hidden="true" tabindex="-1"></a>      <span class="at">F1 =</span> pre_f1,</span>
<span id="cb2-105"><a href="#cb2-105" aria-hidden="true" tabindex="-1"></a>      <span class="at">Kappa =</span> pre_kap,</span>
<span id="cb2-106"><a href="#cb2-106" aria-hidden="true" tabindex="-1"></a>      <span class="at">per_roc =</span> <span class="fu">mean</span>(pre_gene_roc<span class="sc">$</span>roc,<span class="at">na.rm=</span>T)</span>
<span id="cb2-107"><a href="#cb2-107" aria-hidden="true" tabindex="-1"></a>    ) </span>
<span id="cb2-108"><a href="#cb2-108" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb2-109"><a href="#cb2-109" aria-hidden="true" tabindex="-1"></a>  sub_res <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(sub_res)</span>
<span id="cb2-110"><a href="#cb2-110" aria-hidden="true" tabindex="-1"></a>  sub_res<span class="sc">$</span>type <span class="ot">&lt;-</span> all_cv[i]</span>
<span id="cb2-111"><a href="#cb2-111" aria-hidden="true" tabindex="-1"></a>  cv_res[[i]] <span class="ot">&lt;-</span> sub_res</span>
<span id="cb2-112"><a href="#cb2-112" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-113"><a href="#cb2-113" aria-hidden="true" tabindex="-1"></a>cv_res <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(cv_res)</span>
<span id="cb2-114"><a href="#cb2-114" aria-hidden="true" tabindex="-1"></a>cv_res <span class="ot">&lt;-</span> cv_res <span class="sc">%&gt;%</span> </span>
<span id="cb2-115"><a href="#cb2-115" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(<span class="st">&quot;ROC&quot;</span>,<span class="st">&quot;PR&quot;</span>,<span class="st">&quot;F1&quot;</span>,<span class="st">&quot;Kappa&quot;</span>,<span class="st">&quot;per_roc&quot;</span>),</span>
<span id="cb2-116"><a href="#cb2-116" aria-hidden="true" tabindex="-1"></a>                      <span class="at">names_to =</span> <span class="st">&quot;Metric&quot;</span>,<span class="at">values_to =</span> <span class="st">&quot;Value&quot;</span>)</span>
<span id="cb2-117"><a href="#cb2-117" aria-hidden="true" tabindex="-1"></a>cv_res <span class="ot">&lt;-</span> <span class="fu">na.omit</span>(cv_res)</span>
<span id="cb2-118"><a href="#cb2-118" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(cv_res,<span class="at">file =</span> <span class="st">&quot;data/cv_res.rds&quot;</span>)</span></code></pre></div>
<p>Plot results for ablation experiments (<code>Fig1b</code>):</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>cv_res <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/cv_res.rds&quot;</span>)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>cv_res <span class="ot">&lt;-</span> cv_res <span class="sc">%&gt;%</span> </span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Metric <span class="sc">!=</span> <span class="st">&quot;per_f1&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Metric =</span> <span class="fu">ifelse</span>(Metric <span class="sc">==</span> <span class="st">&quot;per_roc&quot;</span>,<span class="st">&quot;Per-ROC&quot;</span>,Metric))</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>cv_1 <span class="ot">&lt;-</span> cv_res <span class="sc">%&gt;%</span> </span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>(type <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;ml_logistic&quot;</span>,<span class="st">&quot;ml_rf&quot;</span>,<span class="st">&quot;ml_svm&quot;</span>)))</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="fu">ggbarplot</span>(cv_1, <span class="at">x =</span> <span class="st">&quot;type&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Value&quot;</span>,<span class="at">fill=</span><span class="st">&quot;Metric&quot;</span>,</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>          <span class="at">add =</span> <span class="st">&quot;mean_se&quot;</span>, <span class="at">label =</span> <span class="cn">TRUE</span>, </span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>          <span class="at">lab.vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>,<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="fl">0.9</span>),</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>          <span class="at">order =</span> <span class="fu">c</span>(<span class="st">&quot;own_model&quot;</span>,<span class="st">&quot;random_feat&quot;</span>,<span class="st">&quot;random_exp&quot;</span>,<span class="st">&quot;no_gnn&quot;</span>,</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>                    <span class="st">&quot;gnn_model_random_y&quot;</span>),</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>          <span class="at">palette =</span> <span class="fu">c</span>(<span class="st">&quot;#2B4F7D&quot;</span>,<span class="st">&quot;#3B77B0&quot;</span>,<span class="st">&quot;#7BC5E3&quot;</span>,<span class="st">&quot;#EEDC96&quot;</span>,<span class="st">&quot;#637951&quot;</span>),</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>          <span class="at">lab.nb.digits =</span> <span class="dv">2</span>, <span class="at">xlab=</span>F)<span class="sc">+</span></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">labels=</span><span class="fu">c</span>(<span class="st">&quot;DeepMeta&quot;</span>,<span class="st">&quot;Random Feature&quot;</span>,</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>                            <span class="st">&quot;Random Expression&quot;</span>,</span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>                            <span class="st">&quot;Without GNN&quot;</span>,<span class="st">&quot;Scrambled label&quot;</span>))<span class="sc">+</span></span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;Type&quot;</span>,<span class="at">y=</span><span class="st">&#39;Value&#39;</span>)</span></code></pre></div>
<p><img src="report_main_files/figure-html/fig1b-1.png" width="2400" /></p>
<p>We also performed scrambling label experiment as negative control to
verify that the achieved performance was not by chance. Besides, our
model achieved an average ten-fold per-ROC (ROC of individual gene) of
0.61, much higher than expected by random chance using the
scrambled-label method (average per-ROC of 0.49).</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">vector</span>(<span class="st">&quot;list&quot;</span>,<span class="dv">10</span>)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>){</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  pre <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">paste0</span>(<span class="st">&quot;~/DeepMeta/data/cv/own_model&quot;</span>,</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>                         <span class="st">&quot;/fold_&quot;</span>,j<span class="dv">-1</span>,<span class="st">&quot;.csv&quot;</span>)) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>X)</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  dt <span class="ot">&lt;-</span> pre <span class="sc">%&gt;%</span> </span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(genes <span class="sc">%in%</span> all_dt_summ<span class="sc">$</span>id) <span class="sc">%&gt;%</span> </span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(genes) <span class="sc">%&gt;%</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">pos_counts =</span> <span class="fu">sum</span>(label <span class="sc">==</span> <span class="dv">1</span>),<span class="at">neg_counts =</span> <span class="fu">sum</span>(label <span class="sc">==</span> <span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> <span class="fu">filter</span>(pos_counts<span class="sc">&gt;</span><span class="dv">0</span> <span class="sc">&amp;</span> neg_counts <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>  pre_gene_roc <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene =</span> <span class="fu">unique</span>(dt<span class="sc">$</span>genes)) <span class="sc">%&gt;%</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">roc =</span> <span class="fu">get_gene_metric</span>(pre,<span class="at">gene_name =</span> gene,<span class="st">&quot;roc&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ungroup</span>()</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>  pre_gene_roc<span class="sc">$</span>fold <span class="ot">&lt;-</span> j</span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>  pre_gene_roc <span class="ot">&lt;-</span> <span class="fu">left_join</span>(</span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>    pre_gene_roc <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">id =</span> gene),</span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>    all_dt_summ</span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a>  res[[j]] <span class="ot">&lt;-</span> pre_gene_roc</span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(res)</span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a>res_gene_summ <span class="ot">&lt;-</span> res <span class="sc">%&gt;%</span> </span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(gene) <span class="sc">%&gt;%</span> </span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">ave_per_roc =</span> <span class="fu">mean</span>(roc)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(res_gene_summ,<span class="at">file =</span> <span class="st">&quot;data/per_roc.rds&quot;</span>)</span></code></pre></div>
<p>Among these genes, the proportion of genes with an ROC exceeding 0.5
is about 85% (<code>Extended Data Fig. 1</code>), indicating that our
model has good predictive performance for most genes:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>res_gene_summ <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/per_roc.rds&quot;</span>)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>res_gene_summ<span class="sc">$</span>tt <span class="ot">&lt;-</span> <span class="st">&quot;&quot;</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggboxplot</span>(res_gene_summ,<span class="at">x=</span><span class="st">&quot;tt&quot;</span>,<span class="at">y=</span><span class="st">&quot;ave_per_roc&quot;</span>,<span class="at">xlab =</span> <span class="cn">FALSE</span>,</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>          <span class="at">ylab =</span> <span class="st">&quot;Average ROC&quot;</span>)</span></code></pre></div>
<p><img src="report_main_files/figure-html/figs1-1.png" width="2400" /></p>
<p>We compared the model performance with three machine learning (ML)
models (see Methods). Since the machine learning model cannot handle
high-dimensional input features (3247 metabolic node features plus
7993-dimensional sample gene expression features), these ML models used
dimension-reduced inputs obtained by principal component analysis (PCA)
and were trained and validated in ten-fold cross-validation
(<code>Fig1c</code>):</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>cv_res <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/cv_res.rds&quot;</span>)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>cv_res <span class="ot">&lt;-</span> cv_res <span class="sc">%&gt;%</span> </span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Metric <span class="sc">!=</span> <span class="st">&quot;per_f1&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Metric =</span> <span class="fu">ifelse</span>(Metric <span class="sc">==</span> <span class="st">&quot;per_roc&quot;</span>,<span class="st">&quot;Per-ROC&quot;</span>,Metric))</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>cv_2 <span class="ot">&lt;-</span> cv_res <span class="sc">%&gt;%</span> </span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(type <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;own_model&quot;</span>,<span class="st">&quot;ml_logistic&quot;</span>,<span class="st">&quot;ml_rf&quot;</span>,<span class="st">&quot;ml_svm&quot;</span>))</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a><span class="fu">ggbarplot</span>(cv_2, <span class="at">x =</span> <span class="st">&quot;type&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Value&quot;</span>,<span class="at">fill=</span><span class="st">&quot;Metric&quot;</span>,</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>          <span class="at">add =</span> <span class="st">&quot;mean_se&quot;</span>, <span class="at">label =</span> <span class="cn">TRUE</span>, </span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>          <span class="at">lab.vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>,<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="fl">0.9</span>),</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>          <span class="at">order =</span> <span class="fu">c</span>(<span class="st">&quot;own_model&quot;</span>,<span class="st">&quot;ml_logistic&quot;</span>,<span class="st">&quot;ml_rf&quot;</span>,<span class="st">&quot;ml_svm&quot;</span>),</span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>          <span class="at">palette =</span> <span class="fu">c</span>(<span class="st">&quot;#2B4F7D&quot;</span>,<span class="st">&quot;#3B77B0&quot;</span>,<span class="st">&quot;#7BC5E3&quot;</span>,<span class="st">&quot;#EEDC96&quot;</span>,<span class="st">&quot;#637951&quot;</span>),</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>          <span class="at">lab.nb.digits =</span> <span class="dv">2</span>, <span class="at">xlab=</span>F)<span class="sc">+</span></span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">labels=</span><span class="fu">c</span>(<span class="st">&quot;DeepMeta&quot;</span>,<span class="st">&quot;Logistic&quot;</span>,<span class="st">&quot;RF&quot;</span>,<span class="st">&quot;SVM&quot;</span>))<span class="sc">+</span></span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;Type&quot;</span>,<span class="at">y=</span><span class="st">&#39;Value&#39;</span>)</span></code></pre></div>
<p><img src="report_main_files/figure-html/Fig1c-1.png" width="2400" /></p>
</div>
<div id="model-performance-evaluation" class="section level1">
<h1>Model performance evaluation</h1>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>enz_gene_mapping <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/enz_gene_mapping.rds&quot;</span>)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>cpg_gene <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/cpg_gene.rds&quot;</span>)</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>ensg2name <span class="ot">&lt;-</span> <span class="cf">function</span>(ensg,mapping){</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>  tt <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(ensg,<span class="st">&quot; and &quot;</span>)[[<span class="dv">1</span>]]</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>  tt_gene <span class="ot">&lt;-</span> mapping<span class="sc">$</span>symbol[<span class="fu">which</span>(mapping<span class="sc">$</span>ensembl_id <span class="sc">%in%</span> tt)] <span class="sc">%&gt;%</span> </span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">unique</span>() </span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>  sym <span class="ot">&lt;-</span> tt_gene <span class="sc">%&gt;%</span> </span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">paste</span>(.,<span class="at">collapse=</span><span class="st">&quot;,&quot;</span>)</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(sym)</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p>In the test dataset (76 cell lines and 1010 genes), DeepMeta achieved
the ROC of 0.94 (<code>Fig2a</code>), illustrating our model enables
accurate prediction of metabolic dependencies:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>pre <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;~/DeepMeta/data/test_preV2.csv&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>X)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>pre <span class="ot">&lt;-</span> pre <span class="sc">%&gt;%</span> </span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">truth =</span> <span class="fu">ifelse</span>(label <span class="sc">==</span> <span class="dv">1</span>, <span class="st">&quot;Class1&quot;</span>,<span class="st">&quot;Class2&quot;</span>),</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">pred_label =</span> <span class="fu">ifelse</span>(preds <span class="sc">==</span> <span class="dv">1</span>, <span class="st">&quot;Class1&quot;</span>,<span class="st">&quot;Class2&quot;</span>))</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>pre<span class="sc">$</span>truth <span class="ot">&lt;-</span> <span class="fu">factor</span>(pre<span class="sc">$</span>truth)</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>pre<span class="sc">$</span>pred_label <span class="ot">&lt;-</span> <span class="fu">factor</span>(pre<span class="sc">$</span>pred_label)</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>pr <span class="ot">&lt;-</span>  <span class="fu">pr_auc</span>(pre, truth, preds_raw)[<span class="st">&quot;.estimate&quot;</span>] <span class="sc">%&gt;%</span> </span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> <span class="fu">unname</span>() <span class="sc">%&gt;%</span> <span class="fu">round</span>(.,<span class="dv">2</span>)</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>roc <span class="ot">&lt;-</span>  <span class="fu">roc_auc</span>(pre, truth, preds_raw)[<span class="st">&quot;.estimate&quot;</span>] <span class="sc">%&gt;%</span> </span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> <span class="fu">unname</span>() <span class="sc">%&gt;%</span> <span class="fu">round</span>(.,<span class="dv">2</span>)</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">roc_curve</span>(pre, truth, preds_raw) <span class="sc">%&gt;%</span></span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">1</span> <span class="sc">-</span> specificity, <span class="at">y =</span> sensitivity)) <span class="sc">+</span></span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_path</span>() <span class="sc">+</span></span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_fixed</span>(<span class="at">xlim =</span> <span class="dv">0</span><span class="sc">:</span><span class="dv">1</span>, <span class="at">ylim =</span> <span class="dv">0</span><span class="sc">:</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="at">geom=</span><span class="st">&quot;text&quot;</span>, <span class="at">x=</span><span class="fl">0.75</span>, <span class="at">y=</span><span class="fl">0.7</span>, <span class="at">label=</span><span class="fu">paste0</span>(<span class="st">&quot;ROC-AUC: &quot;</span>,roc),</span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a>           <span class="at">size=</span><span class="dv">5</span>)</span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a>p1</span></code></pre></div>
<p><img src="report_main_files/figure-html/Fig2a-1.png" width="2400" /></p>
<p>We further validated DeepMeta performance using four independent
datasets: - 1) Another CRISPR screen data conducted by the Sanger
institute using a different CRISPR library8, containing 200 cell lines
overlapped with training data and 25 unique cell lines; - 2) RNAi screen
data9, which apply DEMETER2 to the combination of three large-scale RNAi
screening datasets: the Broad institute project Achilles, Novartis
project DRIVE, and the Marcotte et al. breast cell line dataset. This
dataset contain 417 overlapped cell lines and 160 unique cell lines; -
3) The Childhood Cancer Model Atlas10 (CCMA) dataset which performs
CRISPR-Cas9 screen in 110 paediatric solid tumour cell lines and reports
Z scores; - 4) Pemovska et.al11 drug screen dataset, which performs
metabolic drug library (contains 243 compounds) screen in 15 cancer cell
lines (14 cells with gene expression data) and reports AUC scores,
larger AUC score indicates stronger killing effect on cell.</p>
<p>DeepMeta achieved high AUROC scores in Sanger and RNAi dataset, both
for overlapped and unique cells (<code>Fig2b</code> and
<code>Fig2c</code>):</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>sanger <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;/home/data/sdc/wt/model_data/sanger_gene_dependency_chronos.csv&quot;</span>,<span class="at">data.table =</span> F)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(sanger)[<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(sanger)] <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">s*</span><span class="sc">\\</span><span class="st">([^</span><span class="sc">\\</span><span class="st">)]+</span><span class="sc">\\</span><span class="st">)&quot;</span>,<span class="st">&quot;&quot;</span>,</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>                                         <span class="fu">colnames</span>(sanger)[<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(sanger)])</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>sanger <span class="ot">&lt;-</span> sanger <span class="sc">%&gt;%</span> </span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">colnames</span>(sanger)[<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(sanger)],<span class="at">names_to =</span> <span class="st">&quot;gene&quot;</span>,</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>                      <span class="at">values_to =</span> <span class="st">&quot;score&quot;</span>)</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(sanger)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">&quot;cell&quot;</span></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>sanger <span class="ot">&lt;-</span> sanger <span class="sc">%&gt;%</span></span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(score)) <span class="sc">%&gt;%</span> </span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">type =</span> <span class="fu">case_when</span>(</span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>    score<span class="sc">&gt;</span><span class="fl">0.5</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>    score<span class="sc">&lt;=</span><span class="fl">0.5</span> <span class="sc">~</span> <span class="dv">0</span></span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>  ))</span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a>pre <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;~/DeepMeta/data/sanger_preV2.csv&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>X)</span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a>pre_gene <span class="ot">&lt;-</span>  <span class="fu">data.frame</span>(<span class="at">id =</span> <span class="fu">unique</span>(<span class="fu">c</span>(pre<span class="sc">$</span>gene_name))) <span class="sc">%&gt;%</span> </span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">gene=</span><span class="fu">ensg2name</span>(id,enz_gene_mapping)) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>()</span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a>pre <span class="ot">&lt;-</span> pre <span class="sc">%&gt;%</span> </span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(.,pre_gene <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">gene_name=</span>id)) <span class="sc">%&gt;%</span> </span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">separate_longer_delim</span>(<span class="at">cols =</span> gene,<span class="at">delim =</span> <span class="st">&quot;,&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(.,sanger) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(type)) <span class="sc">%&gt;%</span>  </span>
<span id="cb9-24"><a href="#cb9-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(cell,gene_name) <span class="sc">%&gt;%</span> </span>
<span id="cb9-25"><a href="#cb9-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">preds =</span> <span class="fu">unique</span>(preds),</span>
<span id="cb9-26"><a href="#cb9-26" aria-hidden="true" tabindex="-1"></a>            <span class="at">preds_raw =</span> <span class="fu">unique</span>(preds_raw),</span>
<span id="cb9-27"><a href="#cb9-27" aria-hidden="true" tabindex="-1"></a>            <span class="at">label =</span> <span class="fu">max</span>(type,<span class="at">na.rm =</span> T)) <span class="sc">%&gt;%</span> </span>
<span id="cb9-28"><a href="#cb9-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb9-29"><a href="#cb9-29" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(pre,<span class="at">file =</span> <span class="st">&quot;~/DeepMeta/data/sanger_pre.rds&quot;</span>)</span></code></pre></div>
<p>The results of Sanger dataset:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>pre <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/sanger_pre.rds&quot;</span>)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>pre <span class="ot">&lt;-</span> pre <span class="sc">%&gt;%</span> </span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">truth =</span> <span class="fu">ifelse</span>(label <span class="sc">==</span> <span class="dv">1</span>, <span class="st">&quot;Class1&quot;</span>,<span class="st">&quot;Class2&quot;</span>),</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">pred_label =</span> <span class="fu">ifelse</span>(preds <span class="sc">==</span> <span class="dv">1</span>, <span class="st">&quot;Class1&quot;</span>,<span class="st">&quot;Class2&quot;</span>))</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>pre<span class="sc">$</span>truth <span class="ot">&lt;-</span> <span class="fu">factor</span>(pre<span class="sc">$</span>truth)</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>pre<span class="sc">$</span>pred_label <span class="ot">&lt;-</span> <span class="fu">factor</span>(pre<span class="sc">$</span>pred_label)</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>train_dt <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;/home/data/sdc/wt/model_data/new_model/cell_net_filter_exp/raw/train_cell_info.csv&quot;</span>)</span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>common_cell <span class="ot">&lt;-</span> pre <span class="sc">%&gt;%</span> <span class="fu">filter</span>(cell <span class="sc">%in%</span> train_dt<span class="sc">$</span>cell)</span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>unique_cell <span class="ot">&lt;-</span> pre <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>(cell <span class="sc">%in%</span> train_dt<span class="sc">$</span>cell))</span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>roc_comm <span class="ot">&lt;-</span>  <span class="fu">roc_auc</span>(common_cell, truth, preds_raw)[<span class="st">&quot;.estimate&quot;</span>] <span class="sc">%&gt;%</span> </span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> <span class="fu">unname</span>() <span class="sc">%&gt;%</span> <span class="fu">round</span>(.,<span class="dv">2</span>)</span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>roc_unique <span class="ot">&lt;-</span>  <span class="fu">roc_auc</span>(unique_cell, truth, preds_raw)[<span class="st">&quot;.estimate&quot;</span>] <span class="sc">%&gt;%</span> </span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> <span class="fu">unname</span>() <span class="sc">%&gt;%</span> <span class="fu">round</span>(.,<span class="dv">2</span>)</span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a>p4 <span class="ot">&lt;-</span> <span class="fu">roc_curve</span>(common_cell, truth, preds_raw) <span class="sc">%&gt;%</span></span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">1</span> <span class="sc">-</span> specificity, <span class="at">y =</span> sensitivity)) <span class="sc">+</span></span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_path</span>() <span class="sc">+</span></span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_fixed</span>(<span class="at">xlim =</span> <span class="dv">0</span><span class="sc">:</span><span class="dv">1</span>, <span class="at">ylim =</span> <span class="dv">0</span><span class="sc">:</span><span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb10-23"><a href="#cb10-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="at">geom=</span><span class="st">&quot;text&quot;</span>, <span class="at">x=</span><span class="fl">0.5</span>, <span class="at">y=</span><span class="fl">0.7</span>, <span class="at">label=</span><span class="fu">paste0</span>(<span class="st">&quot;Seen Cells ROC-AUC: &quot;</span>,roc_comm),</span>
<span id="cb10-24"><a href="#cb10-24" aria-hidden="true" tabindex="-1"></a>           <span class="at">size=</span><span class="dv">4</span>)</span>
<span id="cb10-25"><a href="#cb10-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-26"><a href="#cb10-26" aria-hidden="true" tabindex="-1"></a>p5 <span class="ot">&lt;-</span> <span class="fu">roc_curve</span>(unique_cell, truth, preds_raw) <span class="sc">%&gt;%</span></span>
<span id="cb10-27"><a href="#cb10-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">1</span> <span class="sc">-</span> specificity, <span class="at">y =</span> sensitivity)) <span class="sc">+</span></span>
<span id="cb10-28"><a href="#cb10-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_path</span>() <span class="sc">+</span></span>
<span id="cb10-29"><a href="#cb10-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_fixed</span>(<span class="at">xlim =</span> <span class="dv">0</span><span class="sc">:</span><span class="dv">1</span>, <span class="at">ylim =</span> <span class="dv">0</span><span class="sc">:</span><span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb10-30"><a href="#cb10-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb10-31"><a href="#cb10-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="at">geom=</span><span class="st">&quot;text&quot;</span>, <span class="at">x=</span><span class="fl">0.5</span>, <span class="at">y=</span><span class="fl">0.7</span>, <span class="at">label=</span><span class="fu">paste0</span>(<span class="st">&quot;Unseen Cell ROC-AUC: &quot;</span>,roc_unique),</span>
<span id="cb10-32"><a href="#cb10-32" aria-hidden="true" tabindex="-1"></a>           <span class="at">size=</span><span class="dv">4</span>)</span>
<span id="cb10-33"><a href="#cb10-33" aria-hidden="true" tabindex="-1"></a>p4<span class="sc">+</span>p5</span></code></pre></div>
<p><img src="report_main_files/figure-html/Fig2b-1.png" width="2400" /></p>
<p>The results of RNAi dataset:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>pre <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;~/DeepMeta/data/rnai_preV2.csv&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>X)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>train_dt <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/train_dtV2.rds&quot;</span>)</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>pre <span class="ot">&lt;-</span> pre <span class="sc">%&gt;%</span> </span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">truth =</span> <span class="fu">ifelse</span>(label <span class="sc">==</span> <span class="dv">1</span>, <span class="st">&quot;Class1&quot;</span>,<span class="st">&quot;Class2&quot;</span>))</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>pre<span class="sc">$</span>truth <span class="ot">&lt;-</span> <span class="fu">factor</span>(pre<span class="sc">$</span>truth)</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>seen <span class="ot">&lt;-</span> pre <span class="sc">%&gt;%</span> <span class="fu">filter</span>(cell <span class="sc">%in%</span> train_dt<span class="sc">$</span>cell)</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>unseen <span class="ot">&lt;-</span> pre <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>(cell <span class="sc">%in%</span> train_dt<span class="sc">$</span>cell))</span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>seen_roc <span class="ot">&lt;-</span>  <span class="fu">roc_auc</span>(seen, truth, preds_raw)[<span class="st">&quot;.estimate&quot;</span>] <span class="sc">%&gt;%</span> </span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> <span class="fu">unname</span>() <span class="sc">%&gt;%</span> <span class="fu">round</span>(.,<span class="dv">2</span>)</span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>unseen_roc <span class="ot">&lt;-</span>  <span class="fu">roc_auc</span>(unseen, truth, preds_raw)[<span class="st">&quot;.estimate&quot;</span>] <span class="sc">%&gt;%</span> </span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> <span class="fu">unname</span>() <span class="sc">%&gt;%</span> <span class="fu">round</span>(.,<span class="dv">2</span>)</span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">roc_curve</span>(seen, truth, preds_raw) <span class="sc">%&gt;%</span></span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">1</span> <span class="sc">-</span> specificity, <span class="at">y =</span> sensitivity)) <span class="sc">+</span></span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_path</span>() <span class="sc">+</span></span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_fixed</span>(<span class="at">xlim =</span> <span class="dv">0</span><span class="sc">:</span><span class="dv">1</span>, <span class="at">ylim =</span> <span class="dv">0</span><span class="sc">:</span><span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="at">geom=</span><span class="st">&quot;text&quot;</span>, <span class="at">x=</span><span class="fl">0.5</span>, <span class="at">y=</span><span class="fl">0.7</span>, <span class="at">label=</span><span class="fu">paste0</span>(<span class="st">&quot;Seen Cells ROC-AUC: &quot;</span>,seen_roc),</span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a>           <span class="at">size=</span><span class="dv">4</span>)</span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">roc_curve</span>(unseen, truth, preds_raw) <span class="sc">%&gt;%</span></span>
<span id="cb11-23"><a href="#cb11-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">1</span> <span class="sc">-</span> specificity, <span class="at">y =</span> sensitivity)) <span class="sc">+</span></span>
<span id="cb11-24"><a href="#cb11-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_path</span>() <span class="sc">+</span></span>
<span id="cb11-25"><a href="#cb11-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_fixed</span>(<span class="at">xlim =</span> <span class="dv">0</span><span class="sc">:</span><span class="dv">1</span>, <span class="at">ylim =</span> <span class="dv">0</span><span class="sc">:</span><span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb11-26"><a href="#cb11-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb11-27"><a href="#cb11-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="at">geom=</span><span class="st">&quot;text&quot;</span>, <span class="at">x=</span><span class="fl">0.5</span>, <span class="at">y=</span><span class="fl">0.7</span>, <span class="at">label=</span><span class="fu">paste0</span>(<span class="st">&quot;Unseen Cells ROC-AUC: &quot;</span>,</span>
<span id="cb11-28"><a href="#cb11-28" aria-hidden="true" tabindex="-1"></a>                                                   unseen_roc),</span>
<span id="cb11-29"><a href="#cb11-29" aria-hidden="true" tabindex="-1"></a>           <span class="at">size=</span><span class="dv">4</span>)</span>
<span id="cb11-30"><a href="#cb11-30" aria-hidden="true" tabindex="-1"></a>p2<span class="sc">+</span>p3</span></code></pre></div>
<p><img src="report_main_files/figure-html/Fig2c-1.png" width="2400" /></p>
<p>CCMA dataset uses Z score to quantify gene dependency, consistent
with the original paper, we classified cell-gene pair with Z score less
than -1.5 as positive class and others as negative class. Although the
model only achieved a moderate AUROC score in this dataset, there is a
significant difference in the Z scores between different predicted
classes (<code>Fig2d</code>):</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>ccma_dep <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;/home/data/sdc/wt/model_data/CCMA_CRISPRdependencyscore.csv&quot;</span>, <span class="at">data.table =</span> F)</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>pre <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;~/DeepMeta/data/ccma_preV2.csv&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>X)</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>pre_gene <span class="ot">&lt;-</span>  <span class="fu">data.frame</span>(<span class="at">id =</span> <span class="fu">unique</span>(<span class="fu">c</span>(pre<span class="sc">$</span>gene_name))) <span class="sc">%&gt;%</span> </span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">gene=</span><span class="fu">ensg2name</span>(id,enz_gene_mapping)) <span class="sc">%&gt;%</span> </span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>pre <span class="ot">&lt;-</span> pre <span class="sc">%&gt;%</span> </span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(.,pre_gene <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">gene_name=</span>id)) <span class="sc">%&gt;%</span> </span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">separate_longer_delim</span>(<span class="at">cols =</span> gene,<span class="at">delim =</span> <span class="st">&quot;,&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(.,</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>            ccma_dep <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample,gene,z) <span class="sc">%&gt;%</span> </span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>              <span class="fu">rename</span>(<span class="at">cell=</span>sample)) <span class="sc">%&gt;%</span> </span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(z)) <span class="sc">%&gt;%</span>  </span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(cell,gene_name) <span class="sc">%&gt;%</span> </span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">preds =</span> <span class="fu">unique</span>(preds),</span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a>            <span class="at">preds_raw =</span> <span class="fu">unique</span>(preds_raw),</span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a>            <span class="at">z_score =</span> <span class="fu">min</span>(z,<span class="at">na.rm =</span> T)) <span class="sc">%&gt;%</span> </span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">label =</span> <span class="fu">ifelse</span>(z_score <span class="sc">&lt;</span> (<span class="sc">-</span><span class="fl">1.5</span>),<span class="dv">1</span>,<span class="dv">0</span>))</span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(pre,<span class="at">file =</span> <span class="st">&quot;~/DeepMeta/data/ccma_pre.rds&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>pre <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/ccma_pre.rds&quot;</span>)</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>pre <span class="ot">&lt;-</span> pre <span class="sc">%&gt;%</span> </span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">truth =</span> <span class="fu">ifelse</span>(label <span class="sc">==</span> <span class="dv">1</span>, <span class="st">&quot;Class1&quot;</span>,<span class="st">&quot;Class2&quot;</span>),</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">pred_label =</span> <span class="fu">ifelse</span>(preds <span class="sc">==</span> <span class="dv">1</span>, <span class="st">&quot;Class1&quot;</span>,<span class="st">&quot;Class2&quot;</span>))</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>pre<span class="sc">$</span>truth <span class="ot">&lt;-</span> <span class="fu">factor</span>(pre<span class="sc">$</span>truth)</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>pre<span class="sc">$</span>pred_label <span class="ot">&lt;-</span> <span class="fu">factor</span>(pre<span class="sc">$</span>pred_label)</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>roc <span class="ot">&lt;-</span>  <span class="fu">roc_auc</span>(pre, truth, preds_raw)[<span class="st">&quot;.estimate&quot;</span>] <span class="sc">%&gt;%</span> </span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> <span class="fu">unname</span>() <span class="sc">%&gt;%</span> <span class="fu">round</span>(.,<span class="dv">2</span>)</span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>p6 <span class="ot">&lt;-</span> <span class="fu">roc_curve</span>(pre, truth, preds_raw) <span class="sc">%&gt;%</span></span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">1</span> <span class="sc">-</span> specificity, <span class="at">y =</span> sensitivity)) <span class="sc">+</span></span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_path</span>() <span class="sc">+</span></span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_fixed</span>(<span class="at">xlim =</span> <span class="dv">0</span><span class="sc">:</span><span class="dv">1</span>, <span class="at">ylim =</span> <span class="dv">0</span><span class="sc">:</span><span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="at">geom=</span><span class="st">&quot;text&quot;</span>, <span class="at">x=</span><span class="fl">0.4</span>, <span class="at">y=</span><span class="fl">0.85</span>, <span class="at">label=</span><span class="fu">paste0</span>(<span class="st">&quot;ROC-AUC: &quot;</span>,roc),</span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a>           <span class="at">size=</span><span class="dv">5</span>)</span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a>p7 <span class="ot">&lt;-</span> <span class="fu">ggboxplot</span>(<span class="at">data=</span>pre,<span class="at">x=</span><span class="st">&quot;preds&quot;</span>,<span class="at">y=</span><span class="st">&quot;z_score&quot;</span>)<span class="sc">+</span></span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>()<span class="sc">+</span></span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;Model Prediction&quot;</span>,<span class="at">y=</span><span class="st">&quot;Dependency Z scores&quot;</span>)<span class="sc">+</span></span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> (<span class="sc">-</span><span class="fl">1.5</span>),<span class="at">color=</span><span class="st">&quot;red&quot;</span>,<span class="at">linewidth=</span><span class="dv">2</span>,<span class="at">linetype=</span><span class="dv">2</span>)</span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a>p6<span class="sc">+</span>p7</span></code></pre></div>
<p><img src="report_main_files/figure-html/Fig2d-1.png" width="2400" /></p>
<p>For each cell-gene pair in Pemovska dataset, we first obtain the
maximum AUC value of all drugs targeting a specific gene, then rank the
genes in each cell based on this AUC value and assign ranks. We showed
the overlap of the top 5 genes predicted by DeepMeta and the genes with
top 3 AUC rank in each cell (<code>Fig 2e</code>).</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>pre <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;~/DeepMeta/data/drug_preV2.csv&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>X)</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>pre_gene <span class="ot">&lt;-</span>  <span class="fu">data.frame</span>(<span class="at">id =</span> <span class="fu">unique</span>(<span class="fu">c</span>(pre<span class="sc">$</span>gene_name))) <span class="sc">%&gt;%</span> </span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">gene=</span><span class="fu">ensg2name</span>(id,enz_gene_mapping))</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>pre <span class="ot">&lt;-</span> pre <span class="sc">%&gt;%</span> </span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(.,pre_gene <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">gene_name=</span>id))</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>nc_drug <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/nc_drug.rds&quot;</span>)</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a><span class="do">####get gene AUC of drugs</span></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>get_auc <span class="ot">&lt;-</span> <span class="cf">function</span>(cell,gene,drug_dt){</span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>  genes <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(gene,<span class="at">split =</span> <span class="st">&quot;,&quot;</span>)[[<span class="dv">1</span>]]</span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">vector</span>(<span class="st">&quot;list&quot;</span>,<span class="fu">length</span>(genes))</span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(genes)){</span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>    drug_dt_t <span class="ot">&lt;-</span> drug_dt <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">grepl</span>(genes[i],target_gene))</span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>    res[[i]] <span class="ot">&lt;-</span> drug_dt_t</span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(res)</span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">nrow</span>(res)<span class="sc">==</span><span class="dv">0</span>){</span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">list</span>(<span class="cn">NA</span>,<span class="cn">NA</span>))</span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a>  }<span class="cf">else</span>{</span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a>    res <span class="ot">&lt;-</span> res <span class="sc">%&gt;%</span> <span class="fu">select</span>(cell,drug_name)</span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">list</span>(<span class="fu">max</span>(res[,<span class="dv">1</span>]),</span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a>                <span class="fu">paste</span>(res<span class="sc">$</span>drug_name[<span class="fu">unlist</span>(res[,<span class="dv">1</span>]<span class="sc">==</span><span class="fu">max</span>(res[,<span class="dv">1</span>]))],</span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true" tabindex="-1"></a>                      <span class="at">collapse =</span> <span class="st">&quot;,&quot;</span>)))</span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb14-26"><a href="#cb14-26" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb14-27"><a href="#cb14-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-28"><a href="#cb14-28" aria-hidden="true" tabindex="-1"></a>pre_drug <span class="ot">&lt;-</span> pre <span class="sc">%&gt;%</span> </span>
<span id="cb14-29"><a href="#cb14-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb14-30"><a href="#cb14-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">max_auc =</span> <span class="fu">get_auc</span>(cell,gene,nc_drug)[[<span class="dv">1</span>]],</span>
<span id="cb14-31"><a href="#cb14-31" aria-hidden="true" tabindex="-1"></a>         <span class="at">max_auc_drug =</span> <span class="fu">get_auc</span>(cell,gene,nc_drug)[[<span class="dv">2</span>]]) <span class="sc">%&gt;%</span> </span>
<span id="cb14-32"><a href="#cb14-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb14-33"><a href="#cb14-33" aria-hidden="true" tabindex="-1"></a>pre_drug <span class="ot">&lt;-</span> pre_drug <span class="sc">%&gt;%</span> </span>
<span id="cb14-34"><a href="#cb14-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">na.omit</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb14-35"><a href="#cb14-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(cell) <span class="sc">%&gt;%</span> </span>
<span id="cb14-36"><a href="#cb14-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">rank_auc =</span> <span class="fu">rank</span>(<span class="sc">-</span>max_auc)) <span class="sc">%&gt;%</span> </span>
<span id="cb14-37"><a href="#cb14-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb14-38"><a href="#cb14-38" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(pre_drug,<span class="at">file =</span> <span class="st">&quot;~/DeepMeta/data/nc_pre_drug_V2.rds&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>pre_drug <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/nc_pre_drug_V2.rds&quot;</span>)</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>pre_drug_top <span class="ot">&lt;-</span> pre_drug <span class="sc">%&gt;%</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(cell) <span class="sc">%&gt;%</span> </span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_max</span>(<span class="at">order_by =</span> preds_raw, <span class="at">n =</span><span class="dv">5</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a><span class="do">###label rank = 1</span></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>pre_drug_top <span class="ot">&lt;-</span> pre_drug_top <span class="sc">%&gt;%</span> </span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ll =</span> <span class="fu">ifelse</span>(rank_auc <span class="sc">&lt;=</span> <span class="dv">3</span>,</span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">paste0</span>(<span class="st">&quot;Rank: &quot;</span>,rank_auc,<span class="st">&quot;, AUC: &quot;</span>,<span class="fu">round</span>(max_auc,<span class="dv">2</span>)),</span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>                     <span class="st">&quot;&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ll1 =</span> <span class="fu">ifelse</span>(rank_auc <span class="sc">&lt;=</span> <span class="dv">3</span>,</span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">paste0</span>(<span class="st">&quot;Rank: &quot;</span>,rank_auc),<span class="st">&quot;&quot;</span>))</span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a><span class="fu">ggscatter</span>(pre_drug_top, <span class="at">x =</span> <span class="st">&quot;cell&quot;</span>, <span class="at">y =</span> <span class="st">&quot;preds_raw&quot;</span>)<span class="sc">+</span></span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="fu">aes</span>(<span class="at">label =</span> ll),<span class="at">box.padding =</span> <span class="fl">0.5</span>,<span class="at">min.segment.length =</span> <span class="dv">0</span>,</span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a>                  <span class="at">max.overlaps =</span> <span class="cn">Inf</span>) <span class="sc">+</span></span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">color =</span> <span class="fu">ifelse</span>(pre_drug_top<span class="sc">$</span>ll <span class="sc">==</span> <span class="st">&quot;&quot;</span>, <span class="st">&quot;grey50&quot;</span>, <span class="st">&quot;red&quot;</span>),<span class="at">size=</span><span class="dv">4</span>)<span class="sc">+</span></span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rotate_x_text</span>(<span class="dv">45</span>)<span class="sc">+</span></span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;Cell Lines&quot;</span>,<span class="at">y=</span><span class="st">&quot;Model Prediction&quot;</span>)</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-7-1.png" width="2400" /></p>
<p>Interestingly, we found that 11 of the 14 cells contain overlap genes
and the AUC rank of the most overlapped genes is 1 (8/11, median AUC
value of these overlapped gene is 0.85). This result indicates that
DeepMeta predictions can be recapitulated by actual drug screening
experiments. In summary, DeepMeta shows good performance and
generalization ability on different independent datasets.</p>
<p>We also calculated other metrics in these validation dataset
(<code>Extended Data Fig. 3</code>):</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>get_roc <span class="ot">&lt;-</span> <span class="cf">function</span>(dt,<span class="at">need=</span><span class="st">&quot;all&quot;</span>){</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>  dt <span class="ot">&lt;-</span> dt <span class="sc">%&gt;%</span> </span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">truth =</span> <span class="fu">ifelse</span>(label <span class="sc">==</span> <span class="dv">1</span>, <span class="st">&quot;Class1&quot;</span>,<span class="st">&quot;Class2&quot;</span>),</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">pred_label =</span> <span class="fu">ifelse</span>(preds <span class="sc">==</span> <span class="dv">1</span>, <span class="st">&quot;Class1&quot;</span>,<span class="st">&quot;Class2&quot;</span>))</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>  dt<span class="sc">$</span>truth <span class="ot">&lt;-</span> <span class="fu">factor</span>(dt<span class="sc">$</span>truth)</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>  dt<span class="sc">$</span>pred_label <span class="ot">&lt;-</span> <span class="fu">factor</span>(dt<span class="sc">$</span>pred_label)</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>  f1 <span class="ot">&lt;-</span> <span class="fu">try</span>(<span class="fu">f_meas</span>(dt,truth,pred_label)[<span class="st">&quot;.estimate&quot;</span>] <span class="sc">%&gt;%</span></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>              <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> <span class="fu">unname</span>() <span class="sc">%&gt;%</span> <span class="fu">round</span>(.,<span class="dv">2</span>),<span class="at">silent =</span> <span class="cn">TRUE</span>)</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>  kappa <span class="ot">&lt;-</span> <span class="fu">try</span>(<span class="fu">kap</span>(dt,truth,pred_label)[<span class="st">&quot;.estimate&quot;</span>] <span class="sc">%&gt;%</span></span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> <span class="fu">unname</span>() <span class="sc">%&gt;%</span> <span class="fu">round</span>(.,<span class="dv">2</span>),<span class="at">silent =</span> <span class="cn">TRUE</span>)</span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>  f1 <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="st">&#39;try-error&#39;</span> <span class="sc">%in%</span> <span class="fu">class</span>(f1),<span class="cn">NA</span>,f1)</span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>  kappa <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="st">&#39;try-error&#39;</span> <span class="sc">%in%</span> <span class="fu">class</span>(kappa),<span class="cn">NA</span>,kappa)</span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (need <span class="sc">==</span> <span class="st">&quot;all&quot;</span>){</span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>    roc <span class="ot">&lt;-</span>  <span class="fu">roc_auc</span>(dt, truth, preds_raw)[<span class="st">&quot;.estimate&quot;</span>] <span class="sc">%&gt;%</span> </span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a>      <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> <span class="fu">unname</span>() <span class="sc">%&gt;%</span> <span class="fu">round</span>(.,<span class="dv">2</span>)</span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a>    pr <span class="ot">&lt;-</span>  <span class="fu">pr_auc</span>(dt, truth, preds_raw)[<span class="st">&quot;.estimate&quot;</span>] <span class="sc">%&gt;%</span></span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a>      <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> <span class="fu">unname</span>() <span class="sc">%&gt;%</span> <span class="fu">round</span>(.,<span class="dv">2</span>)</span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">c</span>(roc,pr,f1,kappa))</span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a>  }<span class="cf">else</span>{</span>
<span id="cb16-21"><a href="#cb16-21" aria-hidden="true" tabindex="-1"></a>    res <span class="ot">&lt;-</span> <span class="cf">switch</span>(</span>
<span id="cb16-22"><a href="#cb16-22" aria-hidden="true" tabindex="-1"></a>      need,</span>
<span id="cb16-23"><a href="#cb16-23" aria-hidden="true" tabindex="-1"></a>      <span class="st">&quot;roc&quot;</span> <span class="ot">=</span> <span class="fu">roc_auc</span>(dt, truth, preds_raw)[<span class="st">&quot;.estimate&quot;</span>] <span class="sc">%&gt;%</span> </span>
<span id="cb16-24"><a href="#cb16-24" aria-hidden="true" tabindex="-1"></a>        <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> <span class="fu">unname</span>() <span class="sc">%&gt;%</span> <span class="fu">round</span>(.,<span class="dv">2</span>),</span>
<span id="cb16-25"><a href="#cb16-25" aria-hidden="true" tabindex="-1"></a>      <span class="st">&quot;pr&quot;</span> <span class="ot">=</span> <span class="fu">pr_auc</span>(dt, truth, preds_raw)[<span class="st">&quot;.estimate&quot;</span>] <span class="sc">%&gt;%</span></span>
<span id="cb16-26"><a href="#cb16-26" aria-hidden="true" tabindex="-1"></a>        <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> <span class="fu">unname</span>() <span class="sc">%&gt;%</span> <span class="fu">round</span>(.,<span class="dv">2</span>),</span>
<span id="cb16-27"><a href="#cb16-27" aria-hidden="true" tabindex="-1"></a>      <span class="st">&quot;f1&quot;</span> <span class="ot">=</span> f1,</span>
<span id="cb16-28"><a href="#cb16-28" aria-hidden="true" tabindex="-1"></a>      <span class="st">&quot;kappa&quot;</span> <span class="ot">=</span> kappa</span>
<span id="cb16-29"><a href="#cb16-29" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb16-30"><a href="#cb16-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(res)</span>
<span id="cb16-31"><a href="#cb16-31" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb16-32"><a href="#cb16-32" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb16-33"><a href="#cb16-33" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb16-34"><a href="#cb16-34" aria-hidden="true" tabindex="-1"></a>get_gene_metric <span class="ot">&lt;-</span> <span class="cf">function</span>(dt,gene_name,need){</span>
<span id="cb16-35"><a href="#cb16-35" aria-hidden="true" tabindex="-1"></a>  dt <span class="ot">&lt;-</span> dt <span class="sc">%&gt;%</span> </span>
<span id="cb16-36"><a href="#cb16-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">truth =</span> <span class="fu">ifelse</span>(label <span class="sc">==</span> <span class="dv">1</span>, <span class="st">&quot;Class1&quot;</span>,<span class="st">&quot;Class2&quot;</span>),</span>
<span id="cb16-37"><a href="#cb16-37" aria-hidden="true" tabindex="-1"></a>           <span class="at">pred_label =</span> <span class="fu">ifelse</span>(preds <span class="sc">==</span> <span class="dv">1</span>, <span class="st">&quot;Class1&quot;</span>,<span class="st">&quot;Class2&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb16-38"><a href="#cb16-38" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(genes <span class="sc">==</span> gene_name)</span>
<span id="cb16-39"><a href="#cb16-39" aria-hidden="true" tabindex="-1"></a>  dt<span class="sc">$</span>truth <span class="ot">&lt;-</span> <span class="fu">factor</span>(dt<span class="sc">$</span>truth)</span>
<span id="cb16-40"><a href="#cb16-40" aria-hidden="true" tabindex="-1"></a>  dt<span class="sc">$</span>pred_label <span class="ot">&lt;-</span> <span class="fu">factor</span>(dt<span class="sc">$</span>pred_label)</span>
<span id="cb16-41"><a href="#cb16-41" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb16-42"><a href="#cb16-42" aria-hidden="true" tabindex="-1"></a>  f1 <span class="ot">&lt;-</span> <span class="fu">try</span>(<span class="fu">f_meas</span>(dt,truth,pred_label)[<span class="st">&quot;.estimate&quot;</span>] <span class="sc">%&gt;%</span></span>
<span id="cb16-43"><a href="#cb16-43" aria-hidden="true" tabindex="-1"></a>              <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> <span class="fu">unname</span>() <span class="sc">%&gt;%</span> <span class="fu">round</span>(.,<span class="dv">2</span>),<span class="at">silent =</span> <span class="cn">TRUE</span>)</span>
<span id="cb16-44"><a href="#cb16-44" aria-hidden="true" tabindex="-1"></a>  kappa <span class="ot">&lt;-</span> <span class="fu">try</span>(<span class="fu">kap</span>(dt,truth,pred_label)[<span class="st">&quot;.estimate&quot;</span>] <span class="sc">%&gt;%</span></span>
<span id="cb16-45"><a href="#cb16-45" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> <span class="fu">unname</span>() <span class="sc">%&gt;%</span> <span class="fu">round</span>(.,<span class="dv">2</span>),<span class="at">silent =</span> <span class="cn">TRUE</span>)</span>
<span id="cb16-46"><a href="#cb16-46" aria-hidden="true" tabindex="-1"></a>  f1 <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="st">&#39;try-error&#39;</span> <span class="sc">%in%</span> <span class="fu">class</span>(f1),<span class="cn">NA</span>,f1)</span>
<span id="cb16-47"><a href="#cb16-47" aria-hidden="true" tabindex="-1"></a>  kappa <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="st">&#39;try-error&#39;</span> <span class="sc">%in%</span> <span class="fu">class</span>(kappa),<span class="cn">NA</span>,kappa)</span>
<span id="cb16-48"><a href="#cb16-48" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb16-49"><a href="#cb16-49" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="cf">switch</span>(</span>
<span id="cb16-50"><a href="#cb16-50" aria-hidden="true" tabindex="-1"></a>    need,</span>
<span id="cb16-51"><a href="#cb16-51" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;roc&quot;</span> <span class="ot">=</span> <span class="fu">roc_auc</span>(dt, truth, preds_raw)[<span class="st">&quot;.estimate&quot;</span>] <span class="sc">%&gt;%</span> </span>
<span id="cb16-52"><a href="#cb16-52" aria-hidden="true" tabindex="-1"></a>      <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> <span class="fu">unname</span>() <span class="sc">%&gt;%</span> <span class="fu">round</span>(.,<span class="dv">2</span>),</span>
<span id="cb16-53"><a href="#cb16-53" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;pr&quot;</span> <span class="ot">=</span> <span class="fu">pr_auc</span>(dt, truth, preds_raw)[<span class="st">&quot;.estimate&quot;</span>] <span class="sc">%&gt;%</span></span>
<span id="cb16-54"><a href="#cb16-54" aria-hidden="true" tabindex="-1"></a>      <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> <span class="fu">unname</span>() <span class="sc">%&gt;%</span> <span class="fu">round</span>(.,<span class="dv">2</span>),</span>
<span id="cb16-55"><a href="#cb16-55" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;f1&quot;</span> <span class="ot">=</span> f1,</span>
<span id="cb16-56"><a href="#cb16-56" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;kappa&quot;</span> <span class="ot">=</span> kappa</span>
<span id="cb16-57"><a href="#cb16-57" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb16-58"><a href="#cb16-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb16-59"><a href="#cb16-59" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb16-60"><a href="#cb16-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-61"><a href="#cb16-61" aria-hidden="true" tabindex="-1"></a>train_dt <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/train_dtV2.rds&quot;</span>)</span>
<span id="cb16-62"><a href="#cb16-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-63"><a href="#cb16-63" aria-hidden="true" tabindex="-1"></a>test <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;~/DeepMeta/data/test_preV2.csv&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>X)</span>
<span id="cb16-64"><a href="#cb16-64" aria-hidden="true" tabindex="-1"></a>test_m <span class="ot">&lt;-</span> <span class="fu">get_roc</span>(test)</span>
<span id="cb16-65"><a href="#cb16-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-66"><a href="#cb16-66" aria-hidden="true" tabindex="-1"></a>sanger <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/sanger_pre.rds&quot;</span>)</span>
<span id="cb16-67"><a href="#cb16-67" aria-hidden="true" tabindex="-1"></a>sanger_common_cell <span class="ot">&lt;-</span> sanger <span class="sc">%&gt;%</span> <span class="fu">filter</span>(cell <span class="sc">%in%</span> train_dt<span class="sc">$</span>cell)</span>
<span id="cb16-68"><a href="#cb16-68" aria-hidden="true" tabindex="-1"></a>sanger_unique_cell <span class="ot">&lt;-</span> sanger <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>(cell <span class="sc">%in%</span> train_dt<span class="sc">$</span>cell))</span>
<span id="cb16-69"><a href="#cb16-69" aria-hidden="true" tabindex="-1"></a>sanger_seen_m <span class="ot">&lt;-</span> <span class="fu">get_roc</span>(sanger_common_cell)</span>
<span id="cb16-70"><a href="#cb16-70" aria-hidden="true" tabindex="-1"></a>sanger_unseen_m <span class="ot">&lt;-</span> <span class="fu">get_roc</span>(sanger_unique_cell)</span>
<span id="cb16-71"><a href="#cb16-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-72"><a href="#cb16-72" aria-hidden="true" tabindex="-1"></a>rnai <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;~/DeepMeta/data/rnai_preV2.csv&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>X)</span>
<span id="cb16-73"><a href="#cb16-73" aria-hidden="true" tabindex="-1"></a>rnai_seen <span class="ot">&lt;-</span> rnai <span class="sc">%&gt;%</span> <span class="fu">filter</span>(cell <span class="sc">%in%</span> train_dt<span class="sc">$</span>cell)</span>
<span id="cb16-74"><a href="#cb16-74" aria-hidden="true" tabindex="-1"></a>rnai_unseen <span class="ot">&lt;-</span> rnai <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>(cell <span class="sc">%in%</span> train_dt<span class="sc">$</span>cell))</span>
<span id="cb16-75"><a href="#cb16-75" aria-hidden="true" tabindex="-1"></a>rnai_seen_m <span class="ot">&lt;-</span> <span class="fu">get_roc</span>(rnai_seen)</span>
<span id="cb16-76"><a href="#cb16-76" aria-hidden="true" tabindex="-1"></a>rnai_unseen_m <span class="ot">&lt;-</span> <span class="fu">get_roc</span>(rnai_unseen)</span>
<span id="cb16-77"><a href="#cb16-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-78"><a href="#cb16-78" aria-hidden="true" tabindex="-1"></a>ccma <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/ccma_pre.rds&quot;</span>)</span>
<span id="cb16-79"><a href="#cb16-79" aria-hidden="true" tabindex="-1"></a>ccma_m <span class="ot">&lt;-</span> <span class="fu">get_roc</span>(ccma)</span>
<span id="cb16-80"><a href="#cb16-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-81"><a href="#cb16-81" aria-hidden="true" tabindex="-1"></a>dt <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb16-82"><a href="#cb16-82" aria-hidden="true" tabindex="-1"></a>  <span class="at">CCMA =</span> ccma_m,</span>
<span id="cb16-83"><a href="#cb16-83" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">RNAi-Seen</span><span class="st">`</span> <span class="ot">=</span> rnai_seen_m,</span>
<span id="cb16-84"><a href="#cb16-84" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">Sanger-Seen</span><span class="st">`</span> <span class="ot">=</span> sanger_seen_m,</span>
<span id="cb16-85"><a href="#cb16-85" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">RNAi-Unseen</span><span class="st">`</span> <span class="ot">=</span> rnai_unseen_m,</span>
<span id="cb16-86"><a href="#cb16-86" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">Sanger-Unseen</span><span class="st">`</span> <span class="ot">=</span> sanger_unseen_m,</span>
<span id="cb16-87"><a href="#cb16-87" aria-hidden="true" tabindex="-1"></a>  <span class="at">Test =</span> test_m,</span>
<span id="cb16-88"><a href="#cb16-88" aria-hidden="true" tabindex="-1"></a>  <span class="at">Metrics =</span> <span class="fu">c</span>(<span class="st">&quot;AUROC&quot;</span>,<span class="st">&quot;AUPRC&quot;</span>,<span class="st">&quot;F1&quot;</span>,<span class="st">&quot;Kappa&quot;</span>) </span>
<span id="cb16-89"><a href="#cb16-89" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">check.names =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-90"><a href="#cb16-90" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>,<span class="at">names_to =</span> <span class="st">&quot;Type&quot;</span>,<span class="at">values_to =</span> <span class="st">&quot;Value&quot;</span>)</span>
<span id="cb16-91"><a href="#cb16-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-92"><a href="#cb16-92" aria-hidden="true" tabindex="-1"></a><span class="fu">ggbarplot</span>(dt, <span class="at">x =</span> <span class="st">&quot;Type&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Value&quot;</span>,<span class="at">fill=</span><span class="st">&quot;Metrics&quot;</span>,</span>
<span id="cb16-93"><a href="#cb16-93" aria-hidden="true" tabindex="-1"></a>          <span class="at">label =</span> <span class="cn">TRUE</span>, </span>
<span id="cb16-94"><a href="#cb16-94" aria-hidden="true" tabindex="-1"></a>          <span class="at">lab.vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>,<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="fl">0.9</span>),</span>
<span id="cb16-95"><a href="#cb16-95" aria-hidden="true" tabindex="-1"></a>          <span class="at">order =</span> <span class="fu">c</span>(<span class="st">&quot;Test&quot;</span>,<span class="st">&quot;RNAi-Seen&quot;</span>,<span class="st">&quot;RNAi-Unseen&quot;</span>,<span class="st">&quot;Sanger-Seen&quot;</span>,</span>
<span id="cb16-96"><a href="#cb16-96" aria-hidden="true" tabindex="-1"></a>                    <span class="st">&quot;Sanger-Unseen&quot;</span>,<span class="st">&quot;CCMA&quot;</span>),</span>
<span id="cb16-97"><a href="#cb16-97" aria-hidden="true" tabindex="-1"></a>          <span class="at">palette =</span> <span class="fu">c</span>(<span class="st">&quot;#2B4F7D&quot;</span>,<span class="st">&quot;#3B77B0&quot;</span>,<span class="st">&quot;#EEDC96&quot;</span>,<span class="st">&quot;#637951&quot;</span>),</span>
<span id="cb16-98"><a href="#cb16-98" aria-hidden="true" tabindex="-1"></a>          <span class="at">lab.nb.digits =</span> <span class="dv">2</span>, <span class="at">xlab=</span>F)</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-8-1.png" width="2400" /></p>
<p>The slightly low metrics in RNAi dataset
(<code>Extended Data Fig. 4</code>) is partly due to the low correlation
between the RNAi and the CRISPR dataset:</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>rnai <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;/home/data/sdc/wt/model_data/rnai_gene_dependency.csv&quot;</span>,</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>                          <span class="at">data.table =</span> F)</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(rnai)[<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(rnai)] <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">s*</span><span class="sc">\\</span><span class="st">([^</span><span class="sc">\\</span><span class="st">)]+</span><span class="sc">\\</span><span class="st">)&quot;</span>,<span class="st">&quot;&quot;</span>,</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>                                     <span class="fu">colnames</span>(rnai)[<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(rnai)])</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>rnai <span class="ot">&lt;-</span> rnai <span class="sc">%&gt;%</span> </span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">colnames</span>(rnai)[<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(rnai)],<span class="at">names_to =</span> <span class="st">&quot;gene&quot;</span>,</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>                      <span class="at">values_to =</span> <span class="st">&quot;score&quot;</span>)</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(rnai)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">&quot;cell&quot;</span></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>dep_dt <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;/home/data/sdb/wt/model_data/dep_dt.rds&quot;</span>)</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>merge_dt <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>  dep_dt <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">cell =</span> ModelID, <span class="at">cri_score =</span> score),</span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>  rnai <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">rnai_score =</span> score),</span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;cell&quot;</span>,<span class="st">&quot;gene&quot;</span>)</span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">na.omit</span>(.)</span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>merge_dt <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(gene) <span class="sc">%&gt;%</span></span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">cor=</span><span class="fu">cor</span>(cri_score,rnai_score)) <span class="ot">-&gt;</span> tt</span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: There were 5 warnings in `summarise()`.</span></span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The first warning was:</span></span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ℹ In argument: `cor = cor(cri_score, rnai_score)`.</span></span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ℹ In group 10360: `gene = &quot;RAN&quot;`.</span></span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Caused by warning in `cor()`:</span></span>
<span id="cb17-23"><a href="#cb17-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ! the standard deviation is zero</span></span>
<span id="cb17-24"><a href="#cb17-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ℹ Run `dplyr::last_dplyr_warnings()` to see the 4 remaining warnings.</span></span>
<span id="cb17-25"><a href="#cb17-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-26"><a href="#cb17-26" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">gghistogram</span>(tt, <span class="at">x =</span> <span class="st">&quot;cor&quot;</span>, </span>
<span id="cb17-27"><a href="#cb17-27" aria-hidden="true" tabindex="-1"></a>                  <span class="at">fill =</span> <span class="st">&quot;#00AFBB&quot;</span>,</span>
<span id="cb17-28"><a href="#cb17-28" aria-hidden="true" tabindex="-1"></a>                  <span class="at">add =</span> <span class="st">&quot;median&quot;</span>, <span class="at">rug =</span> <span class="cn">TRUE</span>, </span>
<span id="cb17-29"><a href="#cb17-29" aria-hidden="true" tabindex="-1"></a>                  <span class="at">title =</span> <span class="st">&quot;Correlation of dependency score&quot;</span>)<span class="sc">+</span></span>
<span id="cb17-30"><a href="#cb17-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y=</span><span class="st">&quot;Gene Counts&quot;</span>,<span class="at">x=</span><span class="st">&quot;Correlation&quot;</span>)</span>
<span id="cb17-31"><a href="#cb17-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: Using `bins = 30` by default. Pick better value with the argument</span></span>
<span id="cb17-32"><a href="#cb17-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; `bins`.</span></span>
<span id="cb17-33"><a href="#cb17-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: `geom_vline()`: Ignoring `mapping` because `xintercept` was provided.</span></span>
<span id="cb17-34"><a href="#cb17-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: `geom_vline()`: Ignoring `data` because `xintercept` was provided.</span></span>
<span id="cb17-35"><a href="#cb17-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-36"><a href="#cb17-36" aria-hidden="true" tabindex="-1"></a>merge_dt <span class="ot">&lt;-</span> merge_dt <span class="sc">%&gt;%</span> </span>
<span id="cb17-37"><a href="#cb17-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">cri_type =</span> <span class="fu">case_when</span>(</span>
<span id="cb17-38"><a href="#cb17-38" aria-hidden="true" tabindex="-1"></a>    cri_score <span class="sc">&gt;</span> <span class="fl">0.5</span> <span class="sc">~</span> <span class="st">&quot;Positive&quot;</span>,</span>
<span id="cb17-39"><a href="#cb17-39" aria-hidden="true" tabindex="-1"></a>    cri_score <span class="sc">&lt;=</span> <span class="fl">0.5</span> <span class="sc">~</span> <span class="st">&quot;Negative&quot;</span>,</span>
<span id="cb17-40"><a href="#cb17-40" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">&quot;None&quot;</span></span>
<span id="cb17-41"><a href="#cb17-41" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">%&gt;%</span> </span>
<span id="cb17-42"><a href="#cb17-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">rnai_type =</span> <span class="fu">case_when</span>(</span>
<span id="cb17-43"><a href="#cb17-43" aria-hidden="true" tabindex="-1"></a>    rnai_score <span class="sc">&gt;</span> <span class="fl">0.5</span> <span class="sc">~</span> <span class="st">&quot;Positive&quot;</span>,</span>
<span id="cb17-44"><a href="#cb17-44" aria-hidden="true" tabindex="-1"></a>    rnai_score <span class="sc">&lt;=</span> <span class="fl">0.5</span> <span class="sc">~</span> <span class="st">&quot;Negative&quot;</span>,</span>
<span id="cb17-45"><a href="#cb17-45" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">&quot;None&quot;</span></span>
<span id="cb17-46"><a href="#cb17-46" aria-hidden="true" tabindex="-1"></a>  ))</span>
<span id="cb17-47"><a href="#cb17-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-48"><a href="#cb17-48" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggalluvial)</span>
<span id="cb17-49"><a href="#cb17-49" aria-hidden="true" tabindex="-1"></a>merge_summ <span class="ot">&lt;-</span> merge_dt <span class="sc">%&gt;%</span> </span>
<span id="cb17-50"><a href="#cb17-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(cri_type) <span class="sc">%&gt;%</span> </span>
<span id="cb17-51"><a href="#cb17-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">Negative =</span> <span class="fu">sum</span>(rnai_type <span class="sc">==</span> <span class="st">&quot;Negative&quot;</span>),</span>
<span id="cb17-52"><a href="#cb17-52" aria-hidden="true" tabindex="-1"></a>            <span class="at">Positive =</span> <span class="fu">sum</span>(rnai_type <span class="sc">==</span> <span class="st">&quot;Positive&quot;</span>),</span>
<span id="cb17-53"><a href="#cb17-53" aria-hidden="true" tabindex="-1"></a>            <span class="at">None =</span> <span class="fu">sum</span>(rnai_type <span class="sc">==</span> <span class="st">&quot;None&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-54"><a href="#cb17-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb17-55"><a href="#cb17-55" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(<span class="st">&quot;Negative&quot;</span>,<span class="st">&quot;Positive&quot;</span>,<span class="st">&quot;None&quot;</span>),<span class="at">names_to =</span> <span class="st">&quot;Rnai&quot;</span>,</span>
<span id="cb17-56"><a href="#cb17-56" aria-hidden="true" tabindex="-1"></a>                      <span class="at">values_to =</span> <span class="st">&quot;freq&quot;</span>)</span>
<span id="cb17-57"><a href="#cb17-57" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> merge_summ,</span>
<span id="cb17-58"><a href="#cb17-58" aria-hidden="true" tabindex="-1"></a>             <span class="fu">aes</span>(<span class="at">axis1 =</span> cri_type, <span class="at">axis2 =</span> Rnai, <span class="at">y =</span> freq)) <span class="sc">+</span></span>
<span id="cb17-59"><a href="#cb17-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_alluvium</span>(<span class="fu">aes</span>(<span class="at">fill =</span> Rnai),<span class="at">show.legend=</span>T) <span class="sc">+</span></span>
<span id="cb17-60"><a href="#cb17-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_stratum</span>() <span class="sc">+</span></span>
<span id="cb17-61"><a href="#cb17-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">stat =</span> <span class="st">&quot;stratum&quot;</span>,</span>
<span id="cb17-62"><a href="#cb17-62" aria-hidden="true" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">after_stat</span>(stratum))) <span class="sc">+</span></span>
<span id="cb17-63"><a href="#cb17-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_void</span>()<span class="sc">+</span></span>
<span id="cb17-64"><a href="#cb17-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;Change of label between CRISPR and RNAi&quot;</span>)</span>
<span id="cb17-65"><a href="#cb17-65" aria-hidden="true" tabindex="-1"></a>p1 <span class="sc">+</span> p2</span>
<span id="cb17-66"><a href="#cb17-66" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: Removed 5 rows containing non-finite values (`stat_bin()`).</span></span>
<span id="cb17-67"><a href="#cb17-67" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in to_lodes_form(data = data, axes = axis_ind, discern =</span></span>
<span id="cb17-68"><a href="#cb17-68" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; params$discern): Some strata appear at multiple axes.</span></span>
<span id="cb17-69"><a href="#cb17-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-70"><a href="#cb17-70" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in to_lodes_form(data = data, axes = axis_ind, discern =</span></span>
<span id="cb17-71"><a href="#cb17-71" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; params$discern): Some strata appear at multiple axes.</span></span>
<span id="cb17-72"><a href="#cb17-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-73"><a href="#cb17-73" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in to_lodes_form(data = data, axes = axis_ind, discern =</span></span>
<span id="cb17-74"><a href="#cb17-74" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; params$discern): Some strata appear at multiple axes.</span></span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-9-1.png" width="2400" /></p>
<p>Explaining complex deep learning model helps us to understand its
learning capabilities. The GAT model, due to its utilization of
attention mechanisms during model training, is a naturally interpretable
model. Thus, we extracted attention weights from the GAT model to
determine the importance of a gene’s neighbor genes for its prediction
and defined the neighbor importance score:</p>
<p><img
src="https://picgo-wutao.oss-cn-shanghai.aliyuncs.com/image-20240122091735364.png" /></p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>train_dtV2 <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/train_dtV2.rds&quot;</span>) </span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>train_dtV2 <span class="ot">&lt;-</span> train_dtV2 <span class="sc">%&gt;%</span> </span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">index =</span> <span class="fu">paste</span>(cell,id,<span class="at">sep =</span> <span class="st">&quot;-&quot;</span>))</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a><span class="do">##atten weight</span></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">&quot;/home/data/sdc/wt/model_data/autodl_143/autodl-tmp/atten/&quot;</span>,<span class="at">full.names =</span> T)</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>  files,</span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>  <span class="cf">function</span>(x){data.table<span class="sc">::</span><span class="fu">fread</span>(x,<span class="at">data.table =</span> F)}</span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(res)</span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> res <span class="sc">%&gt;%</span> </span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">index =</span> <span class="fu">paste</span>(cell,genes_x,<span class="at">sep =</span> <span class="st">&quot;-&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(index <span class="sc">%in%</span> train_dtV2<span class="sc">$</span>index) <span class="sc">%&gt;%</span> </span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(.,</span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>            train_dtV2 <span class="sc">%&gt;%</span> <span class="fu">select</span>(index,gene,is_dep),</span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>            <span class="at">by =</span> <span class="st">&quot;index&quot;</span>)</span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a>res<span class="sc">$</span>weight <span class="ot">&lt;-</span> <span class="fu">rowMeans</span>(res[,<span class="fu">c</span>(<span class="st">&quot;w1&quot;</span>,<span class="st">&quot;w2&quot;</span>,<span class="st">&quot;w3&quot;</span>)])</span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a>enz_gene_mapping <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/enz_gene_mapping.rds&quot;</span>)</span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a>ensg2name <span class="ot">&lt;-</span> <span class="cf">function</span>(ensg,mapping){</span>
<span id="cb18-23"><a href="#cb18-23" aria-hidden="true" tabindex="-1"></a>  tt <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(ensg,<span class="st">&quot; and &quot;</span>)[[<span class="dv">1</span>]]</span>
<span id="cb18-24"><a href="#cb18-24" aria-hidden="true" tabindex="-1"></a>  tt_gene <span class="ot">&lt;-</span> mapping<span class="sc">$</span>symbol[<span class="fu">which</span>(mapping<span class="sc">$</span>ensembl_id <span class="sc">%in%</span> tt)] <span class="sc">%&gt;%</span> </span>
<span id="cb18-25"><a href="#cb18-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">unique</span>() </span>
<span id="cb18-26"><a href="#cb18-26" aria-hidden="true" tabindex="-1"></a>  sym <span class="ot">&lt;-</span> tt_gene <span class="sc">%&gt;%</span> </span>
<span id="cb18-27"><a href="#cb18-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">paste</span>(.,<span class="at">collapse=</span><span class="st">&quot;,&quot;</span>)</span>
<span id="cb18-28"><a href="#cb18-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(sym)</span>
<span id="cb18-29"><a href="#cb18-29" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb18-30"><a href="#cb18-30" aria-hidden="true" tabindex="-1"></a>all_genes <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">genes_y =</span> <span class="fu">unique</span>(res<span class="sc">$</span>genes_y))</span>
<span id="cb18-31"><a href="#cb18-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-32"><a href="#cb18-32" aria-hidden="true" tabindex="-1"></a>all_genes <span class="ot">&lt;-</span> all_genes <span class="sc">%&gt;%</span> </span>
<span id="cb18-33"><a href="#cb18-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb18-34"><a href="#cb18-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">gene_y =</span> <span class="fu">ensg2name</span>(genes_y, enz_gene_mapping)) <span class="sc">%&gt;%</span> </span>
<span id="cb18-35"><a href="#cb18-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb18-36"><a href="#cb18-36" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">left_join</span>(res,all_genes)</span>
<span id="cb18-37"><a href="#cb18-37" aria-hidden="true" tabindex="-1"></a>res_filter <span class="ot">&lt;-</span> res <span class="sc">%&gt;%</span> <span class="fu">filter</span>(preds_x <span class="sc">==</span> is_dep) <span class="do">##提取预测正确的样本</span></span>
<span id="cb18-38"><a href="#cb18-38" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(res_filter,<span class="at">file =</span> <span class="st">&quot;/home/data/sdc/wt/model_data/atten_weightV2.rds&quot;</span>)</span>
<span id="cb18-39"><a href="#cb18-39" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span>
<span id="cb18-40"><a href="#cb18-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-41"><a href="#cb18-41" aria-hidden="true" tabindex="-1"></a>atten_weight <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;/home/data/sdc/wt/model_data/atten_weightV2.rds&quot;</span>)</span>
<span id="cb18-42"><a href="#cb18-42" aria-hidden="true" tabindex="-1"></a>weight <span class="ot">&lt;-</span> atten_weight <span class="sc">%&gt;%</span> </span>
<span id="cb18-43"><a href="#cb18-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(cell,gene,gene_y,weight) <span class="sc">%&gt;%</span> <span class="fu">distinct_all</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb18-44"><a href="#cb18-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(gene,cell) <span class="sc">%&gt;%</span></span>
<span id="cb18-45"><a href="#cb18-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_max</span>(<span class="at">order_by =</span> weight,<span class="at">n =</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>()</span>
<span id="cb18-46"><a href="#cb18-46" aria-hidden="true" tabindex="-1"></a>cell_counts <span class="ot">&lt;-</span> atten_weight <span class="sc">%&gt;%</span> </span>
<span id="cb18-47"><a href="#cb18-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(cell,gene,gene_y,weight) <span class="sc">%&gt;%</span> </span>
<span id="cb18-48"><a href="#cb18-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(gene) <span class="sc">%&gt;%</span> </span>
<span id="cb18-49"><a href="#cb18-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">counts =</span> <span class="fu">length</span>(<span class="fu">unique</span>(cell))) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>()</span>
<span id="cb18-50"><a href="#cb18-50" aria-hidden="true" tabindex="-1"></a>gene_attr <span class="ot">&lt;-</span> weight <span class="sc">%&gt;%</span> </span>
<span id="cb18-51"><a href="#cb18-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(weight <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb18-52"><a href="#cb18-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(gene,gene_y) <span class="sc">%&gt;%</span> </span>
<span id="cb18-53"><a href="#cb18-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">occour_counts =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb18-54"><a href="#cb18-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb18-55"><a href="#cb18-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(.,cell_counts) <span class="sc">%&gt;%</span> </span>
<span id="cb18-56"><a href="#cb18-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">imp =</span> occour_counts<span class="sc">/</span>counts)</span>
<span id="cb18-57"><a href="#cb18-57" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(gene_attr,<span class="at">file =</span> <span class="st">&quot;data/local_imp.rds&quot;</span>)</span></code></pre></div>
<p>For each pair of gene, we computed the correlation of their DepMap
dependency scores and compared them with the neighbor importance scores.
In comparison to groups with low dependency correlation (R &lt; 0.2),
those with higher dependency correlation (R &gt; 0.8) exhibited
significantly elevated neighbor importance scores:</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="do">###cor</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>local_imp <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/local_imp.rds&quot;</span>)</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>all_genes <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(<span class="fu">paste</span>(<span class="fu">paste</span>(local_imp<span class="sc">$</span>gene,<span class="at">collapse =</span> <span class="st">&quot;,&quot;</span>),</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">paste</span>(local_imp<span class="sc">$</span>gene_y,<span class="at">collapse =</span> <span class="st">&quot;,&quot;</span>),</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>                            <span class="at">collapse =</span> <span class="st">&quot;,&quot;</span>), <span class="at">split =</span> <span class="st">&quot;,&quot;</span>)[[<span class="dv">1</span>]] <span class="sc">%&gt;%</span> <span class="fu">unique</span>()</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>dep_dt <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;/home/data/sdb/wt/model_data/dep_dt.rds&quot;</span>)</span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>dep_dt <span class="ot">&lt;-</span> dep_dt <span class="sc">%&gt;%</span> <span class="fu">filter</span>(gene <span class="sc">%in%</span> all_genes)</span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>dep_dt <span class="ot">&lt;-</span> dep_dt <span class="sc">%&gt;%</span></span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">names_from =</span> <span class="st">&quot;gene&quot;</span>,<span class="at">values_from =</span> <span class="st">&quot;score&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a>cor_res <span class="ot">&lt;-</span> WGCNA<span class="sc">::</span><span class="fu">cor</span>(<span class="fu">as.matrix</span>(dep_dt), <span class="at">use =</span> <span class="st">&quot;pairwise.complete.obs&quot;</span>)</span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in storage.mode(x) &lt;- &quot;double&quot;: NAs introduced by coercion</span></span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a>cor_res <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(cor_res)</span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a>cor_res<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(cor_res)</span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a>cor_res <span class="ot">&lt;-</span> cor_res <span class="sc">%&gt;%</span></span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2438</span>,</span>
<span id="cb19-19"><a href="#cb19-19" aria-hidden="true" tabindex="-1"></a>                      <span class="at">names_to =</span> <span class="st">&quot;targt_genes&quot;</span>,<span class="at">values_to =</span> <span class="st">&quot;cor&quot;</span>)</span>
<span id="cb19-20"><a href="#cb19-20" aria-hidden="true" tabindex="-1"></a>cor_res <span class="ot">&lt;-</span> cor_res <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">index =</span> <span class="fu">paste</span>(gene,targt_genes,<span class="at">sep =</span> <span class="st">&quot;-&quot;</span>))</span>
<span id="cb19-21"><a href="#cb19-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-22"><a href="#cb19-22" aria-hidden="true" tabindex="-1"></a>gene_attr_split <span class="ot">&lt;-</span> local_imp <span class="sc">%&gt;%</span></span>
<span id="cb19-23"><a href="#cb19-23" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">separate_longer_delim</span>(<span class="at">cols =</span> <span class="fu">c</span>(<span class="st">&quot;gene&quot;</span>),<span class="at">delim =</span> <span class="st">&quot;,&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-24"><a href="#cb19-24" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">separate_longer_delim</span>(<span class="at">cols =</span> <span class="fu">c</span>(<span class="st">&quot;gene_y&quot;</span>),<span class="at">delim =</span> <span class="st">&quot;,&quot;</span>)</span>
<span id="cb19-25"><a href="#cb19-25" aria-hidden="true" tabindex="-1"></a>cor_res_all <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(</span>
<span id="cb19-26"><a href="#cb19-26" aria-hidden="true" tabindex="-1"></a>  cor_res,</span>
<span id="cb19-27"><a href="#cb19-27" aria-hidden="true" tabindex="-1"></a>  gene_attr_split <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">index =</span> <span class="fu">paste</span>(gene,gene_y,<span class="at">sep =</span> <span class="st">&quot;-&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-28"><a href="#cb19-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(index,imp)</span>
<span id="cb19-29"><a href="#cb19-29" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-30"><a href="#cb19-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">cor_type =</span> <span class="fu">case_when</span>(</span>
<span id="cb19-31"><a href="#cb19-31" aria-hidden="true" tabindex="-1"></a>    cor <span class="sc">&gt;</span> <span class="fl">0.8</span> <span class="sc">~</span> <span class="st">&quot;High Dependency Cor </span><span class="sc">\n</span><span class="st">(&gt;0.8)&quot;</span>,</span>
<span id="cb19-32"><a href="#cb19-32" aria-hidden="true" tabindex="-1"></a>    cor <span class="sc">&lt;</span> <span class="fl">0.2</span> <span class="sc">~</span> <span class="st">&quot;Low Dependency Cor </span><span class="sc">\n</span><span class="st">(&lt;0.2)&quot;</span>,</span>
<span id="cb19-33"><a href="#cb19-33" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">&quot;others&quot;</span></span>
<span id="cb19-34"><a href="#cb19-34" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(cor_type <span class="sc">!=</span> <span class="st">&quot;others&quot;</span>)</span>
<span id="cb19-35"><a href="#cb19-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(index)`</span></span>
<span id="cb19-36"><a href="#cb19-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-37"><a href="#cb19-37" aria-hidden="true" tabindex="-1"></a><span class="fu">ggboxplot</span>(<span class="at">data=</span>cor_res_all,<span class="at">x=</span><span class="st">&quot;cor_type&quot;</span>,<span class="at">y=</span><span class="st">&quot;imp&quot;</span>,<span class="at">xlab =</span> <span class="cn">FALSE</span>,</span>
<span id="cb19-38"><a href="#cb19-38" aria-hidden="true" tabindex="-1"></a>          <span class="at">ylab =</span> <span class="st">&quot;Local importance score&quot;</span>,</span>
<span id="cb19-39"><a href="#cb19-39" aria-hidden="true" tabindex="-1"></a>          <span class="at">order =</span> <span class="fu">c</span>(<span class="st">&quot;Low Dependency Cor </span><span class="sc">\n</span><span class="st">(&lt;0.2)&quot;</span>,<span class="st">&quot;High Dependency Cor </span><span class="sc">\n</span><span class="st">(&gt;0.8)&quot;</span>))<span class="sc">+</span></span>
<span id="cb19-40"><a href="#cb19-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>()</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-11-1.png" width="2400" /></p>
<p>This demonstrates that our model could utilize attention mechanism to
learn the metabolic dependency associations between genes. We show the
genes that are crucial for predicting the metabolic dependency of the
GPX4 as an example:</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(igraph)</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Attaching package: &#39;igraph&#39;</span></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The following objects are masked from &#39;package:dplyr&#39;:</span></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     as_data_frame, groups, union</span></span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The following objects are masked from &#39;package:stats&#39;:</span></span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     decompose, spectrum</span></span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The following object is masked from &#39;package:base&#39;:</span></span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     union</span></span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a>get_atten_plot <span class="ot">&lt;-</span> <span class="cf">function</span>(pre_dt,gene_name){</span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a>  test <span class="ot">&lt;-</span> pre_dt <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(gene <span class="sc">==</span> gene_name)</span>
<span id="cb20-15"><a href="#cb20-15" aria-hidden="true" tabindex="-1"></a>  <span class="do">##选择权重大于0的top3基因</span></span>
<span id="cb20-16"><a href="#cb20-16" aria-hidden="true" tabindex="-1"></a>  test_summ <span class="ot">&lt;-</span> test <span class="sc">%&gt;%</span></span>
<span id="cb20-17"><a href="#cb20-17" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">group_by</span>(cell) <span class="sc">%&gt;%</span></span>
<span id="cb20-18"><a href="#cb20-18" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">slice_max</span>(<span class="at">order_by =</span> weight, <span class="at">n =</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb20-19"><a href="#cb20-19" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(weight <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb20-20"><a href="#cb20-20" aria-hidden="true" tabindex="-1"></a>  all_cell_counts <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">unique</span>(test_summ<span class="sc">$</span>cell))</span>
<span id="cb20-21"><a href="#cb20-21" aria-hidden="true" tabindex="-1"></a>  gene_summ <span class="ot">&lt;-</span> test_summ <span class="sc">%&gt;%</span> </span>
<span id="cb20-22"><a href="#cb20-22" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">group_by</span>(gene_y) <span class="sc">%&gt;%</span> </span>
<span id="cb20-23"><a href="#cb20-23" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">cell_counts =</span> <span class="fu">length</span>(<span class="fu">unique</span>(cell))) <span class="sc">%&gt;%</span> </span>
<span id="cb20-24"><a href="#cb20-24" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb20-25"><a href="#cb20-25" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">weight =</span> cell_counts<span class="sc">/</span>all_cell_counts) <span class="sc">%&gt;%</span> </span>
<span id="cb20-26"><a href="#cb20-26" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">source =</span> gene_name) <span class="sc">%&gt;%</span> </span>
<span id="cb20-27"><a href="#cb20-27" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">target =</span> gene_y) <span class="sc">%&gt;%</span> </span>
<span id="cb20-28"><a href="#cb20-28" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(source, target, weight) <span class="sc">%&gt;%</span> </span>
<span id="cb20-29"><a href="#cb20-29" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(target <span class="sc">!=</span> gene_name)</span>
<span id="cb20-30"><a href="#cb20-30" aria-hidden="true" tabindex="-1"></a>  gene_g <span class="ot">&lt;-</span> <span class="fu">graph_from_data_frame</span>(gene_summ,<span class="at">directed =</span> F)</span>
<span id="cb20-31"><a href="#cb20-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">V</span>(gene_g)[gene_summ<span class="sc">$</span>target]<span class="sc">$</span>color <span class="ot">&lt;-</span> <span class="st">&quot;#11325D&quot;</span></span>
<span id="cb20-32"><a href="#cb20-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">V</span>(gene_g)[gene_summ<span class="sc">$</span>source]<span class="sc">$</span>color <span class="ot">&lt;-</span> <span class="st">&quot;#F5A673&quot;</span></span>
<span id="cb20-33"><a href="#cb20-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">par</span>(<span class="at">mar=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>)<span class="sc">+</span>.<span class="dv">2</span>)</span>
<span id="cb20-34"><a href="#cb20-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(gene_g,<span class="at">vertex.size=</span><span class="dv">8</span>, <span class="at">edge.arrow.size=</span><span class="fl">0.3</span>,</span>
<span id="cb20-35"><a href="#cb20-35" aria-hidden="true" tabindex="-1"></a>       <span class="at">vertex.label.dist=</span><span class="fl">1.5</span>, <span class="at">vertex.label.cex =</span> <span class="fl">0.8</span>,</span>
<span id="cb20-36"><a href="#cb20-36" aria-hidden="true" tabindex="-1"></a>       <span class="at">edge.width =</span> <span class="fu">E</span>(gene_g)<span class="sc">$</span>weight<span class="sc">*</span><span class="dv">20</span>,<span class="at">layout=</span>layout.circle)</span>
<span id="cb20-37"><a href="#cb20-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(gene_summ)</span>
<span id="cb20-38"><a href="#cb20-38" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb20-39"><a href="#cb20-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-40"><a href="#cb20-40" aria-hidden="true" tabindex="-1"></a>atten_weight <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;/home/data/sdc/wt/model_data/atten_weightV2.rds&quot;</span>)</span>
<span id="cb20-41"><a href="#cb20-41" aria-hidden="true" tabindex="-1"></a><span class="fu">get_atten_plot</span>(atten_weight, <span class="at">gene_name =</span> <span class="st">&quot;GPX4&quot;</span>)</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-12-1.png" width="2400" /></p>
<pre><code>#&gt; # A tibble: 25 × 3
#&gt;    source target   weight
#&gt;    &lt;chr&gt;  &lt;chr&gt;     &lt;dbl&gt;
#&gt;  1 GPX4   ABCC5   0.0136 
#&gt;  2 GPX4   ACSL3   0.0680 
#&gt;  3 GPX4   AQP10   0.00907
#&gt;  4 GPX4   AQP7    0.0862 
#&gt;  5 GPX4   CRAT    0.00907
#&gt;  6 GPX4   CYP3A5  0.0136 
#&gt;  7 GPX4   CYP4F11 0.0635 
#&gt;  8 GPX4   ELOVL7  0.0317 
#&gt;  9 GPX4   ESD     0.0658 
#&gt; 10 GPX4   GSR     0.542  
#&gt; # ℹ 15 more rows</code></pre>
<p>The model reveales that the most essential gene for predicting GPX4’s
metabolic dependency is GSR, which aligns with existing study (Paper:
Metabolic determinants of cancer cell sensitivity to canonical
ferroptosis inducers, Figure1):</p>
<p><img
src="https://picgo-wutao.oss-cn-shanghai.aliyuncs.com/image-20240122141436842.png" /></p>
<p>Soula et al. used CRISPR-Cas9 genetic screens to target 2,998
metabolic genes under GPX4 inhibition by RSL3 and GSR was among top
negative scoring genes under RSL3 treatment, indicting loss of GSR
strongly potentiates RSL3 toxicity.</p>
</div>
<div id="tcga-analysis" class="section level1">
<h1>TCGA analysis</h1>
<p>We applied the DeepMeta to TCGA samples, including 8937 samples of 25
cancer types, to predict metabolic dependencies at the pan-cancer
level.</p>
<div id="pan-cancer-metabolic-dependency" class="section level3">
<h3>Pan-cancer metabolic dependency</h3>
<p>Genes acting in the same cellular pathway should show similar
patterns of essentiality. We calculated the Spearman correlation between
the predicted dependency probabilities of genes.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/kegg_all_pathway.rds&quot;</span>)</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> kegg <span class="sc">%&gt;%</span> </span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">&quot;Metabolism&quot;</span>,class) <span class="sc">|</span> <span class="fu">grepl</span>(<span class="st">&quot;metabolism&quot;</span>,pathway)) <span class="sc">%&gt;%</span> </span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pathway =</span> <span class="fu">gsub</span>(<span class="st">&quot; </span><span class="sc">\\</span><span class="st">- Homo sapiens </span><span class="sc">\\</span><span class="st">(human</span><span class="sc">\\</span><span class="st">)&quot;</span>,<span class="st">&quot;&quot;</span>,pathway))</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>get_pathway <span class="ot">&lt;-</span> <span class="cf">function</span>(gene,pathway){</span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>  gene <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(gene,<span class="at">split =</span> <span class="st">&quot;,&quot;</span>)[[<span class="dv">1</span>]]</span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>  pathway <span class="ot">&lt;-</span> pathway <span class="sc">%&gt;%</span> <span class="fu">filter</span>(genes <span class="sc">%in%</span> gene)</span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">c</span>(<span class="fu">paste</span>(<span class="fu">unique</span>(pathway<span class="sc">$</span>pathway),<span class="at">collapse =</span> <span class="st">&quot;;&quot;</span>),</span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>           <span class="fu">paste</span>(<span class="fu">unique</span>(pathway<span class="sc">$</span>class),<span class="at">collapse =</span> <span class="st">&quot;;&quot;</span>)))</span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a>pre <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/tcga_pre_V2.rds&quot;</span>)</span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a>pre_pathway <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene =</span> <span class="fu">unique</span>(<span class="fu">c</span>(pre<span class="sc">$</span>gene))) <span class="sc">%&gt;%</span> </span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pathway =</span> <span class="fu">get_pathway</span>(gene,kegg)[<span class="dv">1</span>],</span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a>         <span class="at">class =</span> <span class="fu">get_pathway</span>(gene,kegg)[<span class="dv">2</span>]) <span class="sc">%&gt;%</span> </span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">nchar</span>(pathway)<span class="sc">&gt;</span><span class="dv">1</span>)</span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a>pre_pathway <span class="ot">&lt;-</span> pre <span class="sc">%&gt;%</span> </span>
<span id="cb22-21"><a href="#cb22-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(.,pre_pathway) <span class="sc">%&gt;%</span> <span class="fu">na.omit</span>()</span>
<span id="cb22-22"><a href="#cb22-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-23"><a href="#cb22-23" aria-hidden="true" tabindex="-1"></a><span class="do">####两两基因之间的代谢依赖性预测相关性</span></span>
<span id="cb22-24"><a href="#cb22-24" aria-hidden="true" tabindex="-1"></a>all_gene <span class="ot">&lt;-</span> <span class="fu">unique</span>(pre_pathway<span class="sc">$</span>gene)</span>
<span id="cb22-25"><a href="#cb22-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-26"><a href="#cb22-26" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(doParallel)</span>
<span id="cb22-27"><a href="#cb22-27" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(foreach)</span>
<span id="cb22-28"><a href="#cb22-28" aria-hidden="true" tabindex="-1"></a><span class="co">#create the cluster</span></span>
<span id="cb22-29"><a href="#cb22-29" aria-hidden="true" tabindex="-1"></a>my.cluster <span class="ot">&lt;-</span> parallel<span class="sc">::</span><span class="fu">makeCluster</span>(</span>
<span id="cb22-30"><a href="#cb22-30" aria-hidden="true" tabindex="-1"></a>  <span class="dv">80</span>, </span>
<span id="cb22-31"><a href="#cb22-31" aria-hidden="true" tabindex="-1"></a>  <span class="at">type =</span> <span class="st">&quot;PSOCK&quot;</span></span>
<span id="cb22-32"><a href="#cb22-32" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb22-33"><a href="#cb22-33" aria-hidden="true" tabindex="-1"></a><span class="co">#register it to be used by %dopar%</span></span>
<span id="cb22-34"><a href="#cb22-34" aria-hidden="true" tabindex="-1"></a>doParallel<span class="sc">::</span><span class="fu">registerDoParallel</span>(<span class="at">cl =</span> my.cluster)</span>
<span id="cb22-35"><a href="#cb22-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-36"><a href="#cb22-36" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">foreach</span>(</span>
<span id="cb22-37"><a href="#cb22-37" aria-hidden="true" tabindex="-1"></a>  <span class="at">i =</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(all_gene),</span>
<span id="cb22-38"><a href="#cb22-38" aria-hidden="true" tabindex="-1"></a>  <span class="at">.export =</span> <span class="fu">c</span>(<span class="st">&quot;pre_pathway&quot;</span>,<span class="st">&quot;all_gene&quot;</span>),</span>
<span id="cb22-39"><a href="#cb22-39" aria-hidden="true" tabindex="-1"></a>  <span class="at">.packages =</span> <span class="fu">c</span>(<span class="st">&quot;dplyr&quot;</span>)</span>
<span id="cb22-40"><a href="#cb22-40" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%dopar%</span> {</span>
<span id="cb22-41"><a href="#cb22-41" aria-hidden="true" tabindex="-1"></a>  gene_dt <span class="ot">&lt;-</span> pre_pathway <span class="sc">%&gt;%</span> </span>
<span id="cb22-42"><a href="#cb22-42" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(gene <span class="sc">==</span> all_gene[i]) <span class="sc">%&gt;%</span> </span>
<span id="cb22-43"><a href="#cb22-43" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(cell, preds_raw) <span class="sc">%&gt;%</span> </span>
<span id="cb22-44"><a href="#cb22-44" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rename</span>(<span class="at">preds2 =</span> preds_raw)</span>
<span id="cb22-45"><a href="#cb22-45" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb22-46"><a href="#cb22-46" aria-hidden="true" tabindex="-1"></a>  gene_res <span class="ot">&lt;-</span> <span class="fu">sapply</span>(all_gene,</span>
<span id="cb22-47"><a href="#cb22-47" aria-hidden="true" tabindex="-1"></a>                     <span class="cf">function</span>(x){</span>
<span id="cb22-48"><a href="#cb22-48" aria-hidden="true" tabindex="-1"></a>                       dt <span class="ot">&lt;-</span> pre_pathway <span class="sc">%&gt;%</span> <span class="fu">filter</span>(gene <span class="sc">==</span> x)</span>
<span id="cb22-49"><a href="#cb22-49" aria-hidden="true" tabindex="-1"></a>                       dt <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(</span>
<span id="cb22-50"><a href="#cb22-50" aria-hidden="true" tabindex="-1"></a>                         dt <span class="sc">%&gt;%</span> <span class="fu">select</span>(cell,preds_raw),</span>
<span id="cb22-51"><a href="#cb22-51" aria-hidden="true" tabindex="-1"></a>                         gene_dt, <span class="at">by =</span> <span class="st">&quot;cell&quot;</span></span>
<span id="cb22-52"><a href="#cb22-52" aria-hidden="true" tabindex="-1"></a>                       ) </span>
<span id="cb22-53"><a href="#cb22-53" aria-hidden="true" tabindex="-1"></a>                       dt_res <span class="ot">&lt;-</span> <span class="fu">tryCatch</span>({<span class="fu">cor.test</span>(dt<span class="sc">$</span>preds_raw,dt<span class="sc">$</span>preds2,</span>
<span id="cb22-54"><a href="#cb22-54" aria-hidden="true" tabindex="-1"></a>                                                    <span class="at">method =</span> <span class="st">&quot;sp&quot;</span>)},</span>
<span id="cb22-55"><a href="#cb22-55" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">error =</span> <span class="cf">function</span>(e){</span>
<span id="cb22-56"><a href="#cb22-56" aria-hidden="true" tabindex="-1"></a>                                            <span class="fu">return</span>(<span class="fu">data.frame</span>(<span class="at">tt=</span><span class="st">&quot;error&quot;</span>))</span>
<span id="cb22-57"><a href="#cb22-57" aria-hidden="true" tabindex="-1"></a>                                          })</span>
<span id="cb22-58"><a href="#cb22-58" aria-hidden="true" tabindex="-1"></a>                       <span class="cf">if</span> (<span class="fu">class</span>(dt_res) <span class="sc">==</span> <span class="st">&quot;data.frame&quot;</span>){</span>
<span id="cb22-59"><a href="#cb22-59" aria-hidden="true" tabindex="-1"></a>                         <span class="fu">return</span>(<span class="cn">NA</span>)</span>
<span id="cb22-60"><a href="#cb22-60" aria-hidden="true" tabindex="-1"></a>                       }<span class="cf">else</span>{</span>
<span id="cb22-61"><a href="#cb22-61" aria-hidden="true" tabindex="-1"></a>                         <span class="fu">return</span>(dt_res<span class="sc">$</span>estimate)</span>
<span id="cb22-62"><a href="#cb22-62" aria-hidden="true" tabindex="-1"></a>                       }</span>
<span id="cb22-63"><a href="#cb22-63" aria-hidden="true" tabindex="-1"></a>                     })</span>
<span id="cb22-64"><a href="#cb22-64" aria-hidden="true" tabindex="-1"></a>  gene_res <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(gene_res)</span>
<span id="cb22-65"><a href="#cb22-65" aria-hidden="true" tabindex="-1"></a>  gene_res<span class="sc">$</span>genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(gene_res)</span>
<span id="cb22-66"><a href="#cb22-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(gene_res)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">&quot;cor&quot;</span></span>
<span id="cb22-67"><a href="#cb22-67" aria-hidden="true" tabindex="-1"></a>  gene_res<span class="sc">$</span>genes <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;.rho&quot;</span>,<span class="st">&quot;&quot;</span>,gene_res<span class="sc">$</span>genes)</span>
<span id="cb22-68"><a href="#cb22-68" aria-hidden="true" tabindex="-1"></a>  gene_res<span class="sc">$</span>main_gene <span class="ot">&lt;-</span> all_gene[i]</span>
<span id="cb22-69"><a href="#cb22-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(gene_res)</span>
<span id="cb22-70"><a href="#cb22-70" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb22-71"><a href="#cb22-71" aria-hidden="true" tabindex="-1"></a>parallel<span class="sc">::</span><span class="fu">stopCluster</span>(<span class="at">cl =</span> my.cluster)</span>
<span id="cb22-72"><a href="#cb22-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-73"><a href="#cb22-73" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(res)</span>
<span id="cb22-74"><a href="#cb22-74" aria-hidden="true" tabindex="-1"></a>res_dt <span class="ot">&lt;-</span> res <span class="sc">%&gt;%</span> </span>
<span id="cb22-75"><a href="#cb22-75" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">names_from =</span> main_gene,</span>
<span id="cb22-76"><a href="#cb22-76" aria-hidden="true" tabindex="-1"></a>                     <span class="at">values_from =</span> cor) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb22-77"><a href="#cb22-77" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(res_dt) <span class="ot">&lt;-</span> res_dt<span class="sc">$</span>genes</span>
<span id="cb22-78"><a href="#cb22-78" aria-hidden="true" tabindex="-1"></a>res_dt <span class="ot">&lt;-</span> res_dt <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>genes)</span>
<span id="cb22-79"><a href="#cb22-79" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(res_dt,<span class="at">file =</span> <span class="st">&quot;~/DeepMeta/data/tcga_pre_cor_all_gene_rho.rds&quot;</span>)</span>
<span id="cb22-80"><a href="#cb22-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-81"><a href="#cb22-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-82"><a href="#cb22-82" aria-hidden="true" tabindex="-1"></a>tcga_pre_cor <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/tcga_pre_cor_all_gene_rho.rds&quot;</span>)</span>
<span id="cb22-83"><a href="#cb22-83" aria-hidden="true" tabindex="-1"></a>tcga_pre_cor<span class="sc">$</span>main_gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(tcga_pre_cor)</span>
<span id="cb22-84"><a href="#cb22-84" aria-hidden="true" tabindex="-1"></a>tcga_pre_cor <span class="ot">&lt;-</span> tcga_pre_cor <span class="sc">%&gt;%</span> </span>
<span id="cb22-85"><a href="#cb22-85" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(<span class="at">cols =</span> ACSL4<span class="sc">:</span>PGAM4, <span class="at">names_to =</span> <span class="st">&quot;target_gene&quot;</span>,</span>
<span id="cb22-86"><a href="#cb22-86" aria-hidden="true" tabindex="-1"></a>                      <span class="at">values_to =</span> <span class="st">&quot;cor&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb22-87"><a href="#cb22-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb22-88"><a href="#cb22-88" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">main_pathway =</span> <span class="fu">get_pathway</span>(main_gene,kegg)[<span class="dv">1</span>],</span>
<span id="cb22-89"><a href="#cb22-89" aria-hidden="true" tabindex="-1"></a>         <span class="at">target_pathway =</span> <span class="fu">get_pathway</span>(target_gene,kegg)[<span class="dv">1</span>]) <span class="sc">%&gt;%</span> </span>
<span id="cb22-90"><a href="#cb22-90" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb22-91"><a href="#cb22-91" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">nchar</span>(main_pathway)<span class="sc">&gt;</span><span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb22-92"><a href="#cb22-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">nchar</span>(target_pathway)<span class="sc">&gt;</span><span class="dv">1</span>)</span>
<span id="cb22-93"><a href="#cb22-93" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(tcga_pre_cor, <span class="at">file =</span> <span class="st">&quot;~/DeepMeta/data/gene_cor_add_pathway_rho.rds&quot;</span>)</span></code></pre></div>
<p>The ACLY gene encodes ATP citrate lyase that catalyzes the conversion
of citrate and coenzyme A to acetyl-CoA. We display the genes in the
citrate cycle metabolic (TCA) pathway in <code>Fig 3b</code>. The
metabolic dependence correlation between TCA pathway genes and ACLY is
significantly higher than the correlation between other pathway genes
and ACLY:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/kegg_all_pathway.rds&quot;</span>)</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> kegg <span class="sc">%&gt;%</span> </span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">&quot;Metabolism&quot;</span>,class) <span class="sc">|</span> <span class="fu">grepl</span>(<span class="st">&quot;metabolism&quot;</span>,pathway)) <span class="sc">%&gt;%</span> </span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pathway =</span> <span class="fu">gsub</span>(<span class="st">&quot; </span><span class="sc">\\</span><span class="st">- Homo sapiens </span><span class="sc">\\</span><span class="st">(human</span><span class="sc">\\</span><span class="st">)&quot;</span>,<span class="st">&quot;&quot;</span>,pathway))</span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>get_pathway <span class="ot">&lt;-</span> <span class="cf">function</span>(gene,pathway){</span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>  gene <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(gene,<span class="at">split =</span> <span class="st">&quot;,&quot;</span>)[[<span class="dv">1</span>]]</span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>  pathway <span class="ot">&lt;-</span> pathway <span class="sc">%&gt;%</span> <span class="fu">filter</span>(genes <span class="sc">%in%</span> gene)</span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">c</span>(<span class="fu">paste</span>(<span class="fu">unique</span>(pathway<span class="sc">$</span>pathway),<span class="at">collapse =</span> <span class="st">&quot;;&quot;</span>),</span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a>           <span class="fu">paste</span>(<span class="fu">unique</span>(pathway<span class="sc">$</span>class),<span class="at">collapse =</span> <span class="st">&quot;;&quot;</span>)))</span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a>tcga_pre_cor <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/gene_cor_add_pathway_rho.rds&quot;</span>)</span>
<span id="cb23-14"><a href="#cb23-14" aria-hidden="true" tabindex="-1"></a>tca_acly <span class="ot">&lt;-</span> kegg <span class="sc">%&gt;%</span> </span>
<span id="cb23-15"><a href="#cb23-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(pathway <span class="sc">==</span> <span class="st">&quot;Citrate cycle (TCA cycle)&quot;</span>)</span>
<span id="cb23-16"><a href="#cb23-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-17"><a href="#cb23-17" aria-hidden="true" tabindex="-1"></a>acly <span class="ot">&lt;-</span> tcga_pre_cor <span class="sc">%&gt;%</span> </span>
<span id="cb23-18"><a href="#cb23-18" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">separate_longer_delim</span>(<span class="at">cols =</span> <span class="st">&quot;target_pathway&quot;</span>, <span class="at">delim =</span> <span class="st">&quot;;&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb23-19"><a href="#cb23-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(main_gene <span class="sc">==</span> <span class="st">&quot;ACLY&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb23-20"><a href="#cb23-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(target_gene <span class="sc">!=</span> <span class="st">&quot;ACLY&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb23-21"><a href="#cb23-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">type =</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;Citrate cycle </span><span class="sc">\\</span><span class="st">(TCA cycle</span><span class="sc">\\</span><span class="st">)&quot;</span>,target_pathway),</span>
<span id="cb23-22"><a href="#cb23-22" aria-hidden="true" tabindex="-1"></a>                       <span class="st">&quot;TCA cycle pathway&quot;</span>,<span class="st">&quot;Other pathway&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb23-23"><a href="#cb23-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="dv">1</span>,<span class="dv">2</span>,type,cor) <span class="sc">%&gt;%</span> </span>
<span id="cb23-24"><a href="#cb23-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct_all</span>()</span>
<span id="cb23-25"><a href="#cb23-25" aria-hidden="true" tabindex="-1"></a><span class="fu">ggboxplot</span>(<span class="at">data=</span>acly,<span class="at">x=</span><span class="st">&quot;type&quot;</span>,<span class="at">y=</span><span class="st">&quot;cor&quot;</span>,</span>
<span id="cb23-26"><a href="#cb23-26" aria-hidden="true" tabindex="-1"></a>          <span class="at">xlab =</span> <span class="cn">FALSE</span>, <span class="at">ylab =</span> <span class="st">&quot;Spearman Correlation&quot;</span>)<span class="sc">+</span></span>
<span id="cb23-27"><a href="#cb23-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>()</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-14-1.png" width="2400" /></p>
<p>For quantitative representation, we calculated the degree to which
correlated genes (correlation coefficient &gt; 0.8) are enriched in the
metabolic pathway where the target gene are located, relative to all
other pathways, and performed permutation analysis to obtain a null
distribution to calculate the significance (Methods).</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>kegg_enricher <span class="ot">&lt;-</span> <span class="cf">function</span>(kegg_dt, gene_list, pathway_name){</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>  gene_list <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(gene_list,<span class="at">split =</span> <span class="st">&quot;,&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">unlist</span>()</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>  pathway_dt <span class="ot">&lt;-</span> kegg_dt <span class="sc">%&gt;%</span> </span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">isin =</span> <span class="fu">ifelse</span>(genes <span class="sc">%in%</span> gene_list,<span class="st">&quot;yes&quot;</span>,<span class="st">&quot;no&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">pathway_type =</span> <span class="fu">ifelse</span>(pathway <span class="sc">==</span> pathway_name,<span class="st">&quot;yes&quot;</span>,<span class="st">&quot;no&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(genes,pathway_type,isin) <span class="sc">%&gt;%</span> <span class="fu">distinct_all</span>() </span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>  in_pathway <span class="ot">&lt;-</span> pathway_dt <span class="sc">%&gt;%</span> <span class="fu">filter</span>(pathway_type <span class="sc">==</span> <span class="st">&quot;yes&quot;</span>)</span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>  not_in_pathway <span class="ot">&lt;-</span> pathway_dt <span class="sc">%&gt;%</span> <span class="fu">filter</span>(pathway_type <span class="sc">==</span> <span class="st">&quot;no&quot;</span>)</span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>  counts <span class="ot">&lt;-</span> pathway_dt<span class="sc">$</span>genes[pathway_dt<span class="sc">$</span>isin <span class="sc">==</span> <span class="st">&quot;yes&quot;</span> <span class="sc">&amp;</span> </span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>                               pathway_dt<span class="sc">$</span>pathway_type <span class="sc">==</span> <span class="st">&quot;yes&quot;</span>] <span class="sc">%&gt;%</span> </span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">length</span>()</span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a>  fisher_dt <span class="ot">&lt;-</span>  <span class="fu">fisher.test</span>(<span class="fu">cbind</span>(<span class="fu">c</span>(<span class="fu">sum</span>(in_pathway<span class="sc">$</span>isin <span class="sc">==</span> <span class="st">&quot;yes&quot;</span>),</span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">sum</span>(in_pathway<span class="sc">$</span>isin <span class="sc">==</span> <span class="st">&quot;no&quot;</span>)),</span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">c</span>(<span class="fu">sum</span>(not_in_pathway<span class="sc">$</span>isin <span class="sc">==</span> <span class="st">&quot;yes&quot;</span>),</span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">sum</span>(not_in_pathway<span class="sc">$</span>isin <span class="sc">==</span> <span class="st">&quot;no&quot;</span>))),</span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a>                            <span class="at">alternative =</span> <span class="st">&quot;greater&quot;</span>)</span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> fisher_dt<span class="sc">$</span>p.value</span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">list</span>(p,counts)) </span>
<span id="cb24-20"><a href="#cb24-20" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb24-21"><a href="#cb24-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-22"><a href="#cb24-22" aria-hidden="true" tabindex="-1"></a>tcga_pre_cor <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/tcga_pre_cor_all_gene_rho.rds&quot;</span>)</span>
<span id="cb24-23"><a href="#cb24-23" aria-hidden="true" tabindex="-1"></a>all_genes_base <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">genes =</span> <span class="fu">rownames</span>(tcga_pre_cor),</span>
<span id="cb24-24"><a href="#cb24-24" aria-hidden="true" tabindex="-1"></a>                             <span class="at">split_gene =</span> <span class="fu">rownames</span>(tcga_pre_cor)) <span class="sc">%&gt;%</span> </span>
<span id="cb24-25"><a href="#cb24-25" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">separate_longer_delim</span>(<span class="at">cols =</span> <span class="st">&quot;split_gene&quot;</span>,<span class="at">delim =</span> <span class="st">&quot;,&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb24-26"><a href="#cb24-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb24-27"><a href="#cb24-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pathway =</span> <span class="fu">paste</span>(kegg<span class="sc">$</span>pathway[<span class="fu">grep</span>(split_gene,kegg<span class="sc">$</span>genes)],</span>
<span id="cb24-28"><a href="#cb24-28" aria-hidden="true" tabindex="-1"></a>                         <span class="at">collapse =</span> <span class="st">&quot;;&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb24-29"><a href="#cb24-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb24-30"><a href="#cb24-30" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">separate_longer_delim</span>(<span class="at">cols =</span> <span class="st">&quot;pathway&quot;</span>,<span class="at">delim =</span> <span class="st">&quot;;&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb24-31"><a href="#cb24-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(genes,pathway) <span class="sc">%&gt;%</span> <span class="fu">distinct_all</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb24-32"><a href="#cb24-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">nchar</span>(pathway) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb24-33"><a href="#cb24-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb24-34"><a href="#cb24-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">cor_counts =</span> <span class="fu">length</span>(<span class="fu">which</span>(tcga_pre_cor[,genes]<span class="sc">&gt;</span><span class="fl">0.8</span>)) <span class="sc">-</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb24-35"><a href="#cb24-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> <span class="fu">filter</span>(cor_counts <span class="sc">&gt;=</span> <span class="dv">3</span>)</span>
<span id="cb24-36"><a href="#cb24-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-37"><a href="#cb24-37" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(doParallel)</span>
<span id="cb24-38"><a href="#cb24-38" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(foreach)</span>
<span id="cb24-39"><a href="#cb24-39" aria-hidden="true" tabindex="-1"></a><span class="co">#create the cluster</span></span>
<span id="cb24-40"><a href="#cb24-40" aria-hidden="true" tabindex="-1"></a>my.cluster <span class="ot">&lt;-</span> parallel<span class="sc">::</span><span class="fu">makeCluster</span>(</span>
<span id="cb24-41"><a href="#cb24-41" aria-hidden="true" tabindex="-1"></a>  <span class="dv">60</span>, </span>
<span id="cb24-42"><a href="#cb24-42" aria-hidden="true" tabindex="-1"></a>  <span class="at">type =</span> <span class="st">&quot;PSOCK&quot;</span></span>
<span id="cb24-43"><a href="#cb24-43" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb24-44"><a href="#cb24-44" aria-hidden="true" tabindex="-1"></a><span class="co">#register it to be used by %dopar%</span></span>
<span id="cb24-45"><a href="#cb24-45" aria-hidden="true" tabindex="-1"></a>doParallel<span class="sc">::</span><span class="fu">registerDoParallel</span>(<span class="at">cl =</span> my.cluster)</span>
<span id="cb24-46"><a href="#cb24-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-47"><a href="#cb24-47" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">foreach</span>(</span>
<span id="cb24-48"><a href="#cb24-48" aria-hidden="true" tabindex="-1"></a>  <span class="at">i =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span>,</span>
<span id="cb24-49"><a href="#cb24-49" aria-hidden="true" tabindex="-1"></a>  <span class="at">.packages =</span> <span class="fu">c</span>(<span class="st">&quot;dplyr&quot;</span>,<span class="st">&quot;tidyr&quot;</span>)</span>
<span id="cb24-50"><a href="#cb24-50" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%dopar%</span> {</span>
<span id="cb24-51"><a href="#cb24-51" aria-hidden="true" tabindex="-1"></a>  all_genes_sample <span class="ot">&lt;-</span> all_genes_base <span class="sc">%&gt;%</span> </span>
<span id="cb24-52"><a href="#cb24-52" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb24-53"><a href="#cb24-53" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">p =</span> <span class="fu">kegg_enricher</span>(kegg,                    <span class="fu">sample</span>(<span class="fu">rownames</span>(tcga_pre_cor)[<span class="fu">which</span>(<span class="fu">rownames</span>(tcga_pre_cor) <span class="sc">!=</span> genes)],</span>
<span id="cb24-54"><a href="#cb24-54" aria-hidden="true" tabindex="-1"></a>                                    cor_counts,</span>
<span id="cb24-55"><a href="#cb24-55" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">replace=</span><span class="cn">FALSE</span>),</span>
<span id="cb24-56"><a href="#cb24-56" aria-hidden="true" tabindex="-1"></a>                             pathway)[[<span class="dv">1</span>]]) <span class="sc">%&gt;%</span> </span>
<span id="cb24-57"><a href="#cb24-57" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ungroup</span>()</span>
<span id="cb24-58"><a href="#cb24-58" aria-hidden="true" tabindex="-1"></a>  all_genes_sample<span class="sc">$</span>fdr <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(all_genes_sample<span class="sc">$</span>p,<span class="st">&quot;fdr&quot;</span>)</span>
<span id="cb24-59"><a href="#cb24-59" aria-hidden="true" tabindex="-1"></a>  all_genes_sample_summ <span class="ot">&lt;-</span> all_genes_sample <span class="sc">%&gt;%</span> </span>
<span id="cb24-60"><a href="#cb24-60" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(genes) <span class="sc">%&gt;%</span> </span>
<span id="cb24-61"><a href="#cb24-61" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">pathway_min =</span> pathway[<span class="fu">which.min</span>(p)],</span>
<span id="cb24-62"><a href="#cb24-62" aria-hidden="true" tabindex="-1"></a>              <span class="at">min_p =</span> p[<span class="fu">which.min</span>(p)],</span>
<span id="cb24-63"><a href="#cb24-63" aria-hidden="true" tabindex="-1"></a>              <span class="at">fdr_p =</span> fdr[<span class="fu">which.min</span>(p)]) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>()</span>
<span id="cb24-64"><a href="#cb24-64" aria-hidden="true" tabindex="-1"></a>  all_genes_sample_summ<span class="sc">$</span>sample_id <span class="ot">&lt;-</span> i</span>
<span id="cb24-65"><a href="#cb24-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(all_genes_sample_summ)</span>
<span id="cb24-66"><a href="#cb24-66" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb24-67"><a href="#cb24-67" aria-hidden="true" tabindex="-1"></a>parallel<span class="sc">::</span><span class="fu">stopCluster</span>(<span class="at">cl =</span> my.cluster)</span>
<span id="cb24-68"><a href="#cb24-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-69"><a href="#cb24-69" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(res)</span>
<span id="cb24-70"><a href="#cb24-70" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(res,<span class="at">file =</span> <span class="st">&quot;~/DeepMeta/data/pathway_sim_all_genes_rho.rds&quot;</span>)</span>
<span id="cb24-71"><a href="#cb24-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-72"><a href="#cb24-72" aria-hidden="true" tabindex="-1"></a><span class="do">###actual data</span></span>
<span id="cb24-73"><a href="#cb24-73" aria-hidden="true" tabindex="-1"></a>tcga_pre_cor <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/tcga_pre_cor_all_gene_rho.rds&quot;</span>)</span>
<span id="cb24-74"><a href="#cb24-74" aria-hidden="true" tabindex="-1"></a>all_genes_base <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">genes =</span> <span class="fu">rownames</span>(tcga_pre_cor),</span>
<span id="cb24-75"><a href="#cb24-75" aria-hidden="true" tabindex="-1"></a>                             <span class="at">split_gene =</span> <span class="fu">rownames</span>(tcga_pre_cor)) <span class="sc">%&gt;%</span> </span>
<span id="cb24-76"><a href="#cb24-76" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">separate_longer_delim</span>(<span class="at">cols =</span> <span class="st">&quot;split_gene&quot;</span>,<span class="at">delim =</span> <span class="st">&quot;,&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb24-77"><a href="#cb24-77" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb24-78"><a href="#cb24-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pathway =</span> <span class="fu">paste</span>(kegg<span class="sc">$</span>pathway[<span class="fu">grep</span>(split_gene,kegg<span class="sc">$</span>genes)],</span>
<span id="cb24-79"><a href="#cb24-79" aria-hidden="true" tabindex="-1"></a>                         <span class="at">collapse =</span> <span class="st">&quot;;&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb24-80"><a href="#cb24-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb24-81"><a href="#cb24-81" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">separate_longer_delim</span>(<span class="at">cols =</span> <span class="st">&quot;pathway&quot;</span>,<span class="at">delim =</span> <span class="st">&quot;;&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb24-82"><a href="#cb24-82" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(genes,pathway) <span class="sc">%&gt;%</span> <span class="fu">distinct_all</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb24-83"><a href="#cb24-83" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">nchar</span>(pathway) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb24-84"><a href="#cb24-84" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb24-85"><a href="#cb24-85" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">cor_counts =</span> <span class="fu">length</span>(<span class="fu">which</span>(tcga_pre_cor[,genes] <span class="sc">&gt;</span> <span class="fl">0.8</span>) <span class="sc">-</span> <span class="dv">1</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb24-86"><a href="#cb24-86" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> <span class="fu">filter</span>(cor_counts <span class="sc">&gt;=</span> <span class="dv">3</span>)</span>
<span id="cb24-87"><a href="#cb24-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-88"><a href="#cb24-88" aria-hidden="true" tabindex="-1"></a>all_genes <span class="ot">&lt;-</span> all_genes_base <span class="sc">%&gt;%</span> </span>
<span id="cb24-89"><a href="#cb24-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb24-90"><a href="#cb24-90" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">counts =</span> <span class="fu">kegg_enricher</span>(kegg,</span>
<span id="cb24-91"><a href="#cb24-91" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">rownames</span>(tcga_pre_cor)[<span class="fu">which</span>((tcga_pre_cor[,genes]<span class="sc">&gt;</span><span class="fl">0.8</span>) <span class="sc">&amp;</span> (<span class="fu">rownames</span>(tcga_pre_cor) <span class="sc">!=</span> genes))],</span>
<span id="cb24-92"><a href="#cb24-92" aria-hidden="true" tabindex="-1"></a>                                pathway)[[<span class="dv">2</span>]]) <span class="sc">%&gt;%</span> </span>
<span id="cb24-93"><a href="#cb24-93" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p =</span> <span class="fu">kegg_enricher</span>(kegg,</span>
<span id="cb24-94"><a href="#cb24-94" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">rownames</span>(tcga_pre_cor)[<span class="fu">which</span>((tcga_pre_cor[,genes]<span class="sc">&gt;</span><span class="fl">0.8</span>) <span class="sc">&amp;</span> (<span class="fu">rownames</span>(tcga_pre_cor) <span class="sc">!=</span> genes))],</span>
<span id="cb24-95"><a href="#cb24-95" aria-hidden="true" tabindex="-1"></a>                           pathway)[[<span class="dv">1</span>]]) <span class="sc">%&gt;%</span> </span>
<span id="cb24-96"><a href="#cb24-96" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb24-97"><a href="#cb24-97" aria-hidden="true" tabindex="-1"></a>all_genes<span class="sc">$</span>fdr <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(all_genes<span class="sc">$</span>p,<span class="st">&quot;fdr&quot;</span>)</span>
<span id="cb24-98"><a href="#cb24-98" aria-hidden="true" tabindex="-1"></a>all_genes_summ <span class="ot">&lt;-</span> all_genes <span class="sc">%&gt;%</span> </span>
<span id="cb24-99"><a href="#cb24-99" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(genes) <span class="sc">%&gt;%</span> </span>
<span id="cb24-100"><a href="#cb24-100" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">pathway_min =</span> pathway[<span class="fu">which.min</span>(p)],</span>
<span id="cb24-101"><a href="#cb24-101" aria-hidden="true" tabindex="-1"></a>            <span class="at">min_p =</span> p[<span class="fu">which.min</span>(p)],</span>
<span id="cb24-102"><a href="#cb24-102" aria-hidden="true" tabindex="-1"></a>            <span class="at">fdr_p =</span> fdr[<span class="fu">which.min</span>(p)],</span>
<span id="cb24-103"><a href="#cb24-103" aria-hidden="true" tabindex="-1"></a>            <span class="at">cor_counts =</span> <span class="fu">unique</span>(cor_counts),</span>
<span id="cb24-104"><a href="#cb24-104" aria-hidden="true" tabindex="-1"></a>            <span class="at">counts_p =</span> counts[<span class="fu">which.min</span>(p)]) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>()</span>
<span id="cb24-105"><a href="#cb24-105" aria-hidden="true" tabindex="-1"></a><span class="do">##</span></span>
<span id="cb24-106"><a href="#cb24-106" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(all_genes_summ,</span>
<span id="cb24-107"><a href="#cb24-107" aria-hidden="true" tabindex="-1"></a>        <span class="at">file =</span> <span class="st">&quot;~/DeepMeta/data/tcga_codepency_fisher_all_genes_rho.rds&quot;</span>)</span></code></pre></div>
<p>Among 31% of all genes, correlated genes were significantly enriched
(Fisher test P &lt; 0.05) in the metabolic pathway where the target gene
was located and the distribution of this proportion in 1000 permutations
is shown in <code>Fig 3b</code> (median proportion was 14%), resulting
the permutated P value less than 0.001, demonstrating the reliability of
the metabolic dependency analysis in TCGA samples.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/pathway_sim_all_genes_rho.rds&quot;</span>)</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>sampling_res <span class="ot">&lt;-</span> res <span class="sc">%&gt;%</span> </span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample_id) <span class="sc">%&gt;%</span> </span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">p_sig =</span> <span class="fu">mean</span>(min_p <span class="sc">&lt;</span> <span class="fl">0.05</span>),</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">fdr_sig =</span> <span class="fu">mean</span>(fdr_p <span class="sc">&lt;</span> <span class="fl">0.1</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>all_genes_summ <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/tcga_codepency_fisher_all_genes_rho.rds&quot;</span>)</span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a><span class="fu">mean</span>(all_genes_summ<span class="sc">$</span>min_p <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 0.3121547</span></span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> WVPlots<span class="sc">::</span><span class="fu">ShadedDensity</span>(<span class="at">frame =</span> sampling_res, </span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a>                            <span class="at">xvar =</span> <span class="st">&quot;fdr_sig&quot;</span>,</span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a>                            <span class="at">threshold =</span> <span class="fu">mean</span>(all_genes_summ<span class="sc">$</span>min_p <span class="sc">&lt;</span> <span class="fl">0.05</span>),</span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a>                            <span class="at">title =</span> <span class="st">&quot;&quot;</span>,</span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a>                            <span class="at">tail =</span> <span class="st">&quot;right&quot;</span>,<span class="at">linecolor=</span><span class="st">&quot;red&quot;</span>)</span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a>p<span class="sc">$</span>layers[[<span class="dv">1</span>]]<span class="sc">$</span>aes_params<span class="sc">$</span>size <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a>p<span class="sc">$</span>layers[[<span class="dv">2</span>]]<span class="sc">$</span>aes_params<span class="sc">$</span>fill <span class="ot">&lt;-</span> <span class="st">&quot;blue&quot;</span>     <span class="co">#geom_ribbon</span></span>
<span id="cb25-19"><a href="#cb25-19" aria-hidden="true" tabindex="-1"></a>p<span class="sc">$</span>layers[[<span class="dv">3</span>]]<span class="sc">$</span>aes_params<span class="sc">$</span>colour <span class="ot">&lt;-</span> <span class="st">&quot;black&quot;</span> </span>
<span id="cb25-20"><a href="#cb25-20" aria-hidden="true" tabindex="-1"></a>p<span class="sc">$</span>layers[[<span class="dv">3</span>]]<span class="sc">$</span>aes_params<span class="sc">$</span>size <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb25-21"><a href="#cb25-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-22"><a href="#cb25-22" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> p <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;Significant proportion in the simulation (P &lt; 0.05)&quot;</span>,</span>
<span id="cb25-23"><a href="#cb25-23" aria-hidden="true" tabindex="-1"></a>               <span class="at">y=</span><span class="st">&quot;Density&quot;</span>)<span class="sc">+</span></span>
<span id="cb25-24"><a href="#cb25-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_prism</span>()</span>
<span id="cb25-25"><a href="#cb25-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-26"><a href="#cb25-26" aria-hidden="true" tabindex="-1"></a>p2</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-16-1.png" width="2400" /></p>
<p>For each KEGG metabolic pathway, we showed the distribution of the
predicted dependency probability:</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/kegg_all_pathway.rds&quot;</span>)</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> kegg <span class="sc">%&gt;%</span> </span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">&quot;Metabolism&quot;</span>,class) <span class="sc">|</span> <span class="fu">grepl</span>(<span class="st">&quot;metabolism&quot;</span>,pathway)) <span class="sc">%&gt;%</span> </span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pathway =</span> <span class="fu">gsub</span>(<span class="st">&quot; </span><span class="sc">\\</span><span class="st">- Homo sapiens </span><span class="sc">\\</span><span class="st">(human</span><span class="sc">\\</span><span class="st">)&quot;</span>,<span class="st">&quot;&quot;</span>,pathway)) <span class="sc">%&gt;%</span> </span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;Drug&quot;</span>,pathway))</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> kegg <span class="sc">%&gt;%</span> </span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pathway =</span> <span class="fu">case_when</span>(</span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a>    pathway <span class="sc">==</span> <span class="st">&quot;Citrate cycle (TCA cycle)&quot;</span> <span class="sc">~</span> <span class="st">&quot;Citrate cycle&quot;</span>,</span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>    pathway <span class="sc">==</span> <span class="st">&quot;Glycosylphosphatidylinositol (GPI)-anchor biosynthesis&quot;</span> <span class="sc">~</span> <span class="st">&quot;Glycosylphosphatidylinositol&quot;</span>,</span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> pathway</span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a>  ))</span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a>get_pathway <span class="ot">&lt;-</span> <span class="cf">function</span>(gene,pathway){</span>
<span id="cb26-16"><a href="#cb26-16" aria-hidden="true" tabindex="-1"></a>  gene <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(gene,<span class="at">split =</span> <span class="st">&quot;,&quot;</span>)[[<span class="dv">1</span>]]</span>
<span id="cb26-17"><a href="#cb26-17" aria-hidden="true" tabindex="-1"></a>  pathway <span class="ot">&lt;-</span> pathway <span class="sc">%&gt;%</span> <span class="fu">filter</span>(genes <span class="sc">%in%</span> gene)</span>
<span id="cb26-18"><a href="#cb26-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">c</span>(<span class="fu">paste</span>(<span class="fu">unique</span>(pathway<span class="sc">$</span>pathway),<span class="at">collapse =</span> <span class="st">&quot;;&quot;</span>),</span>
<span id="cb26-19"><a href="#cb26-19" aria-hidden="true" tabindex="-1"></a>           <span class="fu">paste</span>(<span class="fu">unique</span>(pathway<span class="sc">$</span>class),<span class="at">collapse =</span> <span class="st">&quot;;&quot;</span>)))</span>
<span id="cb26-20"><a href="#cb26-20" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb26-21"><a href="#cb26-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-22"><a href="#cb26-22" aria-hidden="true" tabindex="-1"></a>pre <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/tcga_pre_V2.rds&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb26-23"><a href="#cb26-23" aria-hidden="true" tabindex="-1"></a>pre_pathway <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene =</span> <span class="fu">unique</span>(<span class="fu">c</span>(pre<span class="sc">$</span>gene))) <span class="sc">%&gt;%</span> </span>
<span id="cb26-24"><a href="#cb26-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb26-25"><a href="#cb26-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pathway =</span> <span class="fu">get_pathway</span>(gene,kegg)[<span class="dv">1</span>],</span>
<span id="cb26-26"><a href="#cb26-26" aria-hidden="true" tabindex="-1"></a>         <span class="at">class =</span> <span class="fu">get_pathway</span>(gene,kegg)[<span class="dv">2</span>]) <span class="sc">%&gt;%</span> </span>
<span id="cb26-27"><a href="#cb26-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb26-28"><a href="#cb26-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">nchar</span>(pathway)<span class="sc">&gt;</span><span class="dv">1</span>)</span>
<span id="cb26-29"><a href="#cb26-29" aria-hidden="true" tabindex="-1"></a>pre_pathway <span class="ot">&lt;-</span> pre <span class="sc">%&gt;%</span> </span>
<span id="cb26-30"><a href="#cb26-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(.,pre_pathway) <span class="sc">%&gt;%</span> <span class="fu">na.omit</span>()</span>
<span id="cb26-31"><a href="#cb26-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(gene)`</span></span>
<span id="cb26-32"><a href="#cb26-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-33"><a href="#cb26-33" aria-hidden="true" tabindex="-1"></a>all_split <span class="ot">&lt;-</span> pre_pathway <span class="sc">%&gt;%</span> </span>
<span id="cb26-34"><a href="#cb26-34" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">separate_longer_delim</span>(<span class="at">cols =</span> <span class="st">&quot;pathway&quot;</span>,<span class="at">delim =</span> <span class="st">&quot;;&quot;</span>)</span>
<span id="cb26-35"><a href="#cb26-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-36"><a href="#cb26-36" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggridges)</span>
<span id="cb26-37"><a href="#cb26-37" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(viridis)</span>
<span id="cb26-38"><a href="#cb26-38" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: viridisLite</span></span>
<span id="cb26-39"><a href="#cb26-39" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(hrbrthemes)</span>
<span id="cb26-40"><a href="#cb26-40" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span><span class="al">NOTE</span><span class="co">: Either Arial Narrow or Roboto Condensed fonts are required to use these themes.</span></span>
<span id="cb26-41"><a href="#cb26-41" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;       Please use hrbrthemes::import_roboto_condensed() to install Roboto Condensed and</span></span>
<span id="cb26-42"><a href="#cb26-42" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;       if Arial Narrow is not on your system, please see https://bit.ly/arialnarrow</span></span>
<span id="cb26-43"><a href="#cb26-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-44"><a href="#cb26-44" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb26-45"><a href="#cb26-45" aria-hidden="true" tabindex="-1"></a>all_split <span class="ot">&lt;-</span> all_split <span class="sc">%&gt;%</span> </span>
<span id="cb26-46"><a href="#cb26-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pathway2 =</span> stringr<span class="sc">::</span><span class="fu">str_replace</span>(pathway,<span class="st">&quot; / &quot;</span>,<span class="st">&quot;-&quot;</span>))</span>
<span id="cb26-47"><a href="#cb26-47" aria-hidden="true" tabindex="-1"></a>gene_summ <span class="ot">&lt;-</span> all_split <span class="sc">%&gt;%</span> </span>
<span id="cb26-48"><a href="#cb26-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(pathway2) <span class="sc">%&gt;%</span> </span>
<span id="cb26-49"><a href="#cb26-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">gene_counts =</span> <span class="fu">length</span>(<span class="fu">unique</span>(gene))) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb26-50"><a href="#cb26-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(gene_counts <span class="sc">&gt;=</span><span class="dv">10</span>)</span>
<span id="cb26-51"><a href="#cb26-51" aria-hidden="true" tabindex="-1"></a>all_split <span class="ot">&lt;-</span> all_split <span class="sc">%&gt;%</span> </span>
<span id="cb26-52"><a href="#cb26-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(pathway2 <span class="sc">%in%</span> gene_summ<span class="sc">$</span>pathway2) <span class="sc">%&gt;%</span> </span>
<span id="cb26-53"><a href="#cb26-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(.,gene_summ) <span class="sc">%&gt;%</span> </span>
<span id="cb26-54"><a href="#cb26-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pathway3 =</span> <span class="fu">paste0</span>(pathway2,<span class="st">&quot; (n=&quot;</span>,gene_counts,<span class="st">&quot;)&quot;</span>))</span>
<span id="cb26-55"><a href="#cb26-55" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(pathway2)`</span></span>
<span id="cb26-56"><a href="#cb26-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-57"><a href="#cb26-57" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(all_split, <span class="fu">aes</span>(<span class="at">x =</span> preds_raw, <span class="at">y =</span> pathway3, <span class="at">fill =</span> <span class="fu">stat</span>(x))) <span class="sc">+</span></span>
<span id="cb26-58"><a href="#cb26-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density_ridges_gradient</span>(<span class="at">scale =</span> <span class="dv">1</span>, <span class="at">rel_min_height =</span> <span class="fl">0.01</span>) <span class="sc">+</span></span>
<span id="cb26-59"><a href="#cb26-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_viridis</span>(<span class="at">name =</span> <span class="st">&quot;Preds&quot;</span>, <span class="at">option =</span> <span class="st">&quot;C&quot;</span>) <span class="sc">+</span></span>
<span id="cb26-60"><a href="#cb26-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb26-61"><a href="#cb26-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb26-62"><a href="#cb26-62" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position=</span><span class="st">&quot;none&quot;</span>,</span>
<span id="cb26-63"><a href="#cb26-63" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.spacing =</span> <span class="fu">unit</span>(<span class="fl">0.1</span>, <span class="st">&quot;lines&quot;</span>),</span>
<span id="cb26-64"><a href="#cb26-64" aria-hidden="true" tabindex="-1"></a>    <span class="at">strip.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>)</span>
<span id="cb26-65"><a href="#cb26-65" aria-hidden="true" tabindex="-1"></a>  )<span class="sc">+</span><span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;Prediction&quot;</span>,<span class="at">y=</span><span class="st">&quot;Pathways&quot;</span>)</span>
<span id="cb26-66"><a href="#cb26-66" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: `stat(x)` was deprecated in ggplot2 3.4.0.</span></span>
<span id="cb26-67"><a href="#cb26-67" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ℹ Please use `after_stat(x)` instead.</span></span>
<span id="cb26-68"><a href="#cb26-68" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; This warning is displayed once every 8 hours.</span></span>
<span id="cb26-69"><a href="#cb26-69" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span>
<span id="cb26-70"><a href="#cb26-70" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; generated.</span></span>
<span id="cb26-71"><a href="#cb26-71" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Picking joint bandwidth of 0.0163</span></span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-17-1.png" width="2400" /></p>
<p>Then we performed enrichment analysis of metabolic pathways to test
whether the genes predicted to be positive in one sample are more
enriched in the specific metabolic pathway, relative to all other
pathways (see Methods). Furthermore, cancer specific metabolic
dependencies should show differences between tumor and normal samples.
Therefore, we applied DeepMeta to TCGA normal tissues and performed the
same enrichment analysis:</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>get_pathway_p <span class="ot">&lt;-</span> <span class="cf">function</span>(pathway_dt,pathway_name,sample_name){</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>  pdt <span class="ot">&lt;-</span> pathway_dt <span class="sc">%&gt;%</span> </span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="fu">grepl</span>(pathway_name,pathway) <span class="sc">&amp;</span> cell <span class="sc">==</span> sample_name)</span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>  npdt <span class="ot">&lt;-</span> pathway_dt <span class="sc">%&gt;%</span> </span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">grepl</span>(pathway_name,pathway) <span class="sc">&amp;</span> cell <span class="sc">==</span> sample_name)</span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">fisher.test</span>(<span class="fu">cbind</span>(</span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="fu">sum</span>(pdt<span class="sc">$</span>preds <span class="sc">==</span> <span class="dv">1</span>),<span class="fu">sum</span>(pdt<span class="sc">$</span>preds <span class="sc">!=</span> <span class="dv">1</span>)),</span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="fu">sum</span>(npdt<span class="sc">$</span>preds <span class="sc">==</span> <span class="dv">1</span>),<span class="fu">sum</span>(npdt<span class="sc">$</span>preds <span class="sc">!=</span> <span class="dv">1</span>))</span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a>  ),<span class="at">alternative =</span> <span class="st">&quot;greater&quot;</span>)</span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">c</span>(res<span class="sc">$</span>p.value,res<span class="sc">$</span>estimate[[<span class="dv">1</span>]]))</span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a>all_samples <span class="ot">&lt;-</span> <span class="fu">unique</span>(pre_pathway<span class="sc">$</span>cell)</span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-15"><a href="#cb27-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(doParallel)</span>
<span id="cb27-16"><a href="#cb27-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(foreach)</span>
<span id="cb27-17"><a href="#cb27-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-18"><a href="#cb27-18" aria-hidden="true" tabindex="-1"></a>all_split <span class="ot">&lt;-</span> pre_pathway <span class="sc">%&gt;%</span> </span>
<span id="cb27-19"><a href="#cb27-19" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">separate_longer_delim</span>(<span class="at">cols =</span> <span class="st">&quot;pathway&quot;</span>,<span class="at">delim =</span> <span class="st">&quot;;&quot;</span>)</span>
<span id="cb27-20"><a href="#cb27-20" aria-hidden="true" tabindex="-1"></a>all_pathways <span class="ot">&lt;-</span> <span class="fu">unique</span>(all_split<span class="sc">$</span>pathway)</span>
<span id="cb27-21"><a href="#cb27-21" aria-hidden="true" tabindex="-1"></a>all_res <span class="ot">&lt;-</span> <span class="fu">vector</span>(<span class="st">&quot;list&quot;</span>,<span class="dv">84</span>)</span>
<span id="cb27-22"><a href="#cb27-22" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(all_pathways)){</span>
<span id="cb27-23"><a href="#cb27-23" aria-hidden="true" tabindex="-1"></a>  tmp <span class="ot">&lt;-</span> all_pathways[i]</span>
<span id="cb27-24"><a href="#cb27-24" aria-hidden="true" tabindex="-1"></a>  my.cluster <span class="ot">&lt;-</span> parallel<span class="sc">::</span><span class="fu">makeCluster</span>(</span>
<span id="cb27-25"><a href="#cb27-25" aria-hidden="true" tabindex="-1"></a>    <span class="dv">60</span>, </span>
<span id="cb27-26"><a href="#cb27-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">type =</span> <span class="st">&quot;PSOCK&quot;</span></span>
<span id="cb27-27"><a href="#cb27-27" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb27-28"><a href="#cb27-28" aria-hidden="true" tabindex="-1"></a>  <span class="co">#register it to be used by %dopar%</span></span>
<span id="cb27-29"><a href="#cb27-29" aria-hidden="true" tabindex="-1"></a>  doParallel<span class="sc">::</span><span class="fu">registerDoParallel</span>(<span class="at">cl =</span> my.cluster)</span>
<span id="cb27-30"><a href="#cb27-30" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb27-31"><a href="#cb27-31" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">foreach</span>(</span>
<span id="cb27-32"><a href="#cb27-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">i =</span> all_samples,</span>
<span id="cb27-33"><a href="#cb27-33" aria-hidden="true" tabindex="-1"></a>    <span class="at">.export =</span> <span class="fu">c</span>(<span class="st">&quot;pre_pathway&quot;</span>,<span class="st">&quot;get_pathway_p&quot;</span>,<span class="st">&quot;tmp&quot;</span>),</span>
<span id="cb27-34"><a href="#cb27-34" aria-hidden="true" tabindex="-1"></a>    <span class="at">.packages =</span> <span class="fu">c</span>(<span class="st">&quot;dplyr&quot;</span>)</span>
<span id="cb27-35"><a href="#cb27-35" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%dopar%</span> {</span>
<span id="cb27-36"><a href="#cb27-36" aria-hidden="true" tabindex="-1"></a>    p_o <span class="ot">&lt;-</span> <span class="fu">get_pathway_p</span>(pre_pathway,tmp,i)</span>
<span id="cb27-37"><a href="#cb27-37" aria-hidden="true" tabindex="-1"></a>    dt <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">p_value =</span> p_o[<span class="dv">1</span>],</span>
<span id="cb27-38"><a href="#cb27-38" aria-hidden="true" tabindex="-1"></a>                     <span class="at">ratio =</span> p_o[<span class="dv">2</span>],</span>
<span id="cb27-39"><a href="#cb27-39" aria-hidden="true" tabindex="-1"></a>                     <span class="at">sample =</span> i,</span>
<span id="cb27-40"><a href="#cb27-40" aria-hidden="true" tabindex="-1"></a>                     <span class="at">pathway =</span> tmp)</span>
<span id="cb27-41"><a href="#cb27-41" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(dt)</span>
<span id="cb27-42"><a href="#cb27-42" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb27-43"><a href="#cb27-43" aria-hidden="true" tabindex="-1"></a>  parallel<span class="sc">::</span><span class="fu">stopCluster</span>(<span class="at">cl =</span> my.cluster)</span>
<span id="cb27-44"><a href="#cb27-44" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(res)</span>
<span id="cb27-45"><a href="#cb27-45" aria-hidden="true" tabindex="-1"></a>  all_res[[i]] <span class="ot">&lt;-</span> res</span>
<span id="cb27-46"><a href="#cb27-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">&quot;Complete &quot;</span>,i)</span>
<span id="cb27-47"><a href="#cb27-47" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb27-48"><a href="#cb27-48" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(all_res)</span>
<span id="cb27-49"><a href="#cb27-49" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(res,<span class="at">file =</span> <span class="st">&quot;~/DeepMeta/data/pancancer_meta_dep.rds&quot;</span>)</span>
<span id="cb27-50"><a href="#cb27-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-51"><a href="#cb27-51" aria-hidden="true" tabindex="-1"></a><span class="do">##normal</span></span>
<span id="cb27-52"><a href="#cb27-52" aria-hidden="true" tabindex="-1"></a>pre <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/tcga_normal_pre_V2.rds&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-53"><a href="#cb27-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb27-54"><a href="#cb27-54" aria-hidden="true" tabindex="-1"></a>pre_pathway <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene =</span> <span class="fu">unique</span>(<span class="fu">c</span>(pre<span class="sc">$</span>gene))) <span class="sc">%&gt;%</span> </span>
<span id="cb27-55"><a href="#cb27-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb27-56"><a href="#cb27-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pathway =</span> <span class="fu">get_pathway</span>(gene,kegg)[<span class="dv">1</span>],</span>
<span id="cb27-57"><a href="#cb27-57" aria-hidden="true" tabindex="-1"></a>         <span class="at">class =</span> <span class="fu">get_pathway</span>(gene,kegg)[<span class="dv">2</span>]) <span class="sc">%&gt;%</span> </span>
<span id="cb27-58"><a href="#cb27-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb27-59"><a href="#cb27-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">nchar</span>(pathway)<span class="sc">&gt;</span><span class="dv">1</span>)</span>
<span id="cb27-60"><a href="#cb27-60" aria-hidden="true" tabindex="-1"></a>pre_pathway <span class="ot">&lt;-</span> pre <span class="sc">%&gt;%</span> </span>
<span id="cb27-61"><a href="#cb27-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(.,pre_pathway) <span class="sc">%&gt;%</span> <span class="fu">na.omit</span>()</span>
<span id="cb27-62"><a href="#cb27-62" aria-hidden="true" tabindex="-1"></a>all_samples <span class="ot">&lt;-</span> <span class="fu">unique</span>(pre_pathway<span class="sc">$</span>cell)</span>
<span id="cb27-63"><a href="#cb27-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-64"><a href="#cb27-64" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(doParallel)</span>
<span id="cb27-65"><a href="#cb27-65" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(foreach)</span>
<span id="cb27-66"><a href="#cb27-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-67"><a href="#cb27-67" aria-hidden="true" tabindex="-1"></a>all_split <span class="ot">&lt;-</span> pre_pathway <span class="sc">%&gt;%</span> </span>
<span id="cb27-68"><a href="#cb27-68" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">separate_longer_delim</span>(<span class="at">cols =</span> <span class="st">&quot;pathway&quot;</span>,<span class="at">delim =</span> <span class="st">&quot;;&quot;</span>)</span>
<span id="cb27-69"><a href="#cb27-69" aria-hidden="true" tabindex="-1"></a>all_pathways <span class="ot">&lt;-</span> <span class="fu">unique</span>(all_split<span class="sc">$</span>pathway)</span>
<span id="cb27-70"><a href="#cb27-70" aria-hidden="true" tabindex="-1"></a>all_res <span class="ot">&lt;-</span> <span class="fu">vector</span>(<span class="st">&quot;list&quot;</span>,<span class="dv">82</span>)</span>
<span id="cb27-71"><a href="#cb27-71" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(all_pathways)){</span>
<span id="cb27-72"><a href="#cb27-72" aria-hidden="true" tabindex="-1"></a>  tmp <span class="ot">&lt;-</span> all_pathways[i]</span>
<span id="cb27-73"><a href="#cb27-73" aria-hidden="true" tabindex="-1"></a>  my.cluster <span class="ot">&lt;-</span> parallel<span class="sc">::</span><span class="fu">makeCluster</span>(</span>
<span id="cb27-74"><a href="#cb27-74" aria-hidden="true" tabindex="-1"></a>    <span class="dv">60</span>, </span>
<span id="cb27-75"><a href="#cb27-75" aria-hidden="true" tabindex="-1"></a>    <span class="at">type =</span> <span class="st">&quot;PSOCK&quot;</span></span>
<span id="cb27-76"><a href="#cb27-76" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb27-77"><a href="#cb27-77" aria-hidden="true" tabindex="-1"></a>  <span class="co">#register it to be used by %dopar%</span></span>
<span id="cb27-78"><a href="#cb27-78" aria-hidden="true" tabindex="-1"></a>  doParallel<span class="sc">::</span><span class="fu">registerDoParallel</span>(<span class="at">cl =</span> my.cluster)</span>
<span id="cb27-79"><a href="#cb27-79" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb27-80"><a href="#cb27-80" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">foreach</span>(</span>
<span id="cb27-81"><a href="#cb27-81" aria-hidden="true" tabindex="-1"></a>    <span class="at">i =</span> all_samples,</span>
<span id="cb27-82"><a href="#cb27-82" aria-hidden="true" tabindex="-1"></a>    <span class="at">.export =</span> <span class="fu">c</span>(<span class="st">&quot;pre_pathway&quot;</span>,<span class="st">&quot;get_pathway_p&quot;</span>,<span class="st">&quot;tmp&quot;</span>),</span>
<span id="cb27-83"><a href="#cb27-83" aria-hidden="true" tabindex="-1"></a>    <span class="at">.packages =</span> <span class="fu">c</span>(<span class="st">&quot;dplyr&quot;</span>)</span>
<span id="cb27-84"><a href="#cb27-84" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%dopar%</span> {</span>
<span id="cb27-85"><a href="#cb27-85" aria-hidden="true" tabindex="-1"></a>    p_o <span class="ot">&lt;-</span> <span class="fu">get_pathway_p</span>(pre_pathway,tmp,i)</span>
<span id="cb27-86"><a href="#cb27-86" aria-hidden="true" tabindex="-1"></a>    dt <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">p_value =</span> p_o[<span class="dv">1</span>],</span>
<span id="cb27-87"><a href="#cb27-87" aria-hidden="true" tabindex="-1"></a>                     <span class="at">ratio =</span> p_o[<span class="dv">2</span>],</span>
<span id="cb27-88"><a href="#cb27-88" aria-hidden="true" tabindex="-1"></a>                     <span class="at">sample =</span> i,</span>
<span id="cb27-89"><a href="#cb27-89" aria-hidden="true" tabindex="-1"></a>                     <span class="at">pathway =</span> tmp)</span>
<span id="cb27-90"><a href="#cb27-90" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(dt)</span>
<span id="cb27-91"><a href="#cb27-91" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb27-92"><a href="#cb27-92" aria-hidden="true" tabindex="-1"></a>  parallel<span class="sc">::</span><span class="fu">stopCluster</span>(<span class="at">cl =</span> my.cluster)</span>
<span id="cb27-93"><a href="#cb27-93" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(res)</span>
<span id="cb27-94"><a href="#cb27-94" aria-hidden="true" tabindex="-1"></a>  all_res[[i]] <span class="ot">&lt;-</span> res</span>
<span id="cb27-95"><a href="#cb27-95" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">&quot;Complete &quot;</span>,i)</span>
<span id="cb27-96"><a href="#cb27-96" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb27-97"><a href="#cb27-97" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(all_res)</span>
<span id="cb27-98"><a href="#cb27-98" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(res,<span class="at">file =</span> <span class="st">&quot;~/DeepMeta/data/pancancer_meta_dep_normal.rds&quot;</span>)</span></code></pre></div>
<p>we found that nucleotide metabolism (including purine metabolism and
pyrimidine metabolism) and glutathione metabolism showed most
significant difference between cancer and normal, both in pan-cancer
(<code>Fig 3c</code>) and individual cancer types:</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>cancer_dep <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/pancancer_meta_dep.rds&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.infinite</span>(ratio)) <span class="sc">%&gt;%</span> </span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">log2ratio=</span><span class="fu">log2</span>(ratio<span class="sc">+</span><span class="dv">1</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">type =</span> <span class="st">&quot;cancer&quot;</span>)</span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>normal_dep <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/pancancer_meta_dep_normal.rds&quot;</span>)<span class="sc">%&gt;%</span></span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.infinite</span>(ratio)) <span class="sc">%&gt;%</span> </span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">log2ratio=</span><span class="fu">log2</span>(ratio<span class="sc">+</span><span class="dv">1</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">type =</span> <span class="st">&quot;normal&quot;</span>)</span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a>both <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(normal_dep,cancer_dep)</span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a>both<span class="sc">$</span>cancer_type <span class="ot">&lt;-</span> EasyBioinfo<span class="sc">::</span><span class="fu">get_cancer_type</span>(both<span class="sc">$</span>sample,</span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a>                                                 <span class="at">cores =</span> <span class="dv">50</span>, </span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a>                                                 <span class="at">parallel =</span> T)</span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; you have 128 cores</span></span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a><span class="do">##全局比较</span></span>
<span id="cb28-16"><a href="#cb28-16" aria-hidden="true" tabindex="-1"></a>cancer_summ <span class="ot">&lt;-</span> both <span class="sc">%&gt;%</span> </span>
<span id="cb28-17"><a href="#cb28-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(pathway,cancer_type) <span class="sc">%&gt;%</span> </span>
<span id="cb28-18"><a href="#cb28-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">counts =</span> <span class="fu">length</span>(<span class="fu">unique</span>(type))) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb28-19"><a href="#cb28-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(counts <span class="sc">&gt;</span> <span class="dv">1</span>)</span>
<span id="cb28-20"><a href="#cb28-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; `summarise()` has grouped output by &#39;pathway&#39;. You can override using the</span></span>
<span id="cb28-21"><a href="#cb28-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; `.groups` argument.</span></span>
<span id="cb28-22"><a href="#cb28-22" aria-hidden="true" tabindex="-1"></a>cancer_summ2 <span class="ot">&lt;-</span> both <span class="sc">%&gt;%</span> </span>
<span id="cb28-23"><a href="#cb28-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(cancer_type,type) <span class="sc">%&gt;%</span> </span>
<span id="cb28-24"><a href="#cb28-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">counts =</span> <span class="fu">length</span>(<span class="fu">unique</span>(sample))) <span class="sc">%&gt;%</span> </span>
<span id="cb28-25"><a href="#cb28-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb28-26"><a href="#cb28-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(type <span class="sc">==</span> <span class="st">&quot;normal&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb28-27"><a href="#cb28-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(counts <span class="sc">&gt;</span> <span class="dv">10</span>)</span>
<span id="cb28-28"><a href="#cb28-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; `summarise()` has grouped output by &#39;cancer_type&#39;. You can override using the</span></span>
<span id="cb28-29"><a href="#cb28-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; `.groups` argument.</span></span>
<span id="cb28-30"><a href="#cb28-30" aria-hidden="true" tabindex="-1"></a>need_cancer <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">unique</span>(cancer_summ<span class="sc">$</span>cancer_type),</span>
<span id="cb28-31"><a href="#cb28-31" aria-hidden="true" tabindex="-1"></a>                         <span class="fu">unique</span>(cancer_summ2<span class="sc">$</span>cancer_type))</span>
<span id="cb28-32"><a href="#cb28-32" aria-hidden="true" tabindex="-1"></a>both_summ <span class="ot">&lt;-</span> both <span class="sc">%&gt;%</span> </span>
<span id="cb28-33"><a href="#cb28-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(cancer_type <span class="sc">%in%</span> need_cancer) <span class="sc">%&gt;%</span> </span>
<span id="cb28-34"><a href="#cb28-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(pathway, type) <span class="sc">%&gt;%</span> </span>
<span id="cb28-35"><a href="#cb28-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">median_or =</span> <span class="fu">median</span>(log2ratio)) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb28-36"><a href="#cb28-36" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">names_from =</span> <span class="st">&quot;type&quot;</span>,<span class="at">values_from =</span> <span class="st">&quot;median_or&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb28-37"><a href="#cb28-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(cancer <span class="sc">&gt;</span> normal)</span>
<span id="cb28-38"><a href="#cb28-38" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; `summarise()` has grouped output by &#39;pathway&#39;. You can override using the</span></span>
<span id="cb28-39"><a href="#cb28-39" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; `.groups` argument.</span></span>
<span id="cb28-40"><a href="#cb28-40" aria-hidden="true" tabindex="-1"></a>dt <span class="ot">&lt;-</span> both <span class="sc">%&gt;%</span> </span>
<span id="cb28-41"><a href="#cb28-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(cancer_type <span class="sc">%in%</span> need_cancer) <span class="sc">%&gt;%</span> </span>
<span id="cb28-42"><a href="#cb28-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(pathway <span class="sc">%in%</span> both_summ<span class="sc">$</span>pathway) <span class="sc">%&gt;%</span> </span>
<span id="cb28-43"><a href="#cb28-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Type =</span> type)</span>
<span id="cb28-44"><a href="#cb28-44" aria-hidden="true" tabindex="-1"></a>dt<span class="sc">$</span>Type <span class="ot">&lt;-</span> <span class="fu">factor</span>(dt<span class="sc">$</span>Type,<span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;normal&quot;</span>,<span class="st">&quot;cancer&quot;</span>))</span>
<span id="cb28-45"><a href="#cb28-45" aria-hidden="true" tabindex="-1"></a><span class="do">###按照 P 值进行排序</span></span>
<span id="cb28-46"><a href="#cb28-46" aria-hidden="true" tabindex="-1"></a>dt_p <span class="ot">&lt;-</span> dt <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(pathway) <span class="sc">%&gt;%</span> </span>
<span id="cb28-47"><a href="#cb28-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">do</span>(<span class="at">w =</span> <span class="fu">wilcox.test</span>(log2ratio <span class="sc">~</span> Type, <span class="at">data=</span>., <span class="at">paired=</span><span class="cn">FALSE</span>, </span>
<span id="cb28-48"><a href="#cb28-48" aria-hidden="true" tabindex="-1"></a>                     <span class="at">alternative =</span> <span class="st">&quot;less&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb28-49"><a href="#cb28-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(pathway, <span class="at">Wilcox =</span> w<span class="sc">$</span>p.value) <span class="sc">%&gt;%</span> </span>
<span id="cb28-50"><a href="#cb28-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Wilcox)</span>
<span id="cb28-51"><a href="#cb28-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-52"><a href="#cb28-52" aria-hidden="true" tabindex="-1"></a><span class="fu">ggboxplot</span>(<span class="at">data =</span> dt, <span class="at">x=</span><span class="st">&quot;pathway&quot;</span>, <span class="at">y=</span> <span class="st">&quot;log2ratio&quot;</span>, <span class="at">fill=</span> <span class="st">&quot;Type&quot;</span>,</span>
<span id="cb28-53"><a href="#cb28-53" aria-hidden="true" tabindex="-1"></a>          <span class="at">ylab =</span> <span class="st">&quot;Log2OR&quot;</span>,<span class="at">xlab =</span> <span class="cn">FALSE</span>,</span>
<span id="cb28-54"><a href="#cb28-54" aria-hidden="true" tabindex="-1"></a>          <span class="at">order =</span> dt_p<span class="sc">$</span>pathway,</span>
<span id="cb28-55"><a href="#cb28-55" aria-hidden="true" tabindex="-1"></a>          <span class="at">palette =</span> <span class="fu">c</span>(<span class="st">&quot;#81E9B8&quot;</span>,<span class="st">&quot;#FE7D81&quot;</span>))<span class="sc">+</span></span>
<span id="cb28-56"><a href="#cb28-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(<span class="fu">aes</span>(<span class="at">group =</span> Type,</span>
<span id="cb28-57"><a href="#cb28-57" aria-hidden="true" tabindex="-1"></a>                         <span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">&quot;p = &quot;</span>, <span class="fu">after_stat</span>(p.format))),</span>
<span id="cb28-58"><a href="#cb28-58" aria-hidden="true" tabindex="-1"></a>                     <span class="at">method =</span> <span class="st">&quot;wilcox&quot;</span>,</span>
<span id="cb28-59"><a href="#cb28-59" aria-hidden="true" tabindex="-1"></a>                     <span class="at">method.args =</span> <span class="fu">list</span>(<span class="at">alternative =</span> <span class="st">&quot;greater&quot;</span>) )<span class="sc">+</span></span>
<span id="cb28-60"><a href="#cb28-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rotate_x_text</span>(<span class="dv">45</span>)</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-19-1.png" width="2400" /></p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a><span class="do">###cancer type</span></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>both_summ <span class="ot">&lt;-</span> both <span class="sc">%&gt;%</span> </span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(pathway, cancer_type, type) <span class="sc">%&gt;%</span> </span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">median_or =</span> <span class="fu">median</span>(log2ratio)) <span class="sc">%&gt;%</span> </span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; `summarise()` has grouped output by &#39;pathway&#39;, &#39;cancer_type&#39;. You can override</span></span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; using the `.groups` argument.</span></span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a>cancer_summ <span class="ot">&lt;-</span> both <span class="sc">%&gt;%</span> </span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(pathway,cancer_type) <span class="sc">%&gt;%</span> </span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">counts =</span> <span class="fu">length</span>(<span class="fu">unique</span>(type))) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(counts <span class="sc">&gt;</span> <span class="dv">1</span>)</span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; `summarise()` has grouped output by &#39;pathway&#39;. You can override using the</span></span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; `.groups` argument.</span></span>
<span id="cb29-15"><a href="#cb29-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-16"><a href="#cb29-16" aria-hidden="true" tabindex="-1"></a>both_summ <span class="ot">&lt;-</span> both_summ <span class="sc">%&gt;%</span> </span>
<span id="cb29-17"><a href="#cb29-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(cancer_type <span class="sc">%in%</span> cancer_summ<span class="sc">$</span>cancer_type) <span class="sc">%&gt;%</span> </span>
<span id="cb29-18"><a href="#cb29-18" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">names_from =</span> <span class="st">&quot;type&quot;</span>,<span class="at">values_from =</span> <span class="st">&quot;median_or&quot;</span>)</span>
<span id="cb29-19"><a href="#cb29-19" aria-hidden="true" tabindex="-1"></a>both_summ <span class="ot">&lt;-</span> both_summ <span class="sc">%&gt;%</span> </span>
<span id="cb29-20"><a href="#cb29-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">diff_or =</span> cancer <span class="sc">-</span> normal) </span>
<span id="cb29-21"><a href="#cb29-21" aria-hidden="true" tabindex="-1"></a>both <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(cancer_type,type) <span class="sc">%&gt;%</span> </span>
<span id="cb29-22"><a href="#cb29-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">counts =</span> <span class="fu">length</span>(<span class="fu">unique</span>(sample))) <span class="sc">%&gt;%</span> </span>
<span id="cb29-23"><a href="#cb29-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(type <span class="sc">==</span> <span class="st">&quot;normal&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb29-24"><a href="#cb29-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(counts <span class="sc">&gt;</span> <span class="dv">10</span>) <span class="ot">-&gt;</span> tt</span>
<span id="cb29-25"><a href="#cb29-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; `summarise()` has grouped output by &#39;cancer_type&#39;. You can override using the</span></span>
<span id="cb29-26"><a href="#cb29-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; `.groups` argument.</span></span>
<span id="cb29-27"><a href="#cb29-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-28"><a href="#cb29-28" aria-hidden="true" tabindex="-1"></a><span class="do">###只选择正常样本数量大于 10 </span></span>
<span id="cb29-29"><a href="#cb29-29" aria-hidden="true" tabindex="-1"></a>dt <span class="ot">&lt;-</span> both_summ <span class="sc">%&gt;%</span> </span>
<span id="cb29-30"><a href="#cb29-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(cancer_type <span class="sc">%in%</span> tt<span class="sc">$</span>cancer_type) <span class="sc">%&gt;%</span> </span>
<span id="cb29-31"><a href="#cb29-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(pathway, cancer_type, diff_or) <span class="sc">%&gt;%</span> </span>
<span id="cb29-32"><a href="#cb29-32" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">names_from =</span> cancer_type,<span class="at">values_from =</span> diff_or) <span class="sc">%&gt;%</span> </span>
<span id="cb29-33"><a href="#cb29-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb29-34"><a href="#cb29-34" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(dt) <span class="ot">&lt;-</span> dt<span class="sc">$</span>pathway</span>
<span id="cb29-35"><a href="#cb29-35" aria-hidden="true" tabindex="-1"></a>dt <span class="ot">&lt;-</span> dt <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>pathway)</span>
<span id="cb29-36"><a href="#cb29-36" aria-hidden="true" tabindex="-1"></a><span class="do">###把全是 0 的去掉，也就是这个通路在所有癌症类型中都没有差异</span></span>
<span id="cb29-37"><a href="#cb29-37" aria-hidden="true" tabindex="-1"></a>which0 <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">apply</span>(dt,<span class="dv">1</span>,<span class="cf">function</span>(x){<span class="fu">all</span>(x <span class="sc">==</span> <span class="dv">0</span>)}))</span>
<span id="cb29-38"><a href="#cb29-38" aria-hidden="true" tabindex="-1"></a>dt <span class="ot">&lt;-</span> dt[<span class="sc">-</span>which0,]</span>
<span id="cb29-39"><a href="#cb29-39" aria-hidden="true" tabindex="-1"></a>dt <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(dt)</span>
<span id="cb29-40"><a href="#cb29-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-41"><a href="#cb29-41" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(circlize)</span>
<span id="cb29-42"><a href="#cb29-42" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ========================================</span></span>
<span id="cb29-43"><a href="#cb29-43" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; circlize version 0.4.15</span></span>
<span id="cb29-44"><a href="#cb29-44" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CRAN page: https://cran.r-project.org/package=circlize</span></span>
<span id="cb29-45"><a href="#cb29-45" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Github page: https://github.com/jokergoo/circlize</span></span>
<span id="cb29-46"><a href="#cb29-46" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Documentation: https://jokergoo.github.io/circlize_book/book/</span></span>
<span id="cb29-47"><a href="#cb29-47" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb29-48"><a href="#cb29-48" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; If you use it in published research, please cite:</span></span>
<span id="cb29-49"><a href="#cb29-49" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Gu, Z. circlize implements and enhances circular visualization</span></span>
<span id="cb29-50"><a href="#cb29-50" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   in R. Bioinformatics 2014.</span></span>
<span id="cb29-51"><a href="#cb29-51" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb29-52"><a href="#cb29-52" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; This message can be suppressed by:</span></span>
<span id="cb29-53"><a href="#cb29-53" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   suppressPackageStartupMessages(library(circlize))</span></span>
<span id="cb29-54"><a href="#cb29-54" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ========================================</span></span>
<span id="cb29-55"><a href="#cb29-55" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb29-56"><a href="#cb29-56" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb29-57"><a href="#cb29-57" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Attaching package: &#39;circlize&#39;</span></span>
<span id="cb29-58"><a href="#cb29-58" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb29-59"><a href="#cb29-59" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The following object is masked from &#39;package:igraph&#39;:</span></span>
<span id="cb29-60"><a href="#cb29-60" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb29-61"><a href="#cb29-61" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     degree</span></span>
<span id="cb29-62"><a href="#cb29-62" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ComplexHeatmap)</span>
<span id="cb29-63"><a href="#cb29-63" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: grid</span></span>
<span id="cb29-64"><a href="#cb29-64" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ========================================</span></span>
<span id="cb29-65"><a href="#cb29-65" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ComplexHeatmap version 2.14.0</span></span>
<span id="cb29-66"><a href="#cb29-66" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/</span></span>
<span id="cb29-67"><a href="#cb29-67" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Github page: https://github.com/jokergoo/ComplexHeatmap</span></span>
<span id="cb29-68"><a href="#cb29-68" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Documentation: http://jokergoo.github.io/ComplexHeatmap-reference</span></span>
<span id="cb29-69"><a href="#cb29-69" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb29-70"><a href="#cb29-70" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; If you use it in published research, please cite either one:</span></span>
<span id="cb29-71"><a href="#cb29-71" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; - Gu, Z. Complex Heatmap Visualization. iMeta 2022.</span></span>
<span id="cb29-72"><a href="#cb29-72" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; - Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional </span></span>
<span id="cb29-73"><a href="#cb29-73" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     genomic data. Bioinformatics 2016.</span></span>
<span id="cb29-74"><a href="#cb29-74" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb29-75"><a href="#cb29-75" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb29-76"><a href="#cb29-76" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The new InteractiveComplexHeatmap package can directly export static </span></span>
<span id="cb29-77"><a href="#cb29-77" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; complex heatmaps into an interactive Shiny app with zero effort. Have a try!</span></span>
<span id="cb29-78"><a href="#cb29-78" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb29-79"><a href="#cb29-79" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; This message can be suppressed by:</span></span>
<span id="cb29-80"><a href="#cb29-80" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   suppressPackageStartupMessages(library(ComplexHeatmap))</span></span>
<span id="cb29-81"><a href="#cb29-81" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ========================================</span></span>
<span id="cb29-82"><a href="#cb29-82" aria-hidden="true" tabindex="-1"></a>col_fun <span class="ot">=</span> <span class="fu">colorRamp2</span>(<span class="fu">c</span>(<span class="fu">min</span>(dt,<span class="at">na.rm =</span> T), <span class="fu">max</span>(dt,<span class="at">na.rm =</span> T)),</span>
<span id="cb29-83"><a href="#cb29-83" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">c</span>(<span class="st">&quot;white&quot;</span>, <span class="st">&quot;red&quot;</span>))</span>
<span id="cb29-84"><a href="#cb29-84" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">Heatmap</span>(dt,<span class="at">cluster_rows =</span> F,<span class="at">cluster_columns =</span> F,</span>
<span id="cb29-85"><a href="#cb29-85" aria-hidden="true" tabindex="-1"></a>              <span class="at">rect_gp =</span> <span class="fu">gpar</span>(<span class="at">col =</span> <span class="st">&quot;grey&quot;</span>, <span class="at">lwd =</span> <span class="dv">2</span>),<span class="at">row_names_side =</span> <span class="st">&quot;left&quot;</span>,</span>
<span id="cb29-86"><a href="#cb29-86" aria-hidden="true" tabindex="-1"></a>              <span class="at">show_heatmap_legend=</span>T,<span class="at">row_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">8</span>),</span>
<span id="cb29-87"><a href="#cb29-87" aria-hidden="true" tabindex="-1"></a>              <span class="at">name=</span><span class="st">&quot;log2OR Cancer VS Normal&quot;</span>,<span class="at">na_col=</span><span class="st">&quot;black&quot;</span>,</span>
<span id="cb29-88"><a href="#cb29-88" aria-hidden="true" tabindex="-1"></a>              <span class="at">row_order =</span> <span class="fu">c</span>(dt_p<span class="sc">$</span>pathway,</span>
<span id="cb29-89"><a href="#cb29-89" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">rownames</span>(dt)[<span class="fu">which</span>(<span class="sc">!</span>(<span class="fu">rownames</span>(dt) <span class="sc">%in%</span> dt_p<span class="sc">$</span>pathway))]))</span>
<span id="cb29-90"><a href="#cb29-90" aria-hidden="true" tabindex="-1"></a>p1</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-19-2.png" width="2400" /></p>
<p>To mitigate the impact of pathway gene set size on enrichment
results, we also conducted the permutation test. For a given pathway in
a particular sample, we randomly sampled genes equal to the number of
genes in that pathway to form a “random gene set”. We calculated the
proportion of genes predicted as positive in this random gene set,
repeated this process 1000 times to obtain 1000 proportion values. Then,
we compared this empirical distribution of 1000 values with the actual
proportion of positive genes in the pathway to obtain a P-value. Thus,
for each sample, we can get such a permutation test P value.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="do">####Obtain P value using permutation test</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="do">##For example, in Terpenoid backbone biosynthesis, there are two genes with predicted values of 1 in the TCGA-95-7947-01 sample. </span></span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a><span class="do">##Then, in this sample, what is the proportion of 2 genes that are randomly selected to be 1? Repeat 1000 times, and the obtained proportions will be compared with the 2/2 above</span></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a><span class="do">###</span></span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>get_random_p <span class="ot">&lt;-</span> <span class="cf">function</span>(pre_dt,orign_value,sampling_counts){</span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>  sampling1000 <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span>,</span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a>                         <span class="cf">function</span>(x,y,num){</span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a>                           dt <span class="ot">&lt;-</span> y <span class="sc">%&gt;%</span> <span class="fu">slice_sample</span>(<span class="at">n=</span>num)</span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">mean</span>(dt<span class="sc">$</span>preds <span class="sc">==</span> <span class="dv">1</span>)</span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a>                         }, pre_dt <span class="sc">%&gt;%</span> <span class="fu">select</span>(preds),</span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a>                         sampling_counts)</span>
<span id="cb30-12"><a href="#cb30-12" aria-hidden="true" tabindex="-1"></a>  p_value <span class="ot">&lt;-</span> <span class="fu">mean</span>(sampling1000 <span class="sc">&gt;=</span> orign_value)</span>
<span id="cb30-13"><a href="#cb30-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(p_value)</span>
<span id="cb30-14"><a href="#cb30-14" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb30-15"><a href="#cb30-15" aria-hidden="true" tabindex="-1"></a>all_split <span class="ot">&lt;-</span> pre_pathway <span class="sc">%&gt;%</span> </span>
<span id="cb30-16"><a href="#cb30-16" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">separate_longer_delim</span>(<span class="at">cols =</span> <span class="st">&quot;pathway&quot;</span>,<span class="at">delim =</span> <span class="st">&quot;;&quot;</span>)</span>
<span id="cb30-17"><a href="#cb30-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-18"><a href="#cb30-18" aria-hidden="true" tabindex="-1"></a><span class="do">##</span></span>
<span id="cb30-19"><a href="#cb30-19" aria-hidden="true" tabindex="-1"></a>all_split_summ <span class="ot">&lt;-</span> all_split <span class="sc">%&gt;%</span> </span>
<span id="cb30-20"><a href="#cb30-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(pathway) <span class="sc">%&gt;%</span> </span>
<span id="cb30-21"><a href="#cb30-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">sample_couts =</span> <span class="fu">length</span>(<span class="fu">unique</span>(cell))) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>()</span>
<span id="cb30-22"><a href="#cb30-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-23"><a href="#cb30-23" aria-hidden="true" tabindex="-1"></a>all_split_summ <span class="ot">&lt;-</span> all_split_summ <span class="sc">%&gt;%</span> </span>
<span id="cb30-24"><a href="#cb30-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(sample_couts <span class="sc">&gt;=</span> <span class="dv">2000</span>)</span>
<span id="cb30-25"><a href="#cb30-25" aria-hidden="true" tabindex="-1"></a>all_split <span class="ot">&lt;-</span> all_split <span class="sc">%&gt;%</span> </span>
<span id="cb30-26"><a href="#cb30-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(pathway <span class="sc">%in%</span> all_split_summ<span class="sc">$</span>pathway)</span>
<span id="cb30-27"><a href="#cb30-27" aria-hidden="true" tabindex="-1"></a>pathway_samples <span class="ot">&lt;-</span> all_split <span class="sc">%&gt;%</span> </span>
<span id="cb30-28"><a href="#cb30-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(cell,pathway) <span class="sc">%&gt;%</span> <span class="fu">distinct_all</span>()</span>
<span id="cb30-29"><a href="#cb30-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-30"><a href="#cb30-30" aria-hidden="true" tabindex="-1"></a><span class="do">###并行计算</span></span>
<span id="cb30-31"><a href="#cb30-31" aria-hidden="true" tabindex="-1"></a>my.cluster <span class="ot">&lt;-</span> parallel<span class="sc">::</span><span class="fu">makeCluster</span>(</span>
<span id="cb30-32"><a href="#cb30-32" aria-hidden="true" tabindex="-1"></a>  <span class="dv">90</span>, </span>
<span id="cb30-33"><a href="#cb30-33" aria-hidden="true" tabindex="-1"></a>  <span class="at">type =</span> <span class="st">&quot;PSOCK&quot;</span></span>
<span id="cb30-34"><a href="#cb30-34" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb30-35"><a href="#cb30-35" aria-hidden="true" tabindex="-1"></a><span class="co">#register it to be used by %dopar%</span></span>
<span id="cb30-36"><a href="#cb30-36" aria-hidden="true" tabindex="-1"></a>doParallel<span class="sc">::</span><span class="fu">registerDoParallel</span>(<span class="at">cl =</span> my.cluster)</span>
<span id="cb30-37"><a href="#cb30-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-38"><a href="#cb30-38" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">foreach</span>(</span>
<span id="cb30-39"><a href="#cb30-39" aria-hidden="true" tabindex="-1"></a>  <span class="at">i =</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(pathway_samples),</span>
<span id="cb30-40"><a href="#cb30-40" aria-hidden="true" tabindex="-1"></a>  <span class="at">.export =</span> <span class="fu">c</span>(<span class="st">&quot;pre_pathway&quot;</span>,<span class="st">&quot;get_random_p&quot;</span>,<span class="st">&quot;pathway_samples&quot;</span>),</span>
<span id="cb30-41"><a href="#cb30-41" aria-hidden="true" tabindex="-1"></a>  <span class="at">.packages =</span> <span class="fu">c</span>(<span class="st">&quot;dplyr&quot;</span>)</span>
<span id="cb30-42"><a href="#cb30-42" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%dopar%</span> {</span>
<span id="cb30-43"><a href="#cb30-43" aria-hidden="true" tabindex="-1"></a>  tt <span class="ot">&lt;-</span> pre_pathway <span class="sc">%&gt;%</span> <span class="fu">filter</span>(cell <span class="sc">==</span> pathway_samples<span class="sc">$</span>cell[i])</span>
<span id="cb30-44"><a href="#cb30-44" aria-hidden="true" tabindex="-1"></a>  tt1 <span class="ot">&lt;-</span> tt <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">grepl</span>(pathway_samples<span class="sc">$</span>pathway[i],pathway))</span>
<span id="cb30-45"><a href="#cb30-45" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">get_random_p</span>(tt,<span class="fu">mean</span>(tt1<span class="sc">$</span>preds <span class="sc">==</span> <span class="dv">1</span>),<span class="fu">nrow</span>(tt1))</span>
<span id="cb30-46"><a href="#cb30-46" aria-hidden="true" tabindex="-1"></a>  dt <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb30-47"><a href="#cb30-47" aria-hidden="true" tabindex="-1"></a>    <span class="at">sample =</span> pathway_samples<span class="sc">$</span>cell[i],</span>
<span id="cb30-48"><a href="#cb30-48" aria-hidden="true" tabindex="-1"></a>    <span class="at">pathway =</span> pathway_samples<span class="sc">$</span>pathway[i],</span>
<span id="cb30-49"><a href="#cb30-49" aria-hidden="true" tabindex="-1"></a>    <span class="at">p_value =</span> p</span>
<span id="cb30-50"><a href="#cb30-50" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb30-51"><a href="#cb30-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(dt)</span>
<span id="cb30-52"><a href="#cb30-52" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb30-53"><a href="#cb30-53" aria-hidden="true" tabindex="-1"></a>parallel<span class="sc">::</span><span class="fu">stopCluster</span>(<span class="at">cl =</span> my.cluster)</span>
<span id="cb30-54"><a href="#cb30-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-55"><a href="#cb30-55" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(res)</span>
<span id="cb30-56"><a href="#cb30-56" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(res,<span class="at">file =</span> <span class="st">&quot;~/DeepMeta/data/tcga_pathway_permutation.rds&quot;</span>)</span>
<span id="cb30-57"><a href="#cb30-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-58"><a href="#cb30-58" aria-hidden="true" tabindex="-1"></a><span class="do">####对正常样本也做这样的分析</span></span>
<span id="cb30-59"><a href="#cb30-59" aria-hidden="true" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/kegg_all_pathway.rds&quot;</span>)</span>
<span id="cb30-60"><a href="#cb30-60" aria-hidden="true" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> kegg <span class="sc">%&gt;%</span> </span>
<span id="cb30-61"><a href="#cb30-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">&quot;Metabolism&quot;</span>,class) <span class="sc">|</span> <span class="fu">grepl</span>(<span class="st">&quot;metabolism&quot;</span>,pathway)) <span class="sc">%&gt;%</span> </span>
<span id="cb30-62"><a href="#cb30-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pathway =</span> <span class="fu">gsub</span>(<span class="st">&quot; </span><span class="sc">\\</span><span class="st">- Homo sapiens </span><span class="sc">\\</span><span class="st">(human</span><span class="sc">\\</span><span class="st">)&quot;</span>,<span class="st">&quot;&quot;</span>,pathway)) <span class="sc">%&gt;%</span> </span>
<span id="cb30-63"><a href="#cb30-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;Drug&quot;</span>,pathway))</span>
<span id="cb30-64"><a href="#cb30-64" aria-hidden="true" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> kegg <span class="sc">%&gt;%</span> </span>
<span id="cb30-65"><a href="#cb30-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pathway =</span> <span class="fu">case_when</span>(</span>
<span id="cb30-66"><a href="#cb30-66" aria-hidden="true" tabindex="-1"></a>    pathway <span class="sc">==</span> <span class="st">&quot;Citrate cycle (TCA cycle)&quot;</span> <span class="sc">~</span> <span class="st">&quot;Citrate cycle&quot;</span>,</span>
<span id="cb30-67"><a href="#cb30-67" aria-hidden="true" tabindex="-1"></a>    pathway <span class="sc">==</span> <span class="st">&quot;Glycosylphosphatidylinositol (GPI)-anchor biosynthesis&quot;</span> <span class="sc">~</span> <span class="st">&quot;Glycosylphosphatidylinositol&quot;</span>,</span>
<span id="cb30-68"><a href="#cb30-68" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> pathway</span>
<span id="cb30-69"><a href="#cb30-69" aria-hidden="true" tabindex="-1"></a>  ))</span>
<span id="cb30-70"><a href="#cb30-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-71"><a href="#cb30-71" aria-hidden="true" tabindex="-1"></a>get_pathway <span class="ot">&lt;-</span> <span class="cf">function</span>(gene,pathway){</span>
<span id="cb30-72"><a href="#cb30-72" aria-hidden="true" tabindex="-1"></a>  gene <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(gene,<span class="at">split =</span> <span class="st">&quot;,&quot;</span>)[[<span class="dv">1</span>]]</span>
<span id="cb30-73"><a href="#cb30-73" aria-hidden="true" tabindex="-1"></a>  pathway <span class="ot">&lt;-</span> pathway <span class="sc">%&gt;%</span> <span class="fu">filter</span>(genes <span class="sc">%in%</span> gene)</span>
<span id="cb30-74"><a href="#cb30-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">c</span>(<span class="fu">paste</span>(<span class="fu">unique</span>(pathway<span class="sc">$</span>pathway),<span class="at">collapse =</span> <span class="st">&quot;;&quot;</span>),</span>
<span id="cb30-75"><a href="#cb30-75" aria-hidden="true" tabindex="-1"></a>           <span class="fu">paste</span>(<span class="fu">unique</span>(pathway<span class="sc">$</span>class),<span class="at">collapse =</span> <span class="st">&quot;;&quot;</span>)))</span>
<span id="cb30-76"><a href="#cb30-76" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb30-77"><a href="#cb30-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-78"><a href="#cb30-78" aria-hidden="true" tabindex="-1"></a>pre <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;data/tcga_normal_pre_V2.rds&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb30-79"><a href="#cb30-79" aria-hidden="true" tabindex="-1"></a>pre_pathway <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene =</span> <span class="fu">unique</span>(<span class="fu">c</span>(pre<span class="sc">$</span>gene))) <span class="sc">%&gt;%</span> </span>
<span id="cb30-80"><a href="#cb30-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb30-81"><a href="#cb30-81" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pathway =</span> <span class="fu">get_pathway</span>(gene,kegg)[<span class="dv">1</span>],</span>
<span id="cb30-82"><a href="#cb30-82" aria-hidden="true" tabindex="-1"></a>         <span class="at">class =</span> <span class="fu">get_pathway</span>(gene,kegg)[<span class="dv">2</span>]) <span class="sc">%&gt;%</span> </span>
<span id="cb30-83"><a href="#cb30-83" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb30-84"><a href="#cb30-84" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">nchar</span>(pathway)<span class="sc">&gt;</span><span class="dv">1</span>)</span>
<span id="cb30-85"><a href="#cb30-85" aria-hidden="true" tabindex="-1"></a>pre_pathway <span class="ot">&lt;-</span> pre <span class="sc">%&gt;%</span> </span>
<span id="cb30-86"><a href="#cb30-86" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(.,pre_pathway) <span class="sc">%&gt;%</span> <span class="fu">na.omit</span>()</span>
<span id="cb30-87"><a href="#cb30-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-88"><a href="#cb30-88" aria-hidden="true" tabindex="-1"></a>all_split <span class="ot">&lt;-</span> pre_pathway <span class="sc">%&gt;%</span> </span>
<span id="cb30-89"><a href="#cb30-89" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">separate_longer_delim</span>(<span class="at">cols =</span> <span class="st">&quot;pathway&quot;</span>,<span class="at">delim =</span> <span class="st">&quot;;&quot;</span>)</span>
<span id="cb30-90"><a href="#cb30-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-91"><a href="#cb30-91" aria-hidden="true" tabindex="-1"></a><span class="do">##</span></span>
<span id="cb30-92"><a href="#cb30-92" aria-hidden="true" tabindex="-1"></a>cancer_res <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/tcga_pathway_permutation.rds&quot;</span>)</span>
<span id="cb30-93"><a href="#cb30-93" aria-hidden="true" tabindex="-1"></a>all_split <span class="ot">&lt;-</span> all_split <span class="sc">%&gt;%</span> </span>
<span id="cb30-94"><a href="#cb30-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(pathway <span class="sc">%in%</span> cancer_res<span class="sc">$</span>pathway)</span>
<span id="cb30-95"><a href="#cb30-95" aria-hidden="true" tabindex="-1"></a>pathway_samples <span class="ot">&lt;-</span> all_split <span class="sc">%&gt;%</span> </span>
<span id="cb30-96"><a href="#cb30-96" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(cell,pathway) <span class="sc">%&gt;%</span> <span class="fu">distinct_all</span>()</span>
<span id="cb30-97"><a href="#cb30-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-98"><a href="#cb30-98" aria-hidden="true" tabindex="-1"></a><span class="do">###run same steps</span></span>
<span id="cb30-99"><a href="#cb30-99" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(res,<span class="at">file =</span> <span class="st">&quot;~/DeepMeta/data/tcga_pathway_permutation_normal.rds&quot;</span>)</span></code></pre></div>
<p>Thus, for each sample, we can get such a permutation test P value. If
this P value was less than 0.05, then the sample was defined as
dependent on this metabolic pathway. The distribution of differences in
percentage of samples show dependency to these pathways between TCGA
cancer and normal tissues shows similar pattern:</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>cancer_res <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/tcga_pathway_permutation.rds&quot;</span>)</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>normal_res <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/tcga_pathway_permutation_normal.rds&quot;</span>)</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>both <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>  cancer_res <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">type =</span> <span class="st">&quot;cancer&quot;</span>),</span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>  normal_res <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">type =</span> <span class="st">&quot;normal&quot;</span>)</span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>both<span class="sc">$</span>cancer_type <span class="ot">&lt;-</span> EasyBioinfo<span class="sc">::</span><span class="fu">get_cancer_type</span>(both<span class="sc">$</span>sample,<span class="at">cores =</span> <span class="dv">50</span>, </span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a>                                                 <span class="at">parallel =</span> <span class="cn">TRUE</span>)</span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; you have 128 cores</span></span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a>cancer_summ2 <span class="ot">&lt;-</span> both <span class="sc">%&gt;%</span> </span>
<span id="cb31-11"><a href="#cb31-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(cancer_type,type) <span class="sc">%&gt;%</span> </span>
<span id="cb31-12"><a href="#cb31-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">counts =</span> <span class="fu">length</span>(<span class="fu">unique</span>(sample))) <span class="sc">%&gt;%</span> </span>
<span id="cb31-13"><a href="#cb31-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb31-14"><a href="#cb31-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(type <span class="sc">==</span> <span class="st">&quot;normal&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb31-15"><a href="#cb31-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(counts <span class="sc">&gt;</span> <span class="dv">10</span>)</span>
<span id="cb31-16"><a href="#cb31-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; `summarise()` has grouped output by &#39;cancer_type&#39;. You can override using the</span></span>
<span id="cb31-17"><a href="#cb31-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; `.groups` argument.</span></span>
<span id="cb31-18"><a href="#cb31-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-19"><a href="#cb31-19" aria-hidden="true" tabindex="-1"></a>both_summ <span class="ot">&lt;-</span> both <span class="sc">%&gt;%</span> </span>
<span id="cb31-20"><a href="#cb31-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(cancer_type <span class="sc">%in%</span> cancer_summ2<span class="sc">$</span>cancer_type) <span class="sc">%&gt;%</span> </span>
<span id="cb31-21"><a href="#cb31-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(pathway,cancer_type,type) <span class="sc">%&gt;%</span> </span>
<span id="cb31-22"><a href="#cb31-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">sig_per =</span> <span class="fu">mean</span>(p_value <span class="sc">&lt;</span> <span class="fl">0.05</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb31-23"><a href="#cb31-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb31-24"><a href="#cb31-24" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">names_from =</span> <span class="st">&quot;type&quot;</span>, <span class="at">values_from =</span> <span class="st">&quot;sig_per&quot;</span>)</span>
<span id="cb31-25"><a href="#cb31-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; `summarise()` has grouped output by &#39;pathway&#39;, &#39;cancer_type&#39;. You can override</span></span>
<span id="cb31-26"><a href="#cb31-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; using the `.groups` argument.</span></span>
<span id="cb31-27"><a href="#cb31-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-28"><a href="#cb31-28" aria-hidden="true" tabindex="-1"></a>both_summ <span class="ot">&lt;-</span> both_summ <span class="sc">%&gt;%</span> </span>
<span id="cb31-29"><a href="#cb31-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">diff =</span> cancer <span class="sc">-</span> normal) <span class="sc">%&gt;%</span></span>
<span id="cb31-30"><a href="#cb31-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(pathway <span class="sc">%in%</span> <span class="fu">rownames</span>(dt))</span>
<span id="cb31-31"><a href="#cb31-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-32"><a href="#cb31-32" aria-hidden="true" tabindex="-1"></a>dt1 <span class="ot">&lt;-</span> both_summ <span class="sc">%&gt;%</span> </span>
<span id="cb31-33"><a href="#cb31-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(pathway, cancer_type, diff) <span class="sc">%&gt;%</span> </span>
<span id="cb31-34"><a href="#cb31-34" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">names_from =</span> cancer_type, <span class="at">values_from =</span> diff) <span class="sc">%&gt;%</span></span>
<span id="cb31-35"><a href="#cb31-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb31-36"><a href="#cb31-36" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(dt1) <span class="ot">&lt;-</span> dt1<span class="sc">$</span>pathway</span>
<span id="cb31-37"><a href="#cb31-37" aria-hidden="true" tabindex="-1"></a>dt1<span class="sc">$</span>pathway <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb31-38"><a href="#cb31-38" aria-hidden="true" tabindex="-1"></a>dt1 <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(dt1)</span>
<span id="cb31-39"><a href="#cb31-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-40"><a href="#cb31-40" aria-hidden="true" tabindex="-1"></a>col_fun <span class="ot">=</span> <span class="fu">colorRamp2</span>(<span class="fu">c</span>(<span class="fu">min</span>(dt1,<span class="at">na.rm =</span> T), <span class="fu">max</span>(dt1,<span class="at">na.rm =</span> T)),</span>
<span id="cb31-41"><a href="#cb31-41" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">c</span>(<span class="st">&quot;white&quot;</span>, <span class="st">&quot;red&quot;</span>))</span>
<span id="cb31-42"><a href="#cb31-42" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">Heatmap</span>(dt1,<span class="at">cluster_rows =</span> F,<span class="at">cluster_columns =</span> F,</span>
<span id="cb31-43"><a href="#cb31-43" aria-hidden="true" tabindex="-1"></a>              <span class="at">rect_gp =</span> <span class="fu">gpar</span>(<span class="at">col =</span> <span class="st">&quot;grey&quot;</span>, <span class="at">lwd =</span> <span class="dv">2</span>),</span>
<span id="cb31-44"><a href="#cb31-44" aria-hidden="true" tabindex="-1"></a>              <span class="at">row_names_side =</span> <span class="st">&quot;left&quot;</span>,</span>
<span id="cb31-45"><a href="#cb31-45" aria-hidden="true" tabindex="-1"></a>              <span class="at">show_heatmap_legend=</span>T,<span class="at">row_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">8</span>),</span>
<span id="cb31-46"><a href="#cb31-46" aria-hidden="true" tabindex="-1"></a>              <span class="at">name=</span><span class="st">&quot;Significant sample proportion </span><span class="sc">\n</span><span class="st">Cancer VS Normal&quot;</span>,</span>
<span id="cb31-47"><a href="#cb31-47" aria-hidden="true" tabindex="-1"></a>              <span class="at">na_col=</span><span class="st">&quot;black&quot;</span>,</span>
<span id="cb31-48"><a href="#cb31-48" aria-hidden="true" tabindex="-1"></a>              <span class="at">row_order =</span> <span class="fu">c</span>(dt_p<span class="sc">$</span>pathway,</span>
<span id="cb31-49"><a href="#cb31-49" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">rownames</span>(dt1)[<span class="fu">which</span>(<span class="sc">!</span>(<span class="fu">rownames</span>(dt1) <span class="sc">%in%</span> dt_p<span class="sc">$</span>pathway))]))</span>
<span id="cb31-50"><a href="#cb31-50" aria-hidden="true" tabindex="-1"></a>p1</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-21-1.png" width="2400" /></p>
<p>To demonstrate the potential clinical applications of DeepMeta, we
divided patients from the TCGA dataset which received anti-pyrimidine
metabolism drug treatment (including capecitabine, pemetrexed,
gemcitabine and fluorouracil) into two groups based on the dependency of
pyrimidine metabolism pathway and calculated the survival difference
between the two groups. We found that among patients who received
anti-pyrimidine metabolism drug treatment, those predicted by DeepMeta
to have a higher dependence on pyrimidine metabolism had significantly
better survival (<code>Fig 3d</code>). Interestingly, among patients who
had not been treated by anti-pyrimidine drugs, the group with higher
dependency exhibits significantly worse prognosis:</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ezcox)</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Welcome to &#39;ezcox&#39; package!</span></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; =======================================================================</span></span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; You are using ezcox version 1.0.4</span></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Project home : https://github.com/ShixiangWang/ezcox</span></span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Documentation: https://shixiangwang.github.io/ezcox</span></span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Cite as      : arXiv:2110.14232</span></span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; =======================================================================</span></span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb32-11"><a href="#cb32-11" aria-hidden="true" tabindex="-1"></a><span class="do">###</span></span>
<span id="cb32-12"><a href="#cb32-12" aria-hidden="true" tabindex="-1"></a>surv <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/pancancer_survial.rds&quot;</span>)</span>
<span id="cb32-13"><a href="#cb32-13" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/pancancer_meta_dep.rds&quot;</span>)</span>
<span id="cb32-14"><a href="#cb32-14" aria-hidden="true" tabindex="-1"></a>py <span class="ot">&lt;-</span> res <span class="sc">%&gt;%</span> <span class="fu">filter</span>(pathway <span class="sc">==</span> <span class="st">&quot;Pyrimidine metabolism&quot;</span>)</span>
<span id="cb32-15"><a href="#cb32-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-16"><a href="#cb32-16" aria-hidden="true" tabindex="-1"></a>tcga_drug <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/tcga_drug.rds&quot;</span>)</span>
<span id="cb32-17"><a href="#cb32-17" aria-hidden="true" tabindex="-1"></a>tcga_drug <span class="ot">&lt;-</span> tcga_drug <span class="sc">%&gt;%</span> </span>
<span id="cb32-18"><a href="#cb32-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">response_type =</span> </span>
<span id="cb32-19"><a href="#cb32-19" aria-hidden="true" tabindex="-1"></a>           <span class="fu">ifelse</span>(response <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Complete Response&quot;</span>,<span class="st">&quot;Partial Response&quot;</span>),<span class="st">&quot;response&quot;</span>,</span>
<span id="cb32-20"><a href="#cb32-20" aria-hidden="true" tabindex="-1"></a>                  <span class="st">&quot;no-response&quot;</span>),</span>
<span id="cb32-21"><a href="#cb32-21" aria-hidden="true" tabindex="-1"></a>         <span class="at">drug_type =</span> <span class="fu">ifelse</span>(drug.name <span class="sc">%in%</span> </span>
<span id="cb32-22"><a href="#cb32-22" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">c</span>(<span class="st">&quot;Capecitabine&quot;</span>,<span class="st">&quot;Pemetrexed&quot;</span>,<span class="st">&quot;Gemcitabine&quot;</span>,<span class="st">&quot;Fluorouracil&quot;</span>),</span>
<span id="cb32-23"><a href="#cb32-23" aria-hidden="true" tabindex="-1"></a>                            <span class="st">&quot;anti_nm&quot;</span>,<span class="st">&quot;no_anti_nm&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-24"><a href="#cb32-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(patient.arr <span class="sc">%in%</span> <span class="fu">substr</span>(py<span class="sc">$</span>sample,<span class="dv">1</span>,<span class="dv">12</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-25"><a href="#cb32-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(patient.arr,drug_type) <span class="sc">%&gt;%</span> </span>
<span id="cb32-26"><a href="#cb32-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct_all</span>()</span>
<span id="cb32-27"><a href="#cb32-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-28"><a href="#cb32-28" aria-hidden="true" tabindex="-1"></a>tcga_drug <span class="ot">&lt;-</span> tcga_drug <span class="sc">%&gt;%</span> </span>
<span id="cb32-29"><a href="#cb32-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(patient.arr) <span class="sc">%&gt;%</span> </span>
<span id="cb32-30"><a href="#cb32-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">drug_type =</span> <span class="fu">ifelse</span>(<span class="st">&quot;anti_nm&quot;</span> <span class="sc">%in%</span> drug_type,<span class="st">&quot;anti_nm&quot;</span>,<span class="st">&quot;no_anti_nm&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-31"><a href="#cb32-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb32-32"><a href="#cb32-32" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb32-33"><a href="#cb32-33" aria-hidden="true" tabindex="-1"></a><span class="do">## anti-nm</span></span>
<span id="cb32-34"><a href="#cb32-34" aria-hidden="true" tabindex="-1"></a>surv_nm <span class="ot">&lt;-</span> <span class="fu">left_join</span>(</span>
<span id="cb32-35"><a href="#cb32-35" aria-hidden="true" tabindex="-1"></a>  tcga_drug <span class="sc">%&gt;%</span> </span>
<span id="cb32-36"><a href="#cb32-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rename</span>(<span class="at">sample=</span>patient.arr) <span class="sc">%&gt;%</span> </span>
<span id="cb32-37"><a href="#cb32-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(drug_type <span class="sc">==</span> <span class="st">&quot;anti_nm&quot;</span>),</span>
<span id="cb32-38"><a href="#cb32-38" aria-hidden="true" tabindex="-1"></a>  surv</span>
<span id="cb32-39"><a href="#cb32-39" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(.,</span>
<span id="cb32-40"><a href="#cb32-40" aria-hidden="true" tabindex="-1"></a>                py <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">substr</span>(sample,<span class="dv">1</span>,<span class="dv">12</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb32-41"><a href="#cb32-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct_all</span>(<span class="at">.keep_all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb32-42"><a href="#cb32-42" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(sample)`</span></span>
<span id="cb32-43"><a href="#cb32-43" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(sample)`</span></span>
<span id="cb32-44"><a href="#cb32-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-45"><a href="#cb32-45" aria-hidden="true" tabindex="-1"></a>surv_nm<span class="sc">$</span>cancers <span class="ot">&lt;-</span> EasyBioinfo<span class="sc">::</span><span class="fu">get_cancer_type</span>(surv_nm<span class="sc">$</span>sample)</span>
<span id="cb32-46"><a href="#cb32-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-47"><a href="#cb32-47" aria-hidden="true" tabindex="-1"></a>cancer_summ <span class="ot">&lt;-</span> surv_nm <span class="sc">%&gt;%</span> </span>
<span id="cb32-48"><a href="#cb32-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(cancers) <span class="sc">%&gt;%</span> </span>
<span id="cb32-49"><a href="#cb32-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">counts =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb32-50"><a href="#cb32-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(counts <span class="sc">&gt;</span> <span class="dv">10</span>)</span>
<span id="cb32-51"><a href="#cb32-51" aria-hidden="true" tabindex="-1"></a>surv_nm_filter <span class="ot">&lt;-</span> surv_nm <span class="sc">%&gt;%</span> </span>
<span id="cb32-52"><a href="#cb32-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(cancers <span class="sc">%in%</span> cancer_summ<span class="sc">$</span>cancers)</span>
<span id="cb32-53"><a href="#cb32-53" aria-hidden="true" tabindex="-1"></a>surv_nm_filter <span class="ot">&lt;-</span> surv_nm_filter <span class="sc">%&gt;%</span> </span>
<span id="cb32-54"><a href="#cb32-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">OR_type =</span> <span class="fu">ifelse</span>(ratio <span class="sc">&gt;</span> <span class="fu">median</span>(surv_nm_filter<span class="sc">$</span>ratio), <span class="st">&quot;high&quot;</span>,<span class="st">&quot;low&quot;</span>)) </span>
<span id="cb32-55"><a href="#cb32-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-56"><a href="#cb32-56" aria-hidden="true" tabindex="-1"></a>surv_nm_filter<span class="sc">$</span>OR_type <span class="ot">&lt;-</span> <span class="fu">factor</span>(surv_nm_filter<span class="sc">$</span>OR_type, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;low&quot;</span>,<span class="st">&quot;high&quot;</span>))</span>
<span id="cb32-57"><a href="#cb32-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-58"><a href="#cb32-58" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">show_forest</span>(surv_nm_filter,<span class="at">covariates =</span> <span class="st">&quot;OR_type&quot;</span>,<span class="at">time =</span> <span class="st">&quot;OS.time&quot;</span>,</span>
<span id="cb32-59"><a href="#cb32-59" aria-hidden="true" tabindex="-1"></a>                  <span class="at">status =</span> <span class="st">&quot;OS&quot;</span>,<span class="at">controls =</span> <span class="st">&quot;cancers&quot;</span>,<span class="at">vars_to_show =</span> <span class="st">&quot;OR_type&quot;</span></span>
<span id="cb32-60"><a href="#cb32-60" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb32-61"><a href="#cb32-61" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; =&gt; Processing variable OR_type</span></span>
<span id="cb32-62"><a href="#cb32-62" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ==&gt; Building Surv object...</span></span>
<span id="cb32-63"><a href="#cb32-63" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ==&gt; Building Cox model...</span></span>
<span id="cb32-64"><a href="#cb32-64" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ==&gt; Done.</span></span>
<span id="cb32-65"><a href="#cb32-65" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Resized limits to included dashed line in forest panel</span></span>
<span id="cb32-66"><a href="#cb32-66" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> EasyBioinfo<span class="sc">::</span><span class="fu">show_km</span>(surv_nm_filter ,<span class="st">&quot;OR_type&quot;</span>,<span class="at">title=</span><span class="st">&quot;All Cancer Type&quot;</span>)</span>
<span id="cb32-67"><a href="#cb32-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-68"><a href="#cb32-68" aria-hidden="true" tabindex="-1"></a><span class="do">###all patient</span></span>
<span id="cb32-69"><a href="#cb32-69" aria-hidden="true" tabindex="-1"></a>surv_all <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(</span>
<span id="cb32-70"><a href="#cb32-70" aria-hidden="true" tabindex="-1"></a>  surv,</span>
<span id="cb32-71"><a href="#cb32-71" aria-hidden="true" tabindex="-1"></a>  py <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">substr</span>(sample,<span class="dv">1</span>,<span class="dv">12</span>))</span>
<span id="cb32-72"><a href="#cb32-72" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">distinct_all</span>(<span class="at">.keep_all =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(OS.time)) </span>
<span id="cb32-73"><a href="#cb32-73" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(sample)`</span></span>
<span id="cb32-74"><a href="#cb32-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-75"><a href="#cb32-75" aria-hidden="true" tabindex="-1"></a><span class="do">##</span></span>
<span id="cb32-76"><a href="#cb32-76" aria-hidden="true" tabindex="-1"></a>surv_all <span class="ot">&lt;-</span> surv_all <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>(sample <span class="sc">%in%</span> surv_nm<span class="sc">$</span>sample))</span>
<span id="cb32-77"><a href="#cb32-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-78"><a href="#cb32-78" aria-hidden="true" tabindex="-1"></a>surv_all<span class="sc">$</span>cancers <span class="ot">&lt;-</span> EasyBioinfo<span class="sc">::</span><span class="fu">get_cancer_type</span>(surv_all<span class="sc">$</span>sample)</span>
<span id="cb32-79"><a href="#cb32-79" aria-hidden="true" tabindex="-1"></a>cancer_summ <span class="ot">&lt;-</span> surv_all <span class="sc">%&gt;%</span></span>
<span id="cb32-80"><a href="#cb32-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(cancers) <span class="sc">%&gt;%</span></span>
<span id="cb32-81"><a href="#cb32-81" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">counts =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb32-82"><a href="#cb32-82" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(counts <span class="sc">&gt;</span> <span class="dv">10</span>)</span>
<span id="cb32-83"><a href="#cb32-83" aria-hidden="true" tabindex="-1"></a>surv_all_filter <span class="ot">&lt;-</span> surv_all <span class="sc">%&gt;%</span> </span>
<span id="cb32-84"><a href="#cb32-84" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(cancers <span class="sc">%in%</span> cancer_summ<span class="sc">$</span>cancers)</span>
<span id="cb32-85"><a href="#cb32-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-86"><a href="#cb32-86" aria-hidden="true" tabindex="-1"></a>surv_all_filter <span class="ot">&lt;-</span> surv_all_filter <span class="sc">%&gt;%</span> </span>
<span id="cb32-87"><a href="#cb32-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">OR_type =</span> <span class="fu">ifelse</span>(ratio <span class="sc">&gt;</span> <span class="fu">median</span>(surv_all_filter<span class="sc">$</span>ratio), <span class="st">&quot;high&quot;</span>,<span class="st">&quot;low&quot;</span>))</span>
<span id="cb32-88"><a href="#cb32-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-89"><a href="#cb32-89" aria-hidden="true" tabindex="-1"></a>surv_all_filter<span class="sc">$</span>OR_type <span class="ot">&lt;-</span> <span class="fu">factor</span>(surv_all_filter<span class="sc">$</span>OR_type, </span>
<span id="cb32-90"><a href="#cb32-90" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;low&quot;</span>,<span class="st">&quot;high&quot;</span>))</span>
<span id="cb32-91"><a href="#cb32-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-92"><a href="#cb32-92" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">show_forest</span>(surv_all_filter,<span class="at">covariates =</span> <span class="st">&quot;OR_type&quot;</span>,<span class="at">time =</span> <span class="st">&quot;OS.time&quot;</span>,</span>
<span id="cb32-93"><a href="#cb32-93" aria-hidden="true" tabindex="-1"></a>                  <span class="at">status =</span> <span class="st">&quot;OS&quot;</span>,<span class="at">controls =</span> <span class="st">&quot;cancers&quot;</span>,<span class="at">vars_to_show =</span> <span class="st">&quot;OR_type&quot;</span>)</span>
<span id="cb32-94"><a href="#cb32-94" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; =&gt; Processing variable OR_type</span></span>
<span id="cb32-95"><a href="#cb32-95" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ==&gt; Building Surv object...</span></span>
<span id="cb32-96"><a href="#cb32-96" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ==&gt; Building Cox model...</span></span>
<span id="cb32-97"><a href="#cb32-97" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ==&gt; Done.</span></span>
<span id="cb32-98"><a href="#cb32-98" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Resized limits to included dashed line in forest panel</span></span>
<span id="cb32-99"><a href="#cb32-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-100"><a href="#cb32-100" aria-hidden="true" tabindex="-1"></a>p4 <span class="ot">&lt;-</span> EasyBioinfo<span class="sc">::</span><span class="fu">show_km</span>(surv_all_filter ,<span class="st">&quot;OR_type&quot;</span>,<span class="at">title=</span><span class="st">&quot;All Cancer Type&quot;</span>)</span>
<span id="cb32-101"><a href="#cb32-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-102"><a href="#cb32-102" aria-hidden="true" tabindex="-1"></a>p5 <span class="ot">&lt;-</span> survminer<span class="sc">::</span><span class="fu">arrange_ggsurvplots</span>(<span class="fu">list</span>(p2,p4))</span>
<span id="cb32-103"><a href="#cb32-103" aria-hidden="true" tabindex="-1"></a>p5</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-22-1.png" width="2400" /></p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>p1 <span class="sc">+</span> p3</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-22-2.png" width="2400" /></p>
<p>In these patients, we compared the enrichment of cell cycle-related
gene signature (through GSVA analysis):</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>CCS_genes <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/meta_target/data/CCS_genes.rds&quot;</span>)</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>tumor_tpm <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;/home/data/sdc/wt/TCGA/tcga_RSEM_gene_tpm.gz&quot;</span>,</span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>                               <span class="at">data.table =</span> F)</span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>tumor_tpm <span class="ot">&lt;-</span> tumor_tpm <span class="sc">%&gt;%</span> </span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample,<span class="fu">which</span>(<span class="fu">as.numeric</span>(<span class="fu">substr</span>(<span class="fu">colnames</span>(tumor_tpm),<span class="dv">14</span>,<span class="dv">15</span>)) <span class="sc">&lt;</span> <span class="dv">10</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">id =</span> sample)</span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>mapping <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;/home/data/sdc/wt/TCGA/probeMap_gencode.v23.annotation.gene.probemap&quot;</span>,</span>
<span id="cb34-8"><a href="#cb34-8" aria-hidden="true" tabindex="-1"></a>                             <span class="at">data.table =</span> F)</span>
<span id="cb34-9"><a href="#cb34-9" aria-hidden="true" tabindex="-1"></a>tumor_tpm <span class="ot">&lt;-</span> <span class="fu">left_join</span>(tumor_tpm,mapping <span class="sc">%&gt;%</span> <span class="fu">select</span>(id,gene)) <span class="sc">%&gt;%</span> </span>
<span id="cb34-10"><a href="#cb34-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>id) <span class="sc">%&gt;%</span> </span>
<span id="cb34-11"><a href="#cb34-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(gene,<span class="fu">everything</span>())</span>
<span id="cb34-12"><a href="#cb34-12" aria-hidden="true" tabindex="-1"></a>tumor_tpm <span class="ot">&lt;-</span> tumor_tpm[<span class="sc">!</span><span class="fu">duplicated</span>(tumor_tpm<span class="sc">$</span>gene),]</span>
<span id="cb34-13"><a href="#cb34-13" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(tumor_tpm) <span class="ot">&lt;-</span> tumor_tpm<span class="sc">$</span>gene</span>
<span id="cb34-14"><a href="#cb34-14" aria-hidden="true" tabindex="-1"></a>tumor_tpm<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb34-15"><a href="#cb34-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-16"><a href="#cb34-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GSVA)</span>
<span id="cb34-17"><a href="#cb34-17" aria-hidden="true" tabindex="-1"></a>gs <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">ccs =</span> CCS_genes<span class="sc">$</span>V1)</span>
<span id="cb34-18"><a href="#cb34-18" aria-hidden="true" tabindex="-1"></a>gsvaPar <span class="ot">&lt;-</span> <span class="fu">gsvaParam</span>(<span class="fu">as.matrix</span>(tumor_tpm), gs)</span>
<span id="cb34-19"><a href="#cb34-19" aria-hidden="true" tabindex="-1"></a>gsva.es <span class="ot">&lt;-</span> <span class="fu">gsva</span>(gsvaPar, <span class="at">verbose=</span><span class="cn">TRUE</span>,</span>
<span id="cb34-20"><a href="#cb34-20" aria-hidden="true" tabindex="-1"></a>                <span class="at">BPPARAM =</span> BiocParallel<span class="sc">::</span><span class="fu">SnowParam</span>(<span class="at">workers =</span> <span class="dv">50</span>, </span>
<span id="cb34-21"><a href="#cb34-21" aria-hidden="true" tabindex="-1"></a>                                                  <span class="at">type =</span> <span class="st">&quot;SOCK&quot;</span>))</span>
<span id="cb34-22"><a href="#cb34-22" aria-hidden="true" tabindex="-1"></a>gsva.es <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(gsva.es))</span>
<span id="cb34-23"><a href="#cb34-23" aria-hidden="true" tabindex="-1"></a>gsva.es<span class="sc">$</span>sample <span class="ot">&lt;-</span> <span class="fu">rownames</span>(gsva.es)</span>
<span id="cb34-24"><a href="#cb34-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-25"><a href="#cb34-25" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/pancancer_meta_dep.rds&quot;</span>)</span>
<span id="cb34-26"><a href="#cb34-26" aria-hidden="true" tabindex="-1"></a>py <span class="ot">&lt;-</span> res <span class="sc">%&gt;%</span> <span class="fu">filter</span>(pathway <span class="sc">==</span> <span class="st">&quot;Pyrimidine metabolism&quot;</span>)</span>
<span id="cb34-27"><a href="#cb34-27" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(</span>
<span id="cb34-28"><a href="#cb34-28" aria-hidden="true" tabindex="-1"></a>  py,</span>
<span id="cb34-29"><a href="#cb34-29" aria-hidden="true" tabindex="-1"></a>  gsva.es</span>
<span id="cb34-30"><a href="#cb34-30" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb34-31"><a href="#cb34-31" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(res,<span class="at">file =</span> <span class="st">&quot;~/DeepMeta/data/tcga_ccs.rds&quot;</span>)</span></code></pre></div>
<p>The results showed that patients with higher dependency on pyrimidine
metabolism pathway had more active expression of cell cycle-related
genes, indicating that tumors which are more dependent on pyrimidine
metabolism may have stronger proliferation phenotype, thus worse
prognosis, while at the same time show strongly improved clinical
response to anti-pyrimidine drugs:</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/pancancer_meta_dep.rds&quot;</span>)</span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>py <span class="ot">&lt;-</span> res <span class="sc">%&gt;%</span> <span class="fu">filter</span>(pathway <span class="sc">==</span> <span class="st">&quot;Pyrimidine metabolism&quot;</span>)</span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>tcga_drug <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/tcga_drug.rds&quot;</span>)</span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>tcga_drug <span class="ot">&lt;-</span> tcga_drug <span class="sc">%&gt;%</span> </span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">drug_type =</span> <span class="fu">ifelse</span>(drug.name <span class="sc">%in%</span>            <span class="fu">c</span>(<span class="st">&quot;Capecitabine&quot;</span>,<span class="st">&quot;Pemetrexed&quot;</span>,<span class="st">&quot;Gemcitabine&quot;</span>,<span class="st">&quot;Fluorouracil&quot;</span>),</span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a>                            <span class="st">&quot;anti_nm&quot;</span>,<span class="st">&quot;no_anti_nm&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(patient.arr <span class="sc">%in%</span> <span class="fu">substr</span>(py<span class="sc">$</span>sample,<span class="dv">1</span>,<span class="dv">12</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(patient.arr,drug_type) <span class="sc">%&gt;%</span> </span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct_all</span>()</span>
<span id="cb35-11"><a href="#cb35-11" aria-hidden="true" tabindex="-1"></a>anti_nm <span class="ot">&lt;-</span> tcga_drug <span class="sc">%&gt;%</span> </span>
<span id="cb35-12"><a href="#cb35-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(patient.arr) <span class="sc">%&gt;%</span> </span>
<span id="cb35-13"><a href="#cb35-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">drug_type =</span> <span class="fu">ifelse</span>(<span class="st">&quot;anti_nm&quot;</span> <span class="sc">%in%</span> drug_type,</span>
<span id="cb35-14"><a href="#cb35-14" aria-hidden="true" tabindex="-1"></a>                               <span class="st">&quot;anti_nm&quot;</span>,<span class="st">&quot;no_anti_nm&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-15"><a href="#cb35-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb35-16"><a href="#cb35-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">sample=</span>patient.arr) <span class="sc">%&gt;%</span> </span>
<span id="cb35-17"><a href="#cb35-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(drug_type <span class="sc">==</span> <span class="st">&quot;anti_nm&quot;</span>)</span>
<span id="cb35-18"><a href="#cb35-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-19"><a href="#cb35-19" aria-hidden="true" tabindex="-1"></a><span class="do">###</span></span>
<span id="cb35-20"><a href="#cb35-20" aria-hidden="true" tabindex="-1"></a>py_no_nm <span class="ot">&lt;-</span> py <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>(<span class="fu">substr</span>(sample,<span class="dv">1</span>,<span class="dv">12</span>) <span class="sc">%in%</span> anti_nm<span class="sc">$</span>sample))</span>
<span id="cb35-21"><a href="#cb35-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-22"><a href="#cb35-22" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/tcga_ccs.rds&quot;</span>)</span>
<span id="cb35-23"><a href="#cb35-23" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> res <span class="sc">%&gt;%</span> </span>
<span id="cb35-24"><a href="#cb35-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(sample <span class="sc">%in%</span> py_no_nm<span class="sc">$</span>sample)</span>
<span id="cb35-25"><a href="#cb35-25" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> res <span class="sc">%&gt;%</span> </span>
<span id="cb35-26"><a href="#cb35-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">type =</span> <span class="fu">ifelse</span>(ratio <span class="sc">&gt;</span> <span class="fu">median</span>(res<span class="sc">$</span>ratio),<span class="st">&quot;high&quot;</span>,<span class="st">&quot;low&quot;</span>))</span>
<span id="cb35-27"><a href="#cb35-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-28"><a href="#cb35-28" aria-hidden="true" tabindex="-1"></a><span class="fu">ggboxplot</span>(res,<span class="at">x=</span><span class="st">&quot;type&quot;</span>,<span class="at">y=</span><span class="st">&quot;ccs&quot;</span>,<span class="at">xlab =</span> <span class="st">&quot;OR_type&quot;</span>,</span>
<span id="cb35-29"><a href="#cb35-29" aria-hidden="true" tabindex="-1"></a>          <span class="at">ylab =</span> <span class="st">&quot;Cell cycle signature score&quot;</span>)<span class="sc">+</span></span>
<span id="cb35-30"><a href="#cb35-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>()</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-24-1.png" width="2400" /></p>
</div>
<div
id="metabolic-vulnerabilities-of-cancers-with-undruggable-driver-genes"
class="section level3">
<h3>Metabolic vulnerabilities of cancers with undruggable driver
genes</h3>
<p>Genetic alterations are major driving forces for cancer, and some
driving genetic alterations such as EGFR-L858R, BRAF-V600E created drug
targets for treating cancers that are dependent on these drivers.
However there are many recurrent cancer driving events, such as genetic
alterations in CTNNB1, MYC, TP53, generate proteins lack accessible
hydrophobic pockets, in which small molecules can bind with high
affinity and thus has been termed “undruggable”. We would like to ask
whether we could indirectly target these cancer driving genetic
alterations from the perspective of metabolic dependency. We considered
three currently undruggable cancer driver genes, including MYC, TP53 and
CTNNB1. For each driver gene, we first screened metabolic genes at the
pan-cancer level that predicted dependencies were correlated with the
mutational status of the driver gene:</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(caret)</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/kegg_all_pathway.rds&quot;</span>)</span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> kegg <span class="sc">%&gt;%</span> </span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">&quot;Metabolism&quot;</span>,class) <span class="sc">|</span> <span class="fu">grepl</span>(<span class="st">&quot;metabolism&quot;</span>,pathway)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pathway =</span> <span class="fu">gsub</span>(<span class="st">&quot; </span><span class="sc">\\</span><span class="st">- Homo sapiens </span><span class="sc">\\</span><span class="st">(human</span><span class="sc">\\</span><span class="st">)&quot;</span>,<span class="st">&quot;&quot;</span>,pathway))</span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a>get_pathway <span class="ot">&lt;-</span> <span class="cf">function</span>(gene,pathway){</span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a>  gene <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(gene,<span class="at">split =</span> <span class="st">&quot;,&quot;</span>)[[<span class="dv">1</span>]]</span>
<span id="cb36-10"><a href="#cb36-10" aria-hidden="true" tabindex="-1"></a>  pathway <span class="ot">&lt;-</span> pathway <span class="sc">%&gt;%</span> <span class="fu">filter</span>(genes <span class="sc">%in%</span> gene)</span>
<span id="cb36-11"><a href="#cb36-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">c</span>(<span class="fu">paste</span>(<span class="fu">unique</span>(pathway<span class="sc">$</span>pathway),<span class="at">collapse =</span> <span class="st">&quot;;&quot;</span>),</span>
<span id="cb36-12"><a href="#cb36-12" aria-hidden="true" tabindex="-1"></a>           <span class="fu">paste</span>(<span class="fu">unique</span>(pathway<span class="sc">$</span>class),<span class="at">collapse =</span> <span class="st">&quot;;&quot;</span>)))</span>
<span id="cb36-13"><a href="#cb36-13" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb36-14"><a href="#cb36-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-15"><a href="#cb36-15" aria-hidden="true" tabindex="-1"></a>pre <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/tcga_pre_V2.rds&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb36-16"><a href="#cb36-16" aria-hidden="true" tabindex="-1"></a>pre_pathway <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene =</span> <span class="fu">unique</span>(<span class="fu">c</span>(pre<span class="sc">$</span>gene))) <span class="sc">%&gt;%</span> </span>
<span id="cb36-17"><a href="#cb36-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb36-18"><a href="#cb36-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pathway =</span> <span class="fu">get_pathway</span>(gene,kegg)[<span class="dv">1</span>],</span>
<span id="cb36-19"><a href="#cb36-19" aria-hidden="true" tabindex="-1"></a>         <span class="at">class =</span> <span class="fu">get_pathway</span>(gene,kegg)[<span class="dv">2</span>]) <span class="sc">%&gt;%</span> </span>
<span id="cb36-20"><a href="#cb36-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb36-21"><a href="#cb36-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">nchar</span>(pathway)<span class="sc">&gt;</span><span class="dv">1</span>)</span>
<span id="cb36-22"><a href="#cb36-22" aria-hidden="true" tabindex="-1"></a>pre_pathway <span class="ot">&lt;-</span> pre <span class="sc">%&gt;%</span> </span>
<span id="cb36-23"><a href="#cb36-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(.,pre_pathway) <span class="sc">%&gt;%</span> <span class="fu">na.omit</span>()</span>
<span id="cb36-24"><a href="#cb36-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-25"><a href="#cb36-25" aria-hidden="true" tabindex="-1"></a>gene_screen <span class="ot">&lt;-</span> <span class="cf">function</span>(pre_dt){</span>
<span id="cb36-26"><a href="#cb36-26" aria-hidden="true" tabindex="-1"></a>  gene_summ <span class="ot">&lt;-</span> pre_dt <span class="sc">%&gt;%</span> </span>
<span id="cb36-27"><a href="#cb36-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(gene) <span class="sc">%&gt;%</span> </span>
<span id="cb36-28"><a href="#cb36-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">counts=</span><span class="fu">mean</span>(preds <span class="sc">==</span> <span class="dv">1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb36-29"><a href="#cb36-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb36-30"><a href="#cb36-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(counts <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> counts <span class="sc">&lt;</span> <span class="dv">1</span>) </span>
<span id="cb36-31"><a href="#cb36-31" aria-hidden="true" tabindex="-1"></a>  gene_summ2 <span class="ot">&lt;-</span> pre_dt <span class="sc">%&gt;%</span> </span>
<span id="cb36-32"><a href="#cb36-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(gene) <span class="sc">%&gt;%</span> </span>
<span id="cb36-33"><a href="#cb36-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">counts=</span><span class="fu">mean</span>(type <span class="sc">==</span> <span class="st">&quot;mut&quot;</span>))  <span class="sc">%&gt;%</span> </span>
<span id="cb36-34"><a href="#cb36-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb36-35"><a href="#cb36-35" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(counts <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> counts <span class="sc">&lt;</span> <span class="dv">1</span>) </span>
<span id="cb36-36"><a href="#cb36-36" aria-hidden="true" tabindex="-1"></a>  all_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(gene_summ<span class="sc">$</span>gene,gene_summ2<span class="sc">$</span>gene)</span>
<span id="cb36-37"><a href="#cb36-37" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb36-38"><a href="#cb36-38" aria-hidden="true" tabindex="-1"></a>  pre_dt <span class="ot">&lt;-</span> pre_dt <span class="sc">%&gt;%</span> <span class="fu">filter</span>(gene <span class="sc">%in%</span> all_genes)</span>
<span id="cb36-39"><a href="#cb36-39" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb36-40"><a href="#cb36-40" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">vector</span>(<span class="st">&quot;list&quot;</span>,<span class="at">length =</span> <span class="fu">length</span>(all_genes))</span>
<span id="cb36-41"><a href="#cb36-41" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(all_genes)){</span>
<span id="cb36-42"><a href="#cb36-42" aria-hidden="true" tabindex="-1"></a>    dt <span class="ot">&lt;-</span> pre_dt <span class="sc">%&gt;%</span> <span class="fu">filter</span>(gene <span class="sc">==</span> all_genes[i])</span>
<span id="cb36-43"><a href="#cb36-43" aria-hidden="true" tabindex="-1"></a>    dt<span class="sc">$</span>type <span class="ot">&lt;-</span> <span class="fu">factor</span>(dt<span class="sc">$</span>type,<span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;non-mut&quot;</span>,<span class="st">&quot;mut&quot;</span>))</span>
<span id="cb36-44"><a href="#cb36-44" aria-hidden="true" tabindex="-1"></a>    dt_cancer <span class="ot">&lt;-</span> dt <span class="sc">%&gt;%</span> <span class="fu">select</span>(cancer)</span>
<span id="cb36-45"><a href="#cb36-45" aria-hidden="true" tabindex="-1"></a>    dummy <span class="ot">&lt;-</span> caret<span class="sc">::</span><span class="fu">dummyVars</span>(<span class="st">&quot; ~ .&quot;</span>, <span class="at">data=</span>dt_cancer)</span>
<span id="cb36-46"><a href="#cb36-46" aria-hidden="true" tabindex="-1"></a>    dt_cancer <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">predict</span>(dummy, <span class="at">newdata=</span>dt_cancer))</span>
<span id="cb36-47"><a href="#cb36-47" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb36-48"><a href="#cb36-48" aria-hidden="true" tabindex="-1"></a>    dt <span class="ot">&lt;-</span> <span class="fu">bind_cols</span>(dt,dt_cancer)</span>
<span id="cb36-49"><a href="#cb36-49" aria-hidden="true" tabindex="-1"></a>    all_caners <span class="ot">&lt;-</span> <span class="fu">colnames</span>(dt_cancer)</span>
<span id="cb36-50"><a href="#cb36-50" aria-hidden="true" tabindex="-1"></a>    dt_model <span class="ot">&lt;-</span> <span class="fu">glm</span>(<span class="at">data =</span> dt, </span>
<span id="cb36-51"><a href="#cb36-51" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">as.formula</span>(<span class="fu">paste0</span>(<span class="st">&quot;preds ~ type + &quot;</span>,</span>
<span id="cb36-52"><a href="#cb36-52" aria-hidden="true" tabindex="-1"></a>                                      <span class="fu">paste</span>(all_caners,<span class="at">collapse =</span> <span class="st">&quot;+&quot;</span>))), </span>
<span id="cb36-53"><a href="#cb36-53" aria-hidden="true" tabindex="-1"></a>                    <span class="at">family =</span><span class="fu">binomial</span>(<span class="at">link =</span> <span class="st">&quot;logit&quot;</span>))</span>
<span id="cb36-54"><a href="#cb36-54" aria-hidden="true" tabindex="-1"></a>    model_summ <span class="ot">&lt;-</span> jtools<span class="sc">::</span><span class="fu">summ</span>(dt_model,<span class="at">exp=</span><span class="cn">TRUE</span>)<span class="sc">$</span>coeftable <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb36-55"><a href="#cb36-55" aria-hidden="true" tabindex="-1"></a>    model_summ<span class="sc">$</span>var <span class="ot">&lt;-</span> <span class="fu">rownames</span>(model_summ)</span>
<span id="cb36-56"><a href="#cb36-56" aria-hidden="true" tabindex="-1"></a>    model_summ <span class="ot">&lt;-</span> model_summ <span class="sc">%&gt;%</span> <span class="fu">filter</span>(var <span class="sc">==</span> <span class="st">&quot;typemut&quot;</span>)</span>
<span id="cb36-57"><a href="#cb36-57" aria-hidden="true" tabindex="-1"></a>    model_summ<span class="sc">$</span>gene <span class="ot">&lt;-</span> all_genes[i]</span>
<span id="cb36-58"><a href="#cb36-58" aria-hidden="true" tabindex="-1"></a>    res[[i]] <span class="ot">&lt;-</span> model_summ</span>
<span id="cb36-59"><a href="#cb36-59" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb36-60"><a href="#cb36-60" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(res)</span>
<span id="cb36-61"><a href="#cb36-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(res)[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;OR&quot;</span>,<span class="st">&quot;lower&quot;</span>,<span class="st">&quot;upper&quot;</span>)</span>
<span id="cb36-62"><a href="#cb36-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(res)[<span class="dv">5</span>] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;P&quot;</span>)</span>
<span id="cb36-63"><a href="#cb36-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(res)[<span class="dv">7</span>] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Gene&quot;</span>)</span>
<span id="cb36-64"><a href="#cb36-64" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> res <span class="sc">%&gt;%</span> <span class="fu">filter</span>(P<span class="sc">&lt;</span><span class="fl">0.05</span> <span class="sc">&amp;</span> OR<span class="sc">&gt;</span><span class="dv">1</span>)</span>
<span id="cb36-65"><a href="#cb36-65" aria-hidden="true" tabindex="-1"></a>  res<span class="sc">$</span>mean <span class="ot">&lt;-</span> res<span class="sc">$</span>OR</span>
<span id="cb36-66"><a href="#cb36-66" aria-hidden="true" tabindex="-1"></a>  res<span class="sc">$</span>OR <span class="ot">&lt;-</span> <span class="fu">round</span>(res<span class="sc">$</span>OR,<span class="dv">3</span>)</span>
<span id="cb36-67"><a href="#cb36-67" aria-hidden="true" tabindex="-1"></a>  res<span class="sc">$</span>P <span class="ot">&lt;-</span> <span class="fu">round</span>(res<span class="sc">$</span>P,<span class="dv">3</span>)</span>
<span id="cb36-68"><a href="#cb36-68" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> res <span class="sc">%&gt;%</span> </span>
<span id="cb36-69"><a href="#cb36-69" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(P,<span class="fu">desc</span>(OR))</span>
<span id="cb36-70"><a href="#cb36-70" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb36-71"><a href="#cb36-71" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb36-72"><a href="#cb36-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-73"><a href="#cb36-73" aria-hidden="true" tabindex="-1"></a><span class="do">####ctnnb</span></span>
<span id="cb36-74"><a href="#cb36-74" aria-hidden="true" tabindex="-1"></a>need_gene <span class="ot">&lt;-</span> <span class="st">&quot;CTNNB1&quot;</span></span>
<span id="cb36-75"><a href="#cb36-75" aria-hidden="true" tabindex="-1"></a>mut_pos <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;p.S45F&quot;</span>,<span class="st">&quot;p.S45P&quot;</span>,<span class="st">&quot;p.S45Y&quot;</span>,<span class="st">&quot;p.S45del&quot;</span>,<span class="st">&quot;p.S45_P52del&quot;</span>,</span>
<span id="cb36-76"><a href="#cb36-76" aria-hidden="true" tabindex="-1"></a>             <span class="st">&quot;p.K335I&quot;</span>,<span class="st">&quot;p.K335T&quot;</span>,<span class="st">&quot;p.N387K&quot;</span>,<span class="st">&quot;p.T41A&quot;</span>,<span class="st">&quot;p.T41I&quot;</span>,</span>
<span id="cb36-77"><a href="#cb36-77" aria-hidden="true" tabindex="-1"></a>             <span class="st">&quot;p.D32A&quot;</span>,<span class="st">&quot;p.D32Y&quot;</span>,<span class="st">&quot;p.D32G&quot;</span>,<span class="st">&quot;p.D32H&quot;</span>,<span class="st">&quot;p.D32V&quot;</span>,<span class="st">&quot;p.D32N&quot;</span>,</span>
<span id="cb36-78"><a href="#cb36-78" aria-hidden="true" tabindex="-1"></a>             <span class="st">&quot;p.S33A&quot;</span>,<span class="st">&quot;p.S33C&quot;</span>,<span class="st">&quot;p.S33F&quot;</span>,<span class="st">&quot;p.S33Y&quot;</span>,<span class="st">&quot;p.S33P&quot;</span>,</span>
<span id="cb36-79"><a href="#cb36-79" aria-hidden="true" tabindex="-1"></a>             <span class="st">&quot;p.G34V&quot;</span>,<span class="st">&quot;p.G34E&quot;</span>,<span class="st">&quot;p.G34R&quot;</span>,</span>
<span id="cb36-80"><a href="#cb36-80" aria-hidden="true" tabindex="-1"></a>             <span class="st">&quot;p.S37A&quot;</span>,<span class="st">&quot;p.S37F&quot;</span>,<span class="st">&quot;p.S37Y&quot;</span>,<span class="st">&quot;p.S37C&quot;</span>)</span>
<span id="cb36-81"><a href="#cb36-81" aria-hidden="true" tabindex="-1"></a>mut_gene <span class="ot">&lt;-</span> mut <span class="sc">%&gt;%</span> </span>
<span id="cb36-82"><a href="#cb36-82" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(gene <span class="sc">==</span> need_gene <span class="sc">&amp;</span> Amino_Acid_Change <span class="sc">%in%</span> mut_pos)</span>
<span id="cb36-83"><a href="#cb36-83" aria-hidden="true" tabindex="-1"></a>all_mut_gene <span class="ot">&lt;-</span> mut <span class="sc">%&gt;%</span> </span>
<span id="cb36-84"><a href="#cb36-84" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(gene <span class="sc">==</span> need_gene)</span>
<span id="cb36-85"><a href="#cb36-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-86"><a href="#cb36-86" aria-hidden="true" tabindex="-1"></a>ctnnb_gene_pre <span class="ot">&lt;-</span> pre_pathway <span class="sc">%&gt;%</span> </span>
<span id="cb36-87"><a href="#cb36-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">type =</span> <span class="fu">case_when</span>(</span>
<span id="cb36-88"><a href="#cb36-88" aria-hidden="true" tabindex="-1"></a>    cell <span class="sc">%in%</span> mut_gene<span class="sc">$</span>sample <span class="sc">~</span> <span class="st">&quot;mut&quot;</span>,</span>
<span id="cb36-89"><a href="#cb36-89" aria-hidden="true" tabindex="-1"></a>    <span class="sc">!</span>(cell <span class="sc">%in%</span> all_mut_gene<span class="sc">$</span>sample) <span class="sc">~</span> <span class="st">&quot;non-mut&quot;</span>,</span>
<span id="cb36-90"><a href="#cb36-90" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">&quot;other&quot;</span></span>
<span id="cb36-91"><a href="#cb36-91" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(type <span class="sc">!=</span> <span class="st">&quot;other&quot;</span>) </span>
<span id="cb36-92"><a href="#cb36-92" aria-hidden="true" tabindex="-1"></a>ctnnb_res <span class="ot">&lt;-</span> <span class="fu">gene_screen</span>(<span class="at">pre_dt =</span> ctnnb_gene_pre)</span>
<span id="cb36-93"><a href="#cb36-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-94"><a href="#cb36-94" aria-hidden="true" tabindex="-1"></a><span class="do">###MYC</span></span>
<span id="cb36-95"><a href="#cb36-95" aria-hidden="true" tabindex="-1"></a>gistic <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;/home/data/sdc/wt/TCGA/GISTIC.focal_data_by_genes.conf_95.txt.gz&quot;</span>,<span class="at">data.table =</span> F)</span>
<span id="cb36-96"><a href="#cb36-96" aria-hidden="true" tabindex="-1"></a>gistic_myc <span class="ot">&lt;-</span> gistic <span class="sc">%&gt;%</span> </span>
<span id="cb36-97"><a href="#cb36-97" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="st">`</span><span class="at">Gene Symbol</span><span class="st">`</span> <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;MYC&quot;</span>,<span class="st">&quot;MYCL1&quot;</span>,<span class="st">&quot;MYCN&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-98"><a href="#cb36-98" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">3</span>)) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb36-99"><a href="#cb36-99" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(gistic_myc) <span class="ot">&lt;-</span> gistic_myc<span class="sc">$</span><span class="st">`</span><span class="at">Gene Symbol</span><span class="st">`</span></span>
<span id="cb36-100"><a href="#cb36-100" aria-hidden="true" tabindex="-1"></a>gistic_myc <span class="ot">&lt;-</span> gistic_myc <span class="sc">%&gt;%</span> </span>
<span id="cb36-101"><a href="#cb36-101" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="st">`</span><span class="at">Gene Symbol</span><span class="st">`</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-102"><a href="#cb36-102" aria-hidden="true" tabindex="-1"></a>  <span class="fu">t</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb36-103"><a href="#cb36-103" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb36-104"><a href="#cb36-104" aria-hidden="true" tabindex="-1"></a>gistic_myc<span class="sc">$</span>sample <span class="ot">&lt;-</span> <span class="fu">rownames</span>(gistic_myc)</span>
<span id="cb36-105"><a href="#cb36-105" aria-hidden="true" tabindex="-1"></a>gistic_myc <span class="ot">&lt;-</span> gistic_myc <span class="sc">%&gt;%</span> </span>
<span id="cb36-106"><a href="#cb36-106" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb36-107"><a href="#cb36-107" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">type =</span> <span class="fu">ifelse</span>(<span class="fu">any</span>(<span class="fu">c</span>(MYC,MYCL1,MYCN)<span class="sc">&gt;</span><span class="dv">0</span>),<span class="st">&quot;amp&quot;</span>,<span class="st">&quot;non-amp&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-108"><a href="#cb36-108" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb36-109"><a href="#cb36-109" aria-hidden="true" tabindex="-1"></a>gistic_myc <span class="ot">&lt;-</span> gistic_myc <span class="sc">%&gt;%</span> </span>
<span id="cb36-110"><a href="#cb36-110" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">substr</span>(sample,<span class="dv">1</span>,<span class="dv">15</span>))</span>
<span id="cb36-111"><a href="#cb36-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-112"><a href="#cb36-112" aria-hidden="true" tabindex="-1"></a>myc_amp <span class="ot">&lt;-</span> gistic_myc <span class="sc">%&gt;%</span> <span class="fu">filter</span>(type <span class="sc">==</span> <span class="st">&quot;amp&quot;</span>)</span>
<span id="cb36-113"><a href="#cb36-113" aria-hidden="true" tabindex="-1"></a>all_mut_gene <span class="ot">&lt;-</span> mut <span class="sc">%&gt;%</span> </span>
<span id="cb36-114"><a href="#cb36-114" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(gene <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;MYC&quot;</span>,<span class="st">&quot;MYCL1&quot;</span>,<span class="st">&quot;MYCN&quot;</span>))</span>
<span id="cb36-115"><a href="#cb36-115" aria-hidden="true" tabindex="-1"></a>all_mut_sample <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">c</span>(all_mut_gene<span class="sc">$</span>sample,myc_amp<span class="sc">$</span>sample))</span>
<span id="cb36-116"><a href="#cb36-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-117"><a href="#cb36-117" aria-hidden="true" tabindex="-1"></a>myc_gene_pre <span class="ot">&lt;-</span> pre_pathway <span class="sc">%&gt;%</span> </span>
<span id="cb36-118"><a href="#cb36-118" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">type =</span> <span class="fu">case_when</span>(</span>
<span id="cb36-119"><a href="#cb36-119" aria-hidden="true" tabindex="-1"></a>    cell <span class="sc">%in%</span> myc_amp<span class="sc">$</span>sample <span class="sc">~</span> <span class="st">&quot;mut&quot;</span>,</span>
<span id="cb36-120"><a href="#cb36-120" aria-hidden="true" tabindex="-1"></a>    <span class="sc">!</span>(cell <span class="sc">%in%</span> all_mut_sample) <span class="sc">~</span> <span class="st">&quot;non-mut&quot;</span>,</span>
<span id="cb36-121"><a href="#cb36-121" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">&quot;other&quot;</span></span>
<span id="cb36-122"><a href="#cb36-122" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(type <span class="sc">!=</span> <span class="st">&quot;other&quot;</span>) </span>
<span id="cb36-123"><a href="#cb36-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-124"><a href="#cb36-124" aria-hidden="true" tabindex="-1"></a>myc_res <span class="ot">&lt;-</span> <span class="fu">gene_screen</span>(<span class="at">pre_dt =</span> myc_gene_pre)</span>
<span id="cb36-125"><a href="#cb36-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-126"><a href="#cb36-126" aria-hidden="true" tabindex="-1"></a><span class="do">###tp53</span></span>
<span id="cb36-127"><a href="#cb36-127" aria-hidden="true" tabindex="-1"></a>tp53_mut <span class="ot">&lt;-</span> mut <span class="sc">%&gt;%</span> </span>
<span id="cb36-128"><a href="#cb36-128" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(gene <span class="sc">==</span> <span class="st">&quot;TP53&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-129"><a href="#cb36-129" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(effect <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Frame_Shift_Del&quot;</span>,<span class="st">&quot;Frame_Shift_Ins&quot;</span>,<span class="st">&quot;In_Frame_Del&quot;</span>,</span>
<span id="cb36-130"><a href="#cb36-130" aria-hidden="true" tabindex="-1"></a>                       <span class="st">&quot;In_Frame_Ins&quot;</span>,<span class="st">&quot;Missense_Mutation&quot;</span>,<span class="st">&quot;Nonsense_Mutation&quot;</span>,</span>
<span id="cb36-131"><a href="#cb36-131" aria-hidden="true" tabindex="-1"></a>                       <span class="st">&quot;Splice_Site&quot;</span>) <span class="sc">|</span> Amino_Acid_Change <span class="sc">==</span> <span class="st">&quot;p.T125T&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-132"><a href="#cb36-132" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mut_index =</span> <span class="fu">paste</span>(chr,start,end,reference,alt,<span class="at">sep =</span> <span class="st">&quot;-&quot;</span>))</span>
<span id="cb36-133"><a href="#cb36-133" aria-hidden="true" tabindex="-1"></a>tp53_mut_ckb <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/tp53_mut_ckb.rds&quot;</span>)</span>
<span id="cb36-134"><a href="#cb36-134" aria-hidden="true" tabindex="-1"></a>tp53_mut_ckb <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;p.&quot;</span>,tp53_mut_ckb<span class="sc">$</span>mut)</span>
<span id="cb36-135"><a href="#cb36-135" aria-hidden="true" tabindex="-1"></a>tp53_mut_ckb <span class="ot">&lt;-</span> tp53_mut_ckb[<span class="fu">which</span>(tp53_mut_ckb <span class="sc">%in%</span> tp53_mut<span class="sc">$</span>Amino_Acid_Change)]</span>
<span id="cb36-136"><a href="#cb36-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-137"><a href="#cb36-137" aria-hidden="true" tabindex="-1"></a>need_gene <span class="ot">&lt;-</span> <span class="st">&quot;TP53&quot;</span></span>
<span id="cb36-138"><a href="#cb36-138" aria-hidden="true" tabindex="-1"></a>mut_pos <span class="ot">&lt;-</span> tp53_mut_ckb</span>
<span id="cb36-139"><a href="#cb36-139" aria-hidden="true" tabindex="-1"></a>mut_gene <span class="ot">&lt;-</span> mut <span class="sc">%&gt;%</span> </span>
<span id="cb36-140"><a href="#cb36-140" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(gene <span class="sc">==</span> need_gene <span class="sc">&amp;</span> Amino_Acid_Change <span class="sc">%in%</span> mut_pos)</span>
<span id="cb36-141"><a href="#cb36-141" aria-hidden="true" tabindex="-1"></a>all_mut_gene <span class="ot">&lt;-</span> mut <span class="sc">%&gt;%</span> </span>
<span id="cb36-142"><a href="#cb36-142" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(gene <span class="sc">==</span> need_gene)</span>
<span id="cb36-143"><a href="#cb36-143" aria-hidden="true" tabindex="-1"></a>tp53_gene_pre <span class="ot">&lt;-</span> pre_pathway <span class="sc">%&gt;%</span> </span>
<span id="cb36-144"><a href="#cb36-144" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">type =</span> <span class="fu">case_when</span>(</span>
<span id="cb36-145"><a href="#cb36-145" aria-hidden="true" tabindex="-1"></a>    cell <span class="sc">%in%</span> mut_gene<span class="sc">$</span>sample <span class="sc">~</span> <span class="st">&quot;mut&quot;</span>,</span>
<span id="cb36-146"><a href="#cb36-146" aria-hidden="true" tabindex="-1"></a>    <span class="sc">!</span>(cell <span class="sc">%in%</span> all_mut_gene<span class="sc">$</span>sample) <span class="sc">~</span> <span class="st">&quot;non-mut&quot;</span>,</span>
<span id="cb36-147"><a href="#cb36-147" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">&quot;other&quot;</span></span>
<span id="cb36-148"><a href="#cb36-148" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(type <span class="sc">!=</span> <span class="st">&quot;other&quot;</span>) </span>
<span id="cb36-149"><a href="#cb36-149" aria-hidden="true" tabindex="-1"></a>tp53_res <span class="ot">&lt;-</span> <span class="fu">gene_screen</span>(<span class="at">pre_dt =</span> tp53_gene_pre)</span>
<span id="cb36-150"><a href="#cb36-150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-151"><a href="#cb36-151" aria-hidden="true" tabindex="-1"></a>all_res <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(</span>
<span id="cb36-152"><a href="#cb36-152" aria-hidden="true" tabindex="-1"></a>  myc_res <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">target =</span> <span class="st">&quot;MYC&quot;</span>),</span>
<span id="cb36-153"><a href="#cb36-153" aria-hidden="true" tabindex="-1"></a>  tp53_res <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">target =</span> <span class="st">&quot;TP53&quot;</span>),</span>
<span id="cb36-154"><a href="#cb36-154" aria-hidden="true" tabindex="-1"></a>  ctnnb_res <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">target =</span> <span class="st">&quot;CTNNB1&quot;</span>)</span>
<span id="cb36-155"><a href="#cb36-155" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb36-156"><a href="#cb36-156" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(all_res,<span class="at">file =</span> <span class="st">&quot;~/DeepMeta/data/merge_driver_dep.rds&quot;</span>)</span>
<span id="cb36-157"><a href="#cb36-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-158"><a href="#cb36-158" aria-hidden="true" tabindex="-1"></a>gene_pre_res <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb36-159"><a href="#cb36-159" aria-hidden="true" tabindex="-1"></a>  <span class="at">MYC =</span> myc_gene_pre,</span>
<span id="cb36-160"><a href="#cb36-160" aria-hidden="true" tabindex="-1"></a>  <span class="at">TP53 =</span> tp53_gene_pre,</span>
<span id="cb36-161"><a href="#cb36-161" aria-hidden="true" tabindex="-1"></a>  <span class="at">CTNNB1 =</span> ctnnb_gene_pre</span>
<span id="cb36-162"><a href="#cb36-162" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb36-163"><a href="#cb36-163" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(gene_pre_res,</span>
<span id="cb36-164"><a href="#cb36-164" aria-hidden="true" tabindex="-1"></a>        <span class="at">file =</span> <span class="st">&quot;/home/data/sdc/wt/TCGA/tcga_driver_pre.rds&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(forestplot)</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: checkmate</span></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: abind</span></span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a>all_res <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/merge_driver_dep.rds&quot;</span>)</span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a>all_res_filter <span class="ot">&lt;-</span> all_res <span class="sc">%&gt;%</span> </span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.infinite</span>(OR)) <span class="sc">%&gt;%</span> </span>
<span id="cb37-8"><a href="#cb37-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">Target-Gene</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">paste</span>(target,Gene,<span class="at">sep =</span> <span class="st">&quot;-&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb37-9"><a href="#cb37-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">Target-Gene</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">ifelse</span>(</span>
<span id="cb37-10"><a href="#cb37-10" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">Target-Gene</span><span class="st">`</span> <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;TP53-GPX4&quot;</span>,<span class="st">&quot;CTNNB1-IMPDH2&quot;</span>,</span>
<span id="cb37-11"><a href="#cb37-11" aria-hidden="true" tabindex="-1"></a>                         <span class="st">&quot;MYC-CAD&quot;</span>,<span class="st">&quot;MYC-TYMS&quot;</span>),</span>
<span id="cb37-12"><a href="#cb37-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">paste0</span>(<span class="st">`</span><span class="at">Target-Gene</span><span class="st">`</span>,<span class="st">&quot; *&quot;</span>), <span class="st">`</span><span class="at">Target-Gene</span><span class="st">`</span></span>
<span id="cb37-13"><a href="#cb37-13" aria-hidden="true" tabindex="-1"></a>  ))</span>
<span id="cb37-14"><a href="#cb37-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-15"><a href="#cb37-15" aria-hidden="true" tabindex="-1"></a>all_res_filter <span class="sc">|&gt;</span></span>
<span id="cb37-16"><a href="#cb37-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">forestplot</span>(<span class="at">labeltext =</span> <span class="fu">c</span>(<span class="st">`</span><span class="at">Target-Gene</span><span class="st">`</span>, OR, P),</span>
<span id="cb37-17"><a href="#cb37-17" aria-hidden="true" tabindex="-1"></a>             <span class="at">xlog =</span> <span class="cn">TRUE</span>,</span>
<span id="cb37-18"><a href="#cb37-18" aria-hidden="true" tabindex="-1"></a>             <span class="at">xlab =</span> <span class="st">&quot;OR&quot;</span>,<span class="at">boxsize =</span> <span class="fl">0.25</span>,) <span class="sc">|&gt;</span></span>
<span id="cb37-19"><a href="#cb37-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">fp_set_style</span>(<span class="at">box =</span> <span class="st">&quot;royalblue&quot;</span>,</span>
<span id="cb37-20"><a href="#cb37-20" aria-hidden="true" tabindex="-1"></a>               <span class="at">line =</span> <span class="st">&quot;darkblue&quot;</span>,</span>
<span id="cb37-21"><a href="#cb37-21" aria-hidden="true" tabindex="-1"></a>               <span class="at">summary =</span> <span class="st">&quot;royalblue&quot;</span>) <span class="sc">|&gt;</span> </span>
<span id="cb37-22"><a href="#cb37-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">fp_add_header</span>(<span class="st">`</span><span class="at">Target-Gene</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Target-Gene&quot;</span>),</span>
<span id="cb37-23"><a href="#cb37-23" aria-hidden="true" tabindex="-1"></a>                <span class="at">OR =</span> <span class="fu">c</span>(<span class="st">&quot;OR&quot;</span>),</span>
<span id="cb37-24"><a href="#cb37-24" aria-hidden="true" tabindex="-1"></a>                <span class="at">P =</span> <span class="fu">c</span>(<span class="st">&quot;P&quot;</span>)) <span class="sc">|&gt;</span></span>
<span id="cb37-25"><a href="#cb37-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">fp_set_zebra_style</span>(<span class="st">&quot;#EFEFEF&quot;</span>)</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-26-1.png" width="2400" /></p>
<p>Subsequently, in each cancer type, for each screened metabolic gene,
we evaluated the dependency difference of the metabolic gene in samples
with functional mutations in the driver gene compared to samples without
the driver gene mutations, using a one-tailed Fisher’s test.</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>gene_screen_one_cancer_fisher <span class="ot">&lt;-</span> <span class="cf">function</span>(pre_dt,need_gene){</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>  dt <span class="ot">&lt;-</span> pre_dt <span class="sc">%&gt;%</span> <span class="fu">filter</span>(gene <span class="sc">==</span> need_gene)</span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>  dt<span class="sc">$</span>type <span class="ot">&lt;-</span> <span class="fu">factor</span>(dt<span class="sc">$</span>type,<span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;mut&quot;</span>,<span class="st">&quot;non-mut&quot;</span>))</span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>  dt<span class="sc">$</span>preds <span class="ot">&lt;-</span> <span class="fu">factor</span>(dt<span class="sc">$</span>preds, <span class="at">levels =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">0</span>))</span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a>  all_cancers <span class="ot">&lt;-</span> <span class="fu">unique</span>(dt<span class="sc">$</span>cancer)</span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a>  cancer_res <span class="ot">&lt;-</span> <span class="fu">vector</span>(<span class="st">&quot;list&quot;</span>,<span class="fu">length</span>(all_cancers))</span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(cancer_res) <span class="ot">&lt;-</span> all_cancers</span>
<span id="cb38-9"><a href="#cb38-9" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(all_cancers)){</span>
<span id="cb38-10"><a href="#cb38-10" aria-hidden="true" tabindex="-1"></a>    dt_cancer <span class="ot">&lt;-</span> dt <span class="sc">%&gt;%</span> <span class="fu">filter</span>(cancer <span class="sc">==</span> all_cancers[i])</span>
<span id="cb38-11"><a href="#cb38-11" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">length</span>(<span class="fu">unique</span>(<span class="fu">as.character</span>(dt_cancer<span class="sc">$</span>preds))) <span class="sc">==</span> <span class="dv">2</span> <span class="sc">&amp;</span> </span>
<span id="cb38-12"><a href="#cb38-12" aria-hidden="true" tabindex="-1"></a>        <span class="fu">length</span>(<span class="fu">unique</span>(<span class="fu">as.character</span>(dt_cancer<span class="sc">$</span>type))) <span class="sc">==</span> <span class="dv">2</span>){</span>
<span id="cb38-13"><a href="#cb38-13" aria-hidden="true" tabindex="-1"></a>      dt_fisher <span class="ot">&lt;-</span> <span class="fu">fisher.test</span>(<span class="fu">table</span>(dt_cancer<span class="sc">$</span>preds,dt_cancer<span class="sc">$</span>type),</span>
<span id="cb38-14"><a href="#cb38-14" aria-hidden="true" tabindex="-1"></a>                               <span class="at">alternative =</span> <span class="st">&quot;greater&quot;</span>)</span>
<span id="cb38-15"><a href="#cb38-15" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb38-16"><a href="#cb38-16" aria-hidden="true" tabindex="-1"></a>      dt_cancer_summ <span class="ot">&lt;-</span> dt_cancer <span class="sc">%&gt;%</span> </span>
<span id="cb38-17"><a href="#cb38-17" aria-hidden="true" tabindex="-1"></a>        <span class="fu">group_by</span>(cancer,type,preds) <span class="sc">%&gt;%</span> </span>
<span id="cb38-18"><a href="#cb38-18" aria-hidden="true" tabindex="-1"></a>        <span class="fu">summarise</span>(<span class="at">counts=</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>()</span>
<span id="cb38-19"><a href="#cb38-19" aria-hidden="true" tabindex="-1"></a>      dt_cancer_summ<span class="sc">$</span>p <span class="ot">&lt;-</span> dt_fisher<span class="sc">$</span>p.value</span>
<span id="cb38-20"><a href="#cb38-20" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb38-21"><a href="#cb38-21" aria-hidden="true" tabindex="-1"></a>      cancer_res[[i]] <span class="ot">&lt;-</span> dt_cancer_summ</span>
<span id="cb38-22"><a href="#cb38-22" aria-hidden="true" tabindex="-1"></a>    }<span class="cf">else</span>{</span>
<span id="cb38-23"><a href="#cb38-23" aria-hidden="true" tabindex="-1"></a>      cancer_res[[i]] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb38-24"><a href="#cb38-24" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb38-25"><a href="#cb38-25" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb38-26"><a href="#cb38-26" aria-hidden="true" tabindex="-1"></a>  cancer_res <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(cancer_res[<span class="fu">which</span>(<span class="fu">lengths</span>(cancer_res)<span class="sc">&gt;</span><span class="dv">1</span>)])</span>
<span id="cb38-27"><a href="#cb38-27" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb38-28"><a href="#cb38-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(cancer_res)</span>
<span id="cb38-29"><a href="#cb38-29" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb38-30"><a href="#cb38-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-31"><a href="#cb38-31" aria-hidden="true" tabindex="-1"></a>get_gene_split_cancer <span class="ot">&lt;-</span> <span class="cf">function</span>(res_dt,pre_dt){</span>
<span id="cb38-32"><a href="#cb38-32" aria-hidden="true" tabindex="-1"></a>  all_genes <span class="ot">&lt;-</span> <span class="fu">unique</span>(res_dt<span class="sc">$</span>Gene)</span>
<span id="cb38-33"><a href="#cb38-33" aria-hidden="true" tabindex="-1"></a>  all_genes_res <span class="ot">&lt;-</span> <span class="fu">vector</span>(<span class="st">&quot;list&quot;</span>,<span class="fu">length</span>(all_genes))</span>
<span id="cb38-34"><a href="#cb38-34" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(all_genes)){</span>
<span id="cb38-35"><a href="#cb38-35" aria-hidden="true" tabindex="-1"></a>    res_cancer <span class="ot">&lt;-</span> <span class="fu">gene_screen_one_cancer_fisher</span>(pre_dt,all_genes[i])</span>
<span id="cb38-36"><a href="#cb38-36" aria-hidden="true" tabindex="-1"></a>    res_cancer<span class="sc">$</span>gene <span class="ot">&lt;-</span> all_genes[i]</span>
<span id="cb38-37"><a href="#cb38-37" aria-hidden="true" tabindex="-1"></a>    all_genes_res[[i]] <span class="ot">&lt;-</span> res_cancer</span>
<span id="cb38-38"><a href="#cb38-38" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb38-39"><a href="#cb38-39" aria-hidden="true" tabindex="-1"></a>  all_genes_res <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(all_genes_res)</span>
<span id="cb38-40"><a href="#cb38-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(all_genes_res)</span>
<span id="cb38-41"><a href="#cb38-41" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb38-42"><a href="#cb38-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-43"><a href="#cb38-43" aria-hidden="true" tabindex="-1"></a>all_res <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/merge_driver_dep.rds&quot;</span>)</span>
<span id="cb38-44"><a href="#cb38-44" aria-hidden="true" tabindex="-1"></a>gene_pre <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;/home/data/sdc/wt/TCGA/tcga_driver_pre.rds&quot;</span>)</span>
<span id="cb38-45"><a href="#cb38-45" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(gene_pre)[<span class="dv">4</span>] <span class="ot">&lt;-</span> <span class="st">&quot;CTNNB1&quot;</span></span>
<span id="cb38-46"><a href="#cb38-46" aria-hidden="true" tabindex="-1"></a>cancer_res <span class="ot">&lt;-</span> <span class="fu">vector</span>(<span class="st">&quot;list&quot;</span>,<span class="dv">4</span>)</span>
<span id="cb38-47"><a href="#cb38-47" aria-hidden="true" tabindex="-1"></a>all_genes <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;TP53&quot;</span>,<span class="st">&quot;MYC&quot;</span>,<span class="st">&quot;CTNNB1&quot;</span>)</span>
<span id="cb38-48"><a href="#cb38-48" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(all_genes)){</span>
<span id="cb38-49"><a href="#cb38-49" aria-hidden="true" tabindex="-1"></a>  dt <span class="ot">&lt;-</span> all_res <span class="sc">%&gt;%</span> <span class="fu">filter</span>(target <span class="sc">==</span> all_genes[i])</span>
<span id="cb38-50"><a href="#cb38-50" aria-hidden="true" tabindex="-1"></a>  dt_gene_pre <span class="ot">&lt;-</span> gene_pre[[all_genes[i]]]</span>
<span id="cb38-51"><a href="#cb38-51" aria-hidden="true" tabindex="-1"></a>  dt_res <span class="ot">&lt;-</span> <span class="fu">get_gene_split_cancer</span>(dt, dt_gene_pre)</span>
<span id="cb38-52"><a href="#cb38-52" aria-hidden="true" tabindex="-1"></a>  dt_res <span class="ot">&lt;-</span> dt_res <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">target =</span> all_genes[i])</span>
<span id="cb38-53"><a href="#cb38-53" aria-hidden="true" tabindex="-1"></a>  cancer_res[[i]] <span class="ot">&lt;-</span> dt_res</span>
<span id="cb38-54"><a href="#cb38-54" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb38-55"><a href="#cb38-55" aria-hidden="true" tabindex="-1"></a>cancer_res <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(cancer_res)</span>
<span id="cb38-56"><a href="#cb38-56" aria-hidden="true" tabindex="-1"></a>cancer_res <span class="ot">&lt;-</span> cancer_res <span class="sc">%&gt;%</span>  <span class="fu">filter</span>(p <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb38-57"><a href="#cb38-57" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(cancer_res,<span class="at">file =</span> <span class="st">&quot;~/DeepMeta/data/split_cancer_res_greater.rds&quot;</span>)</span>
<span id="cb38-58"><a href="#cb38-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-59"><a href="#cb38-59" aria-hidden="true" tabindex="-1"></a><span class="do">###</span></span>
<span id="cb38-60"><a href="#cb38-60" aria-hidden="true" tabindex="-1"></a>all_res <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/split_cancer_res_greater.rds&quot;</span>)</span>
<span id="cb38-61"><a href="#cb38-61" aria-hidden="true" tabindex="-1"></a>all_res_summ <span class="ot">&lt;-</span> all_res <span class="sc">%&gt;%</span> </span>
<span id="cb38-62"><a href="#cb38-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(target,gene,cancer,type) <span class="sc">%&gt;%</span> </span>
<span id="cb38-63"><a href="#cb38-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">percent_1 =</span> counts[<span class="fu">which</span>(preds <span class="sc">==</span> <span class="dv">1</span>)]<span class="sc">/</span><span class="fu">sum</span>(counts),</span>
<span id="cb38-64"><a href="#cb38-64" aria-hidden="true" tabindex="-1"></a>            <span class="at">p_value =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fu">unique</span>(p)),</span>
<span id="cb38-65"><a href="#cb38-65" aria-hidden="true" tabindex="-1"></a>            <span class="at">all_counts =</span> <span class="fu">sum</span>(counts)) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>()</span>
<span id="cb38-66"><a href="#cb38-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-67"><a href="#cb38-67" aria-hidden="true" tabindex="-1"></a><span class="do">##只选择 mut 里面大于 non-mut 里面的</span></span>
<span id="cb38-68"><a href="#cb38-68" aria-hidden="true" tabindex="-1"></a>all_res_summ2 <span class="ot">&lt;-</span> all_res_summ <span class="sc">%&gt;%</span> </span>
<span id="cb38-69"><a href="#cb38-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(target,gene,cancer) <span class="sc">%&gt;%</span> </span>
<span id="cb38-70"><a href="#cb38-70" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">large=</span><span class="fu">ifelse</span>(percent_1[<span class="fu">which</span>(type <span class="sc">==</span> <span class="st">&quot;mut&quot;</span>)] <span class="sc">&gt;</span> percent_1[<span class="fu">which</span>(type <span class="sc">==</span> <span class="st">&quot;non-mut&quot;</span>)],<span class="dv">1</span>,<span class="dv">0</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb38-71"><a href="#cb38-71" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> <span class="fu">filter</span>(large <span class="sc">==</span> <span class="dv">1</span>)</span>
<span id="cb38-72"><a href="#cb38-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-73"><a href="#cb38-73" aria-hidden="true" tabindex="-1"></a>all_res_summ <span class="ot">&lt;-</span> <span class="fu">left_join</span>(</span>
<span id="cb38-74"><a href="#cb38-74" aria-hidden="true" tabindex="-1"></a>  all_res_summ,</span>
<span id="cb38-75"><a href="#cb38-75" aria-hidden="true" tabindex="-1"></a>  all_res_summ2</span>
<span id="cb38-76"><a href="#cb38-76" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb38-77"><a href="#cb38-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-78"><a href="#cb38-78" aria-hidden="true" tabindex="-1"></a>all_res_summ <span class="ot">&lt;-</span> all_res_summ <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(large))</span>
<span id="cb38-79"><a href="#cb38-79" aria-hidden="true" tabindex="-1"></a>all_res_summ <span class="ot">&lt;-</span> all_res_summ <span class="sc">%&gt;%</span> </span>
<span id="cb38-80"><a href="#cb38-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>large) <span class="sc">%&gt;%</span> </span>
<span id="cb38-81"><a href="#cb38-81" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">names_from =</span> <span class="st">&quot;type&quot;</span>,</span>
<span id="cb38-82"><a href="#cb38-82" aria-hidden="true" tabindex="-1"></a>                     <span class="at">values_from =</span> <span class="fu">c</span>(<span class="st">&quot;percent_1&quot;</span>,<span class="st">&quot;all_counts&quot;</span>))</span>
<span id="cb38-83"><a href="#cb38-83" aria-hidden="true" tabindex="-1"></a>all_res_summ <span class="ot">&lt;-</span> all_res_summ <span class="sc">%&gt;%</span> </span>
<span id="cb38-84"><a href="#cb38-84" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">index =</span> <span class="fu">paste</span>(gene,cancer,<span class="at">sep =</span> <span class="st">&quot;-&quot;</span>))</span>
<span id="cb38-85"><a href="#cb38-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-86"><a href="#cb38-86" aria-hidden="true" tabindex="-1"></a><span class="do">###normal</span></span>
<span id="cb38-87"><a href="#cb38-87" aria-hidden="true" tabindex="-1"></a>tcga_normal <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/tcga_normal_pre_V2.rds&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb38-88"><a href="#cb38-88" aria-hidden="true" tabindex="-1"></a>normal_summ <span class="ot">&lt;-</span> tcga_normal <span class="sc">%&gt;%</span> </span>
<span id="cb38-89"><a href="#cb38-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(gene,cancer) <span class="sc">%&gt;%</span> </span>
<span id="cb38-90"><a href="#cb38-90" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">counts =</span> <span class="fu">n</span>(),</span>
<span id="cb38-91"><a href="#cb38-91" aria-hidden="true" tabindex="-1"></a>            <span class="at">pos_percent =</span> <span class="fu">mean</span>(preds <span class="sc">==</span> <span class="dv">1</span>)) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb38-92"><a href="#cb38-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">index =</span> <span class="fu">paste</span>(gene,cancer,<span class="at">sep =</span> <span class="st">&quot;-&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb38-93"><a href="#cb38-93" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(index <span class="sc">%in%</span> all_res_summ<span class="sc">$</span>index)</span>
<span id="cb38-94"><a href="#cb38-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-95"><a href="#cb38-95" aria-hidden="true" tabindex="-1"></a>all_res_summ <span class="ot">&lt;-</span> <span class="fu">left_join</span>(</span>
<span id="cb38-96"><a href="#cb38-96" aria-hidden="true" tabindex="-1"></a>  all_res_summ,</span>
<span id="cb38-97"><a href="#cb38-97" aria-hidden="true" tabindex="-1"></a>  normal_summ <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>cancer,<span class="sc">-</span>gene)</span>
<span id="cb38-98"><a href="#cb38-98" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb38-99"><a href="#cb38-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-100"><a href="#cb38-100" aria-hidden="true" tabindex="-1"></a>dt <span class="ot">&lt;-</span> all_res_summ <span class="sc">%&gt;%</span>  <span class="do">##至少突变3个样本</span></span>
<span id="cb38-101"><a href="#cb38-101" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(all_counts_mut <span class="sc">&gt;=</span><span class="dv">3</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb38-102"><a href="#cb38-102" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ratio1 =</span> <span class="fu">round</span>(percent_1_mut <span class="sc">/</span> <span class="st">`</span><span class="at">percent_1_non-mut</span><span class="st">`</span>,<span class="dv">2</span>),</span>
<span id="cb38-103"><a href="#cb38-103" aria-hidden="true" tabindex="-1"></a>         <span class="at">ratio2 =</span> <span class="fu">round</span>((percent_1_mut <span class="sc">-</span> pos_percent) <span class="sc">/</span> percent_1_mut,<span class="dv">2</span>),</span>
<span id="cb38-104"><a href="#cb38-104" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_value =</span> <span class="fu">round</span>(p_value,<span class="dv">2</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb38-105"><a href="#cb38-105" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(target, gene, cancer, p_value, <span class="fu">c</span>(<span class="dv">12</span>,<span class="dv">13</span>))</span>
<span id="cb38-106"><a href="#cb38-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-107"><a href="#cb38-107" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(dt)[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Target Gene&quot;</span>,<span class="st">&quot;Gene&quot;</span>,<span class="st">&quot;Cancer&quot;</span>)</span>
<span id="cb38-108"><a href="#cb38-108" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(dt,<span class="at">file =</span> <span class="st">&quot;~/DeepMeta/data/split_cancer_webshot2.rds&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>dt <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/DeepMeta/data/split_cancer_webshot2.rds&quot;</span>) </span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(htmltools)</span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reactable)</span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a>bar_chart <span class="ot">&lt;-</span> <span class="cf">function</span>(label, <span class="at">width =</span> <span class="st">&quot;100%&quot;</span>, <span class="at">height =</span> <span class="st">&quot;1rem&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;#00bfc4&quot;</span>, <span class="at">background =</span> <span class="cn">NULL</span>) {</span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a>  bar <span class="ot">&lt;-</span> <span class="fu">div</span>(<span class="at">style =</span> <span class="fu">list</span>(<span class="at">background =</span> fill, <span class="at">width =</span> width, <span class="at">height =</span> height))</span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a>  chart <span class="ot">&lt;-</span> <span class="fu">div</span>(<span class="at">style =</span> <span class="fu">list</span>(<span class="at">flexGrow =</span> <span class="dv">1</span>, <span class="at">marginLeft =</span> <span class="st">&quot;0.5rem&quot;</span>), bar)</span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">div</span>(<span class="at">style =</span> <span class="fu">list</span>(<span class="at">display =</span> <span class="st">&quot;flex&quot;</span>, <span class="at">alignItems =</span> <span class="st">&quot;center&quot;</span>), label, chart)</span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb39-11"><a href="#cb39-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-12"><a href="#cb39-12" aria-hidden="true" tabindex="-1"></a>bar_chart_pos_neg <span class="ot">&lt;-</span> <span class="cf">function</span>(label, <span class="at">width =</span> <span class="st">&quot;100%&quot;</span>, <span class="at">height =</span> <span class="st">&quot;1rem&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;#00bfc4&quot;</span>, <span class="at">background =</span> <span class="cn">NULL</span>) {</span>
<span id="cb39-13"><a href="#cb39-13" aria-hidden="true" tabindex="-1"></a>  neg_chart <span class="ot">&lt;-</span> <span class="fu">div</span>(<span class="at">style =</span> <span class="fu">list</span>(<span class="at">flex =</span> <span class="st">&quot;1 1 0&quot;</span>))</span>
<span id="cb39-14"><a href="#cb39-14" aria-hidden="true" tabindex="-1"></a>  pos_chart <span class="ot">&lt;-</span> <span class="fu">div</span>(<span class="at">style =</span> <span class="fu">list</span>(<span class="at">flex =</span> <span class="st">&quot;1 1 0&quot;</span>))</span>
<span id="cb39-15"><a href="#cb39-15" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (label <span class="sc">&lt;</span> <span class="dv">0</span>){</span>
<span id="cb39-16"><a href="#cb39-16" aria-hidden="true" tabindex="-1"></a>    bar <span class="ot">&lt;-</span> <span class="fu">div</span>(<span class="at">style =</span> <span class="fu">list</span>(<span class="at">background =</span> fill, <span class="at">width =</span> width, <span class="at">height =</span> height, <span class="at">marginLeft =</span> <span class="st">&quot;0.8rem&quot;</span>))</span>
<span id="cb39-17"><a href="#cb39-17" aria-hidden="true" tabindex="-1"></a>    chart <span class="ot">&lt;-</span> <span class="fu">div</span>(<span class="at">style =</span> <span class="fu">list</span>(<span class="at">flexGrow =</span> <span class="dv">1</span>, <span class="at">display =</span> <span class="st">&quot;flex&quot;</span>, <span class="at">alignItems =</span> <span class="st">&quot;center&quot;</span>, <span class="at">justifyContent =</span> <span class="st">&quot;flex-end&quot;</span>), bar)</span>
<span id="cb39-18"><a href="#cb39-18" aria-hidden="true" tabindex="-1"></a>    neg_chart <span class="ot">&lt;-</span> <span class="fu">tagAppendChild</span>(neg_chart, chart)</span>
<span id="cb39-19"><a href="#cb39-19" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb39-20"><a href="#cb39-20" aria-hidden="true" tabindex="-1"></a>    bar <span class="ot">&lt;-</span> <span class="fu">div</span>(<span class="at">style =</span> <span class="fu">list</span>(<span class="at">background =</span> fill, <span class="at">width =</span> width, <span class="at">height =</span> height, <span class="at">marginRight =</span> <span class="st">&quot;0.8rem&quot;</span>))</span>
<span id="cb39-21"><a href="#cb39-21" aria-hidden="true" tabindex="-1"></a>    chart <span class="ot">&lt;-</span> <span class="fu">div</span>(<span class="at">style =</span> <span class="fu">list</span>(<span class="at">flexGrow =</span> <span class="dv">1</span>, <span class="at">display =</span> <span class="st">&quot;flex&quot;</span>, <span class="at">alignItems =</span> <span class="st">&quot;center&quot;</span>), bar)</span>
<span id="cb39-22"><a href="#cb39-22" aria-hidden="true" tabindex="-1"></a>    pos_chart <span class="ot">&lt;-</span> <span class="fu">tagAppendChild</span>(pos_chart, chart)</span>
<span id="cb39-23"><a href="#cb39-23" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb39-24"><a href="#cb39-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">div</span>(<span class="at">style =</span> <span class="fu">list</span>(<span class="at">display =</span> <span class="st">&quot;flex&quot;</span>), label, neg_chart, pos_chart)</span>
<span id="cb39-25"><a href="#cb39-25" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb39-26"><a href="#cb39-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-27"><a href="#cb39-27" aria-hidden="true" tabindex="-1"></a><span class="fu">reactable</span>(</span>
<span id="cb39-28"><a href="#cb39-28" aria-hidden="true" tabindex="-1"></a>  dt,</span>
<span id="cb39-29"><a href="#cb39-29" aria-hidden="true" tabindex="-1"></a>  <span class="at">columns =</span> <span class="fu">list</span>(</span>
<span id="cb39-30"><a href="#cb39-30" aria-hidden="true" tabindex="-1"></a>    <span class="at">ratio1 =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">&quot;Ratio &lt;br&gt; (MP / NMP)&quot;</span>, <span class="at">align =</span> <span class="st">&quot;left&quot;</span>, <span class="at">html =</span> <span class="cn">TRUE</span>, <span class="at">cell =</span> <span class="cf">function</span>(value) {</span>
<span id="cb39-31"><a href="#cb39-31" aria-hidden="true" tabindex="-1"></a>      width <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">abs</span>(value <span class="sc">/</span> <span class="fu">max</span>(dt<span class="sc">$</span>ratio1)) <span class="sc">*</span> <span class="dv">100</span>, <span class="st">&quot;%&quot;</span>)</span>
<span id="cb39-32"><a href="#cb39-32" aria-hidden="true" tabindex="-1"></a>      label <span class="ot">&lt;-</span> <span class="fu">format</span>(value, <span class="at">nsmall =</span> <span class="dv">2</span>)</span>
<span id="cb39-33"><a href="#cb39-33" aria-hidden="true" tabindex="-1"></a>      <span class="fu">bar_chart</span>(label, <span class="at">width =</span> width, <span class="at">fill =</span> <span class="st">&quot;#11325D&quot;</span>, <span class="at">background =</span> <span class="st">&quot;#e1e1e1&quot;</span>)</span>
<span id="cb39-34"><a href="#cb39-34" aria-hidden="true" tabindex="-1"></a>    },<span class="at">minWidth =</span> <span class="dv">100</span>),</span>
<span id="cb39-35"><a href="#cb39-35" aria-hidden="true" tabindex="-1"></a>    <span class="at">ratio2 =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">&quot;Ratio &lt;br&gt; ([MP - NP] / MP)&quot;</span>, <span class="at">align =</span> <span class="st">&quot;left&quot;</span>, <span class="at">html =</span> <span class="cn">TRUE</span>, <span class="at">cell =</span> <span class="cf">function</span>(value) {</span>
<span id="cb39-36"><a href="#cb39-36" aria-hidden="true" tabindex="-1"></a>      width <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">abs</span>(value <span class="sc">/</span> <span class="fu">max</span>(dt<span class="sc">$</span>ratio2)) <span class="sc">*</span> <span class="dv">100</span>, <span class="st">&quot;%&quot;</span>)</span>
<span id="cb39-37"><a href="#cb39-37" aria-hidden="true" tabindex="-1"></a>      label <span class="ot">&lt;-</span> <span class="fu">format</span>(value, <span class="at">nsmall =</span> <span class="dv">2</span>)</span>
<span id="cb39-38"><a href="#cb39-38" aria-hidden="true" tabindex="-1"></a>      <span class="fu">bar_chart_pos_neg</span>(label, <span class="at">width =</span> width, <span class="at">fill =</span> <span class="st">&quot;#f5a673&quot;</span>, <span class="at">background =</span> <span class="st">&quot;#e1e1e1&quot;</span>)</span>
<span id="cb39-39"><a href="#cb39-39" aria-hidden="true" tabindex="-1"></a>    },<span class="at">minWidth =</span> <span class="dv">120</span>),</span>
<span id="cb39-40"><a href="#cb39-40" aria-hidden="true" tabindex="-1"></a>    <span class="at">p_value =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">&quot;P Value (-log10)&quot;</span>, <span class="at">align =</span> <span class="st">&quot;left&quot;</span>, <span class="at">cell =</span> <span class="cf">function</span>(value) {</span>
<span id="cb39-41"><a href="#cb39-41" aria-hidden="true" tabindex="-1"></a>      width <span class="ot">&lt;-</span> <span class="fu">paste0</span>(value <span class="sc">/</span> <span class="fu">max</span>(dt<span class="sc">$</span>p_value) <span class="sc">*</span> <span class="dv">100</span>, <span class="st">&quot;%&quot;</span>)</span>
<span id="cb39-42"><a href="#cb39-42" aria-hidden="true" tabindex="-1"></a>      label <span class="ot">&lt;-</span> <span class="fu">format</span>(value, <span class="at">nsmall =</span> <span class="dv">2</span>)</span>
<span id="cb39-43"><a href="#cb39-43" aria-hidden="true" tabindex="-1"></a>      <span class="fu">bar_chart</span>(label, <span class="at">width =</span> width, <span class="at">fill =</span> <span class="st">&quot;#B783AF&quot;</span>, <span class="at">background =</span> <span class="st">&quot;#e1e1e1&quot;</span>)</span>
<span id="cb39-44"><a href="#cb39-44" aria-hidden="true" tabindex="-1"></a>    },<span class="at">minWidth =</span> <span class="dv">100</span>)</span>
<span id="cb39-45"><a href="#cb39-45" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb39-46"><a href="#cb39-46" aria-hidden="true" tabindex="-1"></a>  <span class="at">defaultPageSize =</span> <span class="dv">40</span></span>
<span id="cb39-47"><a href="#cb39-47" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<div class="reactable html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-d598cc81091e499be576" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-d598cc81091e499be576">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"Target Gene":["CTNNB1","CTNNB1","CTNNB1","CTNNB1","CTNNB1","CTNNB1","CTNNB1","CTNNB1","CTNNB1","CTNNB1","CTNNB1","CTNNB1","CTNNB1","MYC","MYC","MYC","MYC","MYC","MYC","MYC","MYC","MYC","MYC","MYC","MYC","MYC","MYC","MYC","MYC","MYC","TP53","TP53","TP53","TP53"],"Gene":["ACACA","COASY","IMPDH2","IMPDH2","IMPDH2","OGDH","PDPK1","SLC2A1","TYMS","TYMS","TYMS","TYMS","UAP1","CAD","CAD","CDS2","EHMT2","FPGS","HSD17B10","HSD17B10,HADHA,HADH","HSD17B10,HADHA,HADH","SPTLC2,SPTLC1,SPTLC3","TKT","TXNRD1","TXNRD1","TXNRD1","TXNRD1","TYMS","TYMS","TYMS","DHFR","GPX4","GPX4","UROD"],"Cancer":["LIHC","BLCA","BLCA","LIHC","PRAD","BLCA","LIHC","LIHC","BLCA","ESCA","LIHC","LUAD","LIHC","LUAD","SKCM","LGG","LUSC","SKCM","LGG","LGG","SKCM","LGG","SKCM","BRCA","ESCA","KIRP","SKCM","BRCA","SKCM","STAD","BRCA","LUAD","PAAD","LGG"],"p_value":[3.45,2.61,3.09,2.7,1.48,2.72,4.59,1.55,2.01,1.72,1.61,1.86,3.63,2.22,3.19,1.65,1.77,2.42,1.58,1.68,4.24,1.81,2.51,1.97,1.51,2.54,2.15,1.96,1.94,1.38,4.31,4.02,1.33,1.4],"ratio1":[1.51,4.48,8.95,1.73,2.99,2.8,1.4,3.16,6.72,8.97,1.14,3.23,1.54,3.79,2.36,1.45,1.11,2.49,1.59,1.25,3.06,1.4,2.03,2.34,1.61,2.53,1.26,1.61,1.4,1.23,2.82,1.19,2.07,1.05],"ratio2":[-0.28,0.83,1,0.95,0.62,0.1,0.19,0.73,0.86,1,-0.07,1,-0.28,1,1,"NA",0.54,1,"NA","NA",1,"NA",1,0.82,0.79,0.8,1,0.16,1,0.22,0.36,-0.08,1,"NA"]},"columns":[{"id":"Target Gene","name":"Target Gene","type":"character"},{"id":"Gene","name":"Gene","type":"character"},{"id":"Cancer","name":"Cancer","type":"character"},{"id":"p_value","name":"P Value (-log10)","type":"numeric","cell":[{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["3.45",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#B783AF","width":"75.1633986928105%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["2.61",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#B783AF","width":"56.8627450980392%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["3.09",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#B783AF","width":"67.3202614379085%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["2.70",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#B783AF","width":"58.8235294117647%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["1.48",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#B783AF","width":"32.244008714597%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["2.72",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#B783AF","width":"59.2592592592593%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["4.59",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#B783AF","width":"100%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["1.55",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#B783AF","width":"33.7690631808279%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["2.01",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#B783AF","width":"43.7908496732026%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["1.72",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#B783AF","width":"37.4727668845316%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["1.61",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#B783AF","width":"35.0762527233116%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["1.86",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#B783AF","width":"40.5228758169935%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["3.63",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#B783AF","width":"79.0849673202614%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["2.22",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#B783AF","width":"48.3660130718954%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["3.19",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#B783AF","width":"69.4989106753813%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["1.65",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#B783AF","width":"35.9477124183007%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["1.77",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#B783AF","width":"38.562091503268%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["2.42",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#B783AF","width":"52.723311546841%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["1.58",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#B783AF","width":"34.4226579520697%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["1.68",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#B783AF","width":"36.6013071895425%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["4.24",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#B783AF","width":"92.3747276688453%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["1.81",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#B783AF","width":"39.4335511982571%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["2.51",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#B783AF","width":"54.6840958605665%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["1.97",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#B783AF","width":"42.9193899782135%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["1.51",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#B783AF","width":"32.8976034858388%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["2.54",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#B783AF","width":"55.3376906318083%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["2.15",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#B783AF","width":"46.8409586056645%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["1.96",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#B783AF","width":"42.7015250544662%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["1.94",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#B783AF","width":"42.2657952069717%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["1.38",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#B783AF","width":"30.0653594771242%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["4.31",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#B783AF","width":"93.8997821350763%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["4.02",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#B783AF","width":"87.5816993464052%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["1.33",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#B783AF","width":"28.9760348583878%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["1.40",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#B783AF","width":"30.5010893246187%","height":"1rem"}},"children":[]}]}]}],"minWidth":100,"align":"left"},{"id":"ratio1","name":"Ratio <br> (MP / NMP)","type":"numeric","cell":[{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["1.51",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#11325D","width":"16.8338907469342%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["4.48",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#11325D","width":"49.9442586399108%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["8.95",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#11325D","width":"99.7770345596432%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["1.73",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#11325D","width":"19.2865105908584%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["2.99",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#11325D","width":"33.3333333333333%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["2.80",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#11325D","width":"31.2151616499443%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["1.40",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#11325D","width":"15.6075808249721%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["3.16",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#11325D","width":"35.2285395763657%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["6.72",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#11325D","width":"74.9163879598662%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["8.97",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#11325D","width":"100%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["1.14",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#11325D","width":"12.7090301003344%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["3.23",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#11325D","width":"36.0089186176143%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["1.54",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#11325D","width":"17.1683389074693%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["3.79",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#11325D","width":"42.2519509476031%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["2.36",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#11325D","width":"26.3099219620959%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["1.45",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#11325D","width":"16.164994425864%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["1.11",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#11325D","width":"12.3745819397993%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["2.49",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#11325D","width":"27.7591973244147%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["1.59",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#11325D","width":"17.7257525083612%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["1.25",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#11325D","width":"13.9353400222965%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["3.06",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#11325D","width":"34.1137123745819%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["1.40",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#11325D","width":"15.6075808249721%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["2.03",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#11325D","width":"22.6309921962096%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["2.34",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#11325D","width":"26.0869565217391%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["1.61",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#11325D","width":"17.9487179487179%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["2.53",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#11325D","width":"28.2051282051282%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["1.26",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#11325D","width":"14.0468227424749%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["1.61",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#11325D","width":"17.9487179487179%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["1.40",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#11325D","width":"15.6075808249721%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["1.23",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#11325D","width":"13.7123745819398%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["2.82",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#11325D","width":"31.438127090301%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["1.19",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#11325D","width":"13.2664437012263%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["2.07",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#11325D","width":"23.0769230769231%","height":"1rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"display":"flex","alignItems":"center"}},"children":["1.05",{"name":"div","attribs":{"style":{"flexGrow":1,"marginLeft":"0.5rem"}},"children":[{"name":"div","attribs":{"style":{"background":"#11325D","width":"11.7056856187291%","height":"1rem"}},"children":[]}]}]}],"html":true,"minWidth":100,"align":"left"},{"id":"ratio2","name":"Ratio <br> ([MP - NP] / MP)","type":"numeric","cell":[{"name":"div","attribs":{"style":{"display":"flex"}},"children":["-0.28",{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[{"name":"div","attribs":{"style":{"flexGrow":1,"display":"flex","alignItems":"center","justifyContent":"flex-end"}},"children":[{"name":"div","attribs":{"style":{"background":"#f5a673","width":"NA%","height":"1rem","marginLeft":"0.8rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[]}]},{"name":"div","attribs":{"style":{"display":"flex"}},"children":["0.83",{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[]},{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[{"name":"div","attribs":{"style":{"flexGrow":1,"display":"flex","alignItems":"center"}},"children":[{"name":"div","attribs":{"style":{"background":"#f5a673","width":"NA%","height":"1rem","marginRight":"0.8rem"}},"children":[]}]}]}]},{"name":"div","attribs":{"style":{"display":"flex"}},"children":["1.00",{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[]},{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[{"name":"div","attribs":{"style":{"flexGrow":1,"display":"flex","alignItems":"center"}},"children":[{"name":"div","attribs":{"style":{"background":"#f5a673","width":"NA%","height":"1rem","marginRight":"0.8rem"}},"children":[]}]}]}]},{"name":"div","attribs":{"style":{"display":"flex"}},"children":["0.95",{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[]},{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[{"name":"div","attribs":{"style":{"flexGrow":1,"display":"flex","alignItems":"center"}},"children":[{"name":"div","attribs":{"style":{"background":"#f5a673","width":"NA%","height":"1rem","marginRight":"0.8rem"}},"children":[]}]}]}]},{"name":"div","attribs":{"style":{"display":"flex"}},"children":["0.62",{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[]},{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[{"name":"div","attribs":{"style":{"flexGrow":1,"display":"flex","alignItems":"center"}},"children":[{"name":"div","attribs":{"style":{"background":"#f5a673","width":"NA%","height":"1rem","marginRight":"0.8rem"}},"children":[]}]}]}]},{"name":"div","attribs":{"style":{"display":"flex"}},"children":["0.10",{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[]},{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[{"name":"div","attribs":{"style":{"flexGrow":1,"display":"flex","alignItems":"center"}},"children":[{"name":"div","attribs":{"style":{"background":"#f5a673","width":"NA%","height":"1rem","marginRight":"0.8rem"}},"children":[]}]}]}]},{"name":"div","attribs":{"style":{"display":"flex"}},"children":["0.19",{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[]},{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[{"name":"div","attribs":{"style":{"flexGrow":1,"display":"flex","alignItems":"center"}},"children":[{"name":"div","attribs":{"style":{"background":"#f5a673","width":"NA%","height":"1rem","marginRight":"0.8rem"}},"children":[]}]}]}]},{"name":"div","attribs":{"style":{"display":"flex"}},"children":["0.73",{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[]},{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[{"name":"div","attribs":{"style":{"flexGrow":1,"display":"flex","alignItems":"center"}},"children":[{"name":"div","attribs":{"style":{"background":"#f5a673","width":"NA%","height":"1rem","marginRight":"0.8rem"}},"children":[]}]}]}]},{"name":"div","attribs":{"style":{"display":"flex"}},"children":["0.86",{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[]},{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[{"name":"div","attribs":{"style":{"flexGrow":1,"display":"flex","alignItems":"center"}},"children":[{"name":"div","attribs":{"style":{"background":"#f5a673","width":"NA%","height":"1rem","marginRight":"0.8rem"}},"children":[]}]}]}]},{"name":"div","attribs":{"style":{"display":"flex"}},"children":["1.00",{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[]},{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[{"name":"div","attribs":{"style":{"flexGrow":1,"display":"flex","alignItems":"center"}},"children":[{"name":"div","attribs":{"style":{"background":"#f5a673","width":"NA%","height":"1rem","marginRight":"0.8rem"}},"children":[]}]}]}]},{"name":"div","attribs":{"style":{"display":"flex"}},"children":["-0.07",{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[{"name":"div","attribs":{"style":{"flexGrow":1,"display":"flex","alignItems":"center","justifyContent":"flex-end"}},"children":[{"name":"div","attribs":{"style":{"background":"#f5a673","width":"NA%","height":"1rem","marginLeft":"0.8rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[]}]},{"name":"div","attribs":{"style":{"display":"flex"}},"children":["1.00",{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[]},{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[{"name":"div","attribs":{"style":{"flexGrow":1,"display":"flex","alignItems":"center"}},"children":[{"name":"div","attribs":{"style":{"background":"#f5a673","width":"NA%","height":"1rem","marginRight":"0.8rem"}},"children":[]}]}]}]},{"name":"div","attribs":{"style":{"display":"flex"}},"children":["-0.28",{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[{"name":"div","attribs":{"style":{"flexGrow":1,"display":"flex","alignItems":"center","justifyContent":"flex-end"}},"children":[{"name":"div","attribs":{"style":{"background":"#f5a673","width":"NA%","height":"1rem","marginLeft":"0.8rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[]}]},{"name":"div","attribs":{"style":{"display":"flex"}},"children":["1.00",{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[]},{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[{"name":"div","attribs":{"style":{"flexGrow":1,"display":"flex","alignItems":"center"}},"children":[{"name":"div","attribs":{"style":{"background":"#f5a673","width":"NA%","height":"1rem","marginRight":"0.8rem"}},"children":[]}]}]}]},{"name":"div","attribs":{"style":{"display":"flex"}},"children":["1.00",{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[]},{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[{"name":"div","attribs":{"style":{"flexGrow":1,"display":"flex","alignItems":"center"}},"children":[{"name":"div","attribs":{"style":{"background":"#f5a673","width":"NA%","height":"1rem","marginRight":"0.8rem"}},"children":[]}]}]}]},{"name":"div","attribs":{"style":{"display":"flex"}},"children":["NA",{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[]},{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[{"name":"div","attribs":{"style":{"flexGrow":1,"display":"flex","alignItems":"center"}},"children":[{"name":"div","attribs":{"style":{"background":"#f5a673","width":"NA%","height":"1rem","marginRight":"0.8rem"}},"children":[]}]}]}]},{"name":"div","attribs":{"style":{"display":"flex"}},"children":["0.54",{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[]},{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[{"name":"div","attribs":{"style":{"flexGrow":1,"display":"flex","alignItems":"center"}},"children":[{"name":"div","attribs":{"style":{"background":"#f5a673","width":"NA%","height":"1rem","marginRight":"0.8rem"}},"children":[]}]}]}]},{"name":"div","attribs":{"style":{"display":"flex"}},"children":["1.00",{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[]},{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[{"name":"div","attribs":{"style":{"flexGrow":1,"display":"flex","alignItems":"center"}},"children":[{"name":"div","attribs":{"style":{"background":"#f5a673","width":"NA%","height":"1rem","marginRight":"0.8rem"}},"children":[]}]}]}]},{"name":"div","attribs":{"style":{"display":"flex"}},"children":["NA",{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[]},{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[{"name":"div","attribs":{"style":{"flexGrow":1,"display":"flex","alignItems":"center"}},"children":[{"name":"div","attribs":{"style":{"background":"#f5a673","width":"NA%","height":"1rem","marginRight":"0.8rem"}},"children":[]}]}]}]},{"name":"div","attribs":{"style":{"display":"flex"}},"children":["NA",{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[]},{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[{"name":"div","attribs":{"style":{"flexGrow":1,"display":"flex","alignItems":"center"}},"children":[{"name":"div","attribs":{"style":{"background":"#f5a673","width":"NA%","height":"1rem","marginRight":"0.8rem"}},"children":[]}]}]}]},{"name":"div","attribs":{"style":{"display":"flex"}},"children":["1.00",{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[]},{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[{"name":"div","attribs":{"style":{"flexGrow":1,"display":"flex","alignItems":"center"}},"children":[{"name":"div","attribs":{"style":{"background":"#f5a673","width":"NA%","height":"1rem","marginRight":"0.8rem"}},"children":[]}]}]}]},{"name":"div","attribs":{"style":{"display":"flex"}},"children":["NA",{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[]},{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[{"name":"div","attribs":{"style":{"flexGrow":1,"display":"flex","alignItems":"center"}},"children":[{"name":"div","attribs":{"style":{"background":"#f5a673","width":"NA%","height":"1rem","marginRight":"0.8rem"}},"children":[]}]}]}]},{"name":"div","attribs":{"style":{"display":"flex"}},"children":["1.00",{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[]},{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[{"name":"div","attribs":{"style":{"flexGrow":1,"display":"flex","alignItems":"center"}},"children":[{"name":"div","attribs":{"style":{"background":"#f5a673","width":"NA%","height":"1rem","marginRight":"0.8rem"}},"children":[]}]}]}]},{"name":"div","attribs":{"style":{"display":"flex"}},"children":["0.82",{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[]},{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[{"name":"div","attribs":{"style":{"flexGrow":1,"display":"flex","alignItems":"center"}},"children":[{"name":"div","attribs":{"style":{"background":"#f5a673","width":"NA%","height":"1rem","marginRight":"0.8rem"}},"children":[]}]}]}]},{"name":"div","attribs":{"style":{"display":"flex"}},"children":["0.79",{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[]},{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[{"name":"div","attribs":{"style":{"flexGrow":1,"display":"flex","alignItems":"center"}},"children":[{"name":"div","attribs":{"style":{"background":"#f5a673","width":"NA%","height":"1rem","marginRight":"0.8rem"}},"children":[]}]}]}]},{"name":"div","attribs":{"style":{"display":"flex"}},"children":["0.80",{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[]},{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[{"name":"div","attribs":{"style":{"flexGrow":1,"display":"flex","alignItems":"center"}},"children":[{"name":"div","attribs":{"style":{"background":"#f5a673","width":"NA%","height":"1rem","marginRight":"0.8rem"}},"children":[]}]}]}]},{"name":"div","attribs":{"style":{"display":"flex"}},"children":["1.00",{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[]},{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[{"name":"div","attribs":{"style":{"flexGrow":1,"display":"flex","alignItems":"center"}},"children":[{"name":"div","attribs":{"style":{"background":"#f5a673","width":"NA%","height":"1rem","marginRight":"0.8rem"}},"children":[]}]}]}]},{"name":"div","attribs":{"style":{"display":"flex"}},"children":["0.16",{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[]},{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[{"name":"div","attribs":{"style":{"flexGrow":1,"display":"flex","alignItems":"center"}},"children":[{"name":"div","attribs":{"style":{"background":"#f5a673","width":"NA%","height":"1rem","marginRight":"0.8rem"}},"children":[]}]}]}]},{"name":"div","attribs":{"style":{"display":"flex"}},"children":["1.00",{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[]},{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[{"name":"div","attribs":{"style":{"flexGrow":1,"display":"flex","alignItems":"center"}},"children":[{"name":"div","attribs":{"style":{"background":"#f5a673","width":"NA%","height":"1rem","marginRight":"0.8rem"}},"children":[]}]}]}]},{"name":"div","attribs":{"style":{"display":"flex"}},"children":["0.22",{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[]},{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[{"name":"div","attribs":{"style":{"flexGrow":1,"display":"flex","alignItems":"center"}},"children":[{"name":"div","attribs":{"style":{"background":"#f5a673","width":"NA%","height":"1rem","marginRight":"0.8rem"}},"children":[]}]}]}]},{"name":"div","attribs":{"style":{"display":"flex"}},"children":["0.36",{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[]},{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[{"name":"div","attribs":{"style":{"flexGrow":1,"display":"flex","alignItems":"center"}},"children":[{"name":"div","attribs":{"style":{"background":"#f5a673","width":"NA%","height":"1rem","marginRight":"0.8rem"}},"children":[]}]}]}]},{"name":"div","attribs":{"style":{"display":"flex"}},"children":["-0.08",{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[{"name":"div","attribs":{"style":{"flexGrow":1,"display":"flex","alignItems":"center","justifyContent":"flex-end"}},"children":[{"name":"div","attribs":{"style":{"background":"#f5a673","width":"NA%","height":"1rem","marginLeft":"0.8rem"}},"children":[]}]}]},{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[]}]},{"name":"div","attribs":{"style":{"display":"flex"}},"children":["1.00",{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[]},{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[{"name":"div","attribs":{"style":{"flexGrow":1,"display":"flex","alignItems":"center"}},"children":[{"name":"div","attribs":{"style":{"background":"#f5a673","width":"NA%","height":"1rem","marginRight":"0.8rem"}},"children":[]}]}]}]},{"name":"div","attribs":{"style":{"display":"flex"}},"children":["NA",{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[]},{"name":"div","attribs":{"style":{"flex":"1 1 0"}},"children":[{"name":"div","attribs":{"style":{"flexGrow":1,"display":"flex","alignItems":"center"}},"children":[{"name":"div","attribs":{"style":{"background":"#f5a673","width":"NA%","height":"1rem","marginRight":"0.8rem"}},"children":[]}]}]}]}],"html":true,"minWidth":120,"align":"left"}],"defaultPageSize":40,"dataKey":"54d92f2ff8d7e4b637646e3393a7ee28"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</div>

   
   
            
      

  <script>
    $(document).ready(function () {

			
 		
	    });
  </script>



    <!-- dynamically load mathjax for compatibility with self-contained -->
  <script>
    (function () {
	var script = document.createElement("script");
	script.type = "text/javascript";
	script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
	document.getElementsByTagName("head")[0].appendChild(script);
    })();
  </script>
  
</body>
</html>
